{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "1ebf0dbb",
   "metadata": {},
   "source": [
    "# Environment Setup and Data Loading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "b6eff0eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import os\n",
    "import sys\n",
    "import anndata as ad\n",
    "import random\n",
    "import functions_degs\n",
    "import importlib\n",
    "\n",
    "PROJECT_DIR = \"D:/Github/SRF_Linda_RNA\"\n",
    "WORKING_DIR = os.path.join(PROJECT_DIR, \"combine_data\")\n",
    "os.chdir(WORKING_DIR)\n",
    "sys.path.insert(0, WORKING_DIR)\n",
    "\n",
    "importlib.reload(functions_degs)\n",
    "\n",
    "# Set seeds for all random number generators\n",
    "random_seed = 0\n",
    "np.random.seed(random_seed)\n",
    "random.seed(random_seed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "305ffa36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Input directory: D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\n",
      "Plot output directory (Overall/Genotype): D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\plots\n",
      "DGE output directory (Overall/Genotype): D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\n",
      "Biomarker output directory: D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\n"
     ]
    }
   ],
   "source": [
    "# Set up directories\n",
    "\n",
    "# REMOVE_DOUBLETS = True\n",
    "REMOVE_DOUBLETS = False\n",
    "\n",
    "FIX_TRESHOLD = True\n",
    "# FIX_TRESHOLD = False\n",
    "\n",
    "if FIX_TRESHOLD:\n",
    "    BASE_RESULTS_DIR = os.path.join(WORKING_DIR, \"results_from_raw\")\n",
    "else:\n",
    "    if REMOVE_DOUBLETS:\n",
    "        BASE_RESULTS_DIR = os.path.join(WORKING_DIR, \"results_from_raw_percentile_threshold\", \"doublets_removed\")\n",
    "    else:\n",
    "        BASE_RESULTS_DIR = os.path.join(WORKING_DIR, \"results_from_raw_percentile_threshold\")\n",
    "\n",
    "INPUT_DIR = BASE_RESULTS_DIR\n",
    "adata_path = os.path.join(INPUT_DIR, 'annotation_final.h5ad')\n",
    "\n",
    "PARENT_OUTPUT_DIR = os.path.join(INPUT_DIR, \"DEGs_mapmycells_L2\")\n",
    "\n",
    "# CUSTOM_ANALYSIS =  None\n",
    "CUSTOM_ANALYSIS =  \"FC_0_25\"\n",
    "\n",
    "DEG_BY = \"mapmycells_second_layer\"\n",
    "\n",
    "if CUSTOM_ANALYSIS is not None:\n",
    "    PARENT_OUTPUT_DIR = PARENT_OUTPUT_DIR + CUSTOM_ANALYSIS\n",
    "\n",
    "PLOT_OUTPUT_DIR = os.path.join(PARENT_OUTPUT_DIR, 'plots')\n",
    "DGE_OUTPUT_DIR = os.path.join(PARENT_OUTPUT_DIR, 'dge_res')\n",
    "BIOMARKER_OUTPUT_DIR = os.path.join(PARENT_OUTPUT_DIR, 'biomarkers')\n",
    "\n",
    "# Create output directories if they don't exist\n",
    "os.makedirs(PLOT_OUTPUT_DIR, exist_ok=True)\n",
    "os.makedirs(DGE_OUTPUT_DIR, exist_ok=True)\n",
    "os.makedirs(BIOMARKER_OUTPUT_DIR, exist_ok=True)\n",
    "\n",
    "print(f\"Input directory: {INPUT_DIR}\")\n",
    "print(f\"Plot output directory (Overall/Genotype): {PLOT_OUTPUT_DIR}\")\n",
    "print(f\"DGE output directory (Overall/Genotype): {DGE_OUTPUT_DIR}\")\n",
    "print(f\"Biomarker output directory: {BIOMARKER_OUTPUT_DIR}\")\n",
    "\n",
    "# Configure scanpy settings\n",
    "sc.settings.figdir = PLOT_OUTPUT_DIR\n",
    "sc.settings.set_figure_params(dpi=150, facecolor='white')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "31cc9407",
   "metadata": {
    "lines_to_next_cell": 2
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Loading dataset from D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\annotation_final.h5ad\n"
     ]
    }
   ],
   "source": [
    "# Load data\n",
    "print(f\"\\nLoading dataset from {adata_path}\")\n",
    "adata = sc.read_h5ad(adata_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bd447167",
   "metadata": {},
   "source": [
    "# Data Preprocessing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8e80d3c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original mapmycells_second_layer clusters:\n",
      "mapmycells_second_layer\n",
      "DG Glut                               3724\n",
      "CA1-ProS Glut                         3379\n",
      "Oligo NN                              1828\n",
      "CA3 Glut                              1546\n",
      "L2/3 IT ENT Glut                      1231\n",
      "L2/3 IT PPP Glut                      1041\n",
      "Astro-TE NN                           1005\n",
      "L6 CT CTX Glut                         991\n",
      "Microglia NN                           931\n",
      "L2/3 IT CTX Glut                       848\n",
      "MEA Slc17a7 Glut                       823\n",
      "OPC NN                                 822\n",
      "Sst Gaba                               706\n",
      "L2/3 IT PIR-ENTl Glut                  696\n",
      "SUB-ProS Glut                          658\n",
      "LA-BLA-BMA-PA Glut                     640\n",
      "L6b/CT ENT Glut                        545\n",
      "L4/5 IT CTX Glut                       498\n",
      "Pvalb Gaba                             427\n",
      "L2 IT ENT-po Glut                      316\n",
      "Vip Gaba                               315\n",
      "L5/6 IT TPE-ENT Glut                   276\n",
      "L5 ET CTX Glut                         255\n",
      "NP SUB Glut                            246\n",
      "CT SUB Glut                            236\n",
      "RHP-COA Ndnf Gaba                      211\n",
      "Lamp5 Lhx6 Gaba                        206\n",
      "IT AON-TT-DP Glut                      202\n",
      "CA2-FC-IG Glut                         187\n",
      "Lamp5 Gaba                             184\n",
      "ENTmv-PA-COAp Glut                     175\n",
      "L6 IT CTX Glut                         159\n",
      "DG-PIR Ex IMN                          158\n",
      "Sncg Gaba                              149\n",
      "VLMC NN                                137\n",
      "L2 IT PPP-APr Glut                     131\n",
      "IT EP-CLA Glut                         125\n",
      "Endo NN                                122\n",
      "ABC NN                                  87\n",
      "CLA-EPd-CTX Car3 Glut                   86\n",
      "OB-in Frmd7 Gaba                        77\n",
      "L5 NP CTX Glut                          75\n",
      "HPF CR Glut                             73\n",
      "L5 IT CTX Glut                          63\n",
      "OB-STR-CTX Inh IMN                      59\n",
      "Pvalb chandelier Gaba                   57\n",
      "COAp Grxcr2 Glut                        56\n",
      "OB-out Frmd7 Gaba                       55\n",
      "L6b CTX Glut                            50\n",
      "Peri NN                                 46\n",
      "Lymphoid NN                             41\n",
      "OB Meis2 Thsd7b Gaba                    41\n",
      "OEC NN                                  40\n",
      "STR D2 Gaba                             39\n",
      "Astro-NT NN                             36\n",
      "COAa-PAA-MEA Barhl2 Glut                36\n",
      "STR-PAL Chst9 Gaba                      29\n",
      "BAM NN                                  27\n",
      "L2/3 IT RSP Glut                        25\n",
      "OB Dopa-Gaba                            25\n",
      "NP PPP Glut                             24\n",
      "L6b EPd Glut                            21\n",
      "Sst Chodl Gaba                          20\n",
      "NLL Gata3 Gly-Gaba                      20\n",
      "STR D1 Gaba                             19\n",
      "NTS Phox2b Glut                         18\n",
      "SI-MA-LPO-LHA Skor1 Glut                18\n",
      "Ependymal NN                            18\n",
      "OB Eomes Ms4a15 Glut                    17\n",
      "IC Six3 En2 Gaba                        17\n",
      "IC Tfap2d Maf Glut                      17\n",
      "PRT Tcf7l2 Gaba                         17\n",
      "PGRN-PARN-MDRN Hoxb5 Glut               17\n",
      "MEA-BST Sox6 Gaba                       16\n",
      "SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut      16\n",
      "SCsg Gabrr2 Gaba                        15\n",
      "SCig Foxb1 Glut                         15\n",
      "SMC NN                                  15\n",
      "LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba        13\n",
      "DC NN                                   13\n",
      "L5 PPP Glut                             13\n",
      "MH Tac2 Glut                            13\n",
      "LSX Nkx2-1 Gaba                         12\n",
      "CEA-BST Six3 Cyp26b1 Gaba               12\n",
      "LSX Sall3 Lmo1 Gaba                     11\n",
      "Astro-OLF NN                            10\n",
      "STR Prox1 Lhx6 Gaba                     10\n",
      "CHOR NN                                 10\n",
      "SCs Pax7 Nfia Gaba                       9\n",
      "LSX Prdm12 Slit2 Gaba                    9\n",
      "CB Granule Glut                          9\n",
      "IA Mgp Gaba                              9\n",
      "TH Prkcd Grin2c Glut                     9\n",
      "PRNc Otp Gly-Gaba                        8\n",
      "PRT Mecom Gaba                           8\n",
      "MEA-BST Lhx6 Nfib Gaba                   8\n",
      "L4 RSP-ACA Glut                          8\n",
      "SPVI-SPVC Tlx3 Ebf3 Glut                 8\n",
      "SC-PAG Lef1 Emx2 Gaba                    7\n",
      "CEA-BST Ebf1 Pdyn Gaba                   7\n",
      "LSX Prdm12 Zeb2 Gaba                     7\n",
      "NTS-PARN Neurod2 Gly-Gaba                6\n",
      "PAG-SC Neurod2 Meis2 Glut                6\n",
      "MY Lhx1 Gly-Gaba                         6\n",
      "LDT-PCG-CS Gata3 Lhx1 Gaba               5\n",
      "NDB-SI-MA-STRv Lhx8 Gaba                 5\n",
      "TRS-BAC Sln Glut                         5\n",
      "PAG-RN Nkx2-2 Otx1 Gaba                  5\n",
      "OB Trdn Gaba                             5\n",
      "MEA-BST Lhx6 Nr2e1 Gaba                  5\n",
      "ADP-MPO Trp73 Glut                       4\n",
      "PH-SUM Foxa1 Glut                        4\n",
      "Tanycyte NN                              4\n",
      "PVHd-DMH Lhx6 Gaba                       4\n",
      "Monocytes NN                             4\n",
      "MS-SF Bsx Glut                           4\n",
      "LH Pou4f1 Sox1 Glut                      4\n",
      "PRC-PAG Pax6 Glut                        3\n",
      "PRC-PAG Tcf7l2 Irx2 Glut                 3\n",
      "PVR Six3 Sox3 Gaba                       3\n",
      "ZI Pax6 Gaba                             3\n",
      "VCO Mafa Meis2 Glut                      3\n",
      "SPVI-SPVC Sall3 Nfib Gly-Gaba            3\n",
      "NDB-SI-ant Prdm12 Gaba                   3\n",
      "SCm-PAG Cdh23 Gaba                       3\n",
      "PH-LHA Foxb1 Glut                        3\n",
      "MDRNd Prox1 Pax6 Gly-Gaba                3\n",
      "PAG-PPN Pax5 Sox21 Gaba                  3\n",
      "PAG-SC Pou4f1 Zic1 Glut                  3\n",
      "DTN-LDT-IPN Otp Pax3 Gaba                3\n",
      "CB PLI Gly-Gaba                          3\n",
      "LSX Otx2 Gaba                            3\n",
      "MV-SPIV Zic4 Neurod2 Glut                3\n",
      "PG-TRN-LRN Fat2 Glut                     3\n",
      "SC Lef1 Otx2 Gaba                        3\n",
      "Pineal Crx Glut                          3\n",
      "OB-mi Frmd7 Gaba                         3\n",
      "BST-MPN Six3 Nrgn Gaba                   2\n",
      "LHA Barhl2 Glut                          2\n",
      "CEA-AAA-BST Six3 Sp9 Gaba                2\n",
      "AHN-SBPV-PVHd Pdrm12 Gaba                2\n",
      "PVpo-VMPO-MPN Hmx2 Gaba                  2\n",
      "PRP Otp Gly-Gaba                         2\n",
      "SC Tnnt1 Gli3 Gaba                       2\n",
      "MG-POL-SGN Nts Glut                      2\n",
      "MEA-BST Lhx6 Sp9 Gaba                    2\n",
      "MPO-ADP Lhx8 Gaba                        2\n",
      "SPVC Mafa Glut                           2\n",
      "SCsg Pde5a Glut                          2\n",
      "SNc-VTA-RAmb Foxa1 Dopa                  2\n",
      "PVT-PT Ntrk1 Glut                        2\n",
      "SCig-an-PPT Foxb1 Glut                   2\n",
      "OT D3 Folh1 Gaba                         2\n",
      "LDT Fgf7 Gaba                            2\n",
      "DMH Hmx2 Gaba                            2\n",
      "CS-PRNr-DR En1 Sox2 Gaba                 2\n",
      "IF-RL-CLI-PAG Foxa1 Glut                 2\n",
      "NLOT Rho Glut                            2\n",
      "PAG Pou4f2 Mesi2 Glut                    2\n",
      "NI-RPO Gata3 Nr4a2 Gaba                  2\n",
      "PH-ant-LHA Otp Bsx Glut                  2\n",
      "PAG-MRN-RN Foxa2 Gaba                    2\n",
      "PAG-ND-PCG Onecut1 Gaba                  2\n",
      "PARN-MDRNd-NTS Gbx2 Gly-Gaba             2\n",
      "LDT-PCG Vsx2 Lhx4 Glut                   1\n",
      "CS-PRNr-PCG Tmem163 Otp Gaba             1\n",
      "APN C1ql4 Glut                           1\n",
      "Astroependymal NN                        1\n",
      "Bergmann NN                              1\n",
      "CU-ECU-SPVI Foxb1 Glut                   1\n",
      "CS-RPO Meis2 Gaba                        1\n",
      "DMH-LHA Gsx1 Gaba                        1\n",
      "DCO Il22 Gly-Gaba                        1\n",
      "LGv-ZI Otx2 Gaba                         1\n",
      "LDT Vsx2 Nkx6-1 Nfib Glut                1\n",
      "LDT-PCG St18 Gaba                        1\n",
      "PDTg-PCG Pax6 Gaba                       1\n",
      "PPN-CUN-PCG Otp En1 Gaba                 1\n",
      "PH-an Pitx2 Glut                         1\n",
      "MPN-MPO-PVpo Hmx2 Glut                   1\n",
      "MRN-PPN-CUN Pax8 Gaba                    1\n",
      "LSX Sall3 Pax6 Gaba                      1\n",
      "MEA-COA-BMA Ccdc42 Glut                  1\n",
      "MB-MY Tph2 Glut-Sero                     1\n",
      "PB Sst Gly-Gaba                          1\n",
      "MY Prox1 Lmo7 Gly-Gaba                   1\n",
      "MV-SPIV Phox2b Ebf3 Lbx1 Glut            1\n",
      "PVH-SO-PVa Otp Glut                      1\n",
      "PVHd-SBPV Six3 Prox1 Gaba                1\n",
      "SCs Lef1 Gli3 Gaba                       1\n",
      "SCs Dmbx1 Gaba                           1\n",
      "SCop Sln Glut                            1\n",
      "SCH Six6 Cdc14a Gaba                     1\n",
      "RT-ZI Gnb3 Gaba                          1\n",
      "RPA Pax6 Hoxb5 Gly-Gaba                  1\n",
      "SC Bnc2 Glut                             1\n",
      "SBPV-PVa Six6 Satb2 Gaba                 1\n",
      "RN Spp1 Glut                             1\n",
      "STN-PSTN Pitx2 Glut                      1\n",
      "SI-MA-ACB Ebf1 Bnc2 Gaba                 1\n",
      "SI-MPO-LPO Lhx8 Gaba                     1\n",
      "STR D1 Sema5a Gaba                       1\n",
      "STR Lhx8 Gaba                            1\n",
      "TU-ARH Otp Six6 Gaba                     1\n",
      "VMH Nr5a1 Glut                           1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# Save current settings\n",
    "original_max_rows = pd.get_option('display.max_rows')\n",
    "\n",
    "# Set to None to display all rows\n",
    "pd.set_option('display.max_rows', None)\n",
    "\n",
    "# Check original clusters:\n",
    "print(f\"Original {DEG_BY} clusters:\")\n",
    "print(adata.obs[DEG_BY].value_counts())\n",
    "\n",
    "# Reset to original settings\n",
    "pd.set_option('display.max_rows', original_max_rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "f0edc229",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "AnnData object summary:\n",
      "AnnData object with n_obs × n_vars = 28026 × 26870\n",
      "    obs: 'sample', 'condition', 'genotype', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden_0.4', 'ISO_majority_voting', 'ISO_conf_score', 'DG_majority_voting', 'DG_conf_score', 'mapmycells_first_layer', 'mapmycells_second_layer', 'cell_type', 'highlight'\n",
      "    var: 'mt', 'n_cells_by_counts', 'mean_counts', 'pct_dropout_by_counts', 'total_counts', 'n_cells', 'highly_variable', 'highly_variable_rank', 'means', 'variances', 'variances_norm', 'mean', 'std'\n",
      "    uns: 'DG_majority_voting_colors', 'ISO_majority_voting_colors', 'cell_type_colors', 'condition_colors', 'genotype_colors', 'highlight_colors', 'hvg', 'leiden_0.4', 'leiden_0.4_colors', 'log1p', 'neighbors', 'pca', 'sample_colors', 'umap'\n",
      "    obsm: 'X_pca', 'X_umap'\n",
      "    varm: 'PCs'\n",
      "    obsp: 'connectivities', 'distances'\n",
      "\n",
      "Available layers: []\n",
      "Raw data available: True\n",
      "Raw data shape: (28026, 33685)\n"
     ]
    }
   ],
   "source": [
    "# Check adata structure\n",
    "print(\"\\nAnnData object summary:\")\n",
    "print(adata)\n",
    "print(\"\\nAvailable layers:\", list(adata.layers.keys()))\n",
    "print(\"Raw data available:\", adata.raw is not None)\n",
    "if adata.raw:\n",
    "    print(\"Raw data shape:\", adata.raw.X.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0854da00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Creating 'for_DEGs' layer...\n",
      "Normalizing total counts (target_sum=1e4)...\n",
      "Applying log1p transformation...\n",
      "Storing result in adata.layers['for_DEGs']...\n",
      "'for_DEGs' layer created with shape: (28026, 26870)\n"
     ]
    }
   ],
   "source": [
    "# Create the 'for_DEGs' layer: Normalized and log1p transformed counts\n",
    "# Assuming raw counts are in adata.raw.X\n",
    "print(\"\\nCreating 'for_DEGs' layer...\")\n",
    "if adata.raw is not None and adata.raw.X is not None:\n",
    "    # Create a temporary AnnData with raw counts to perform normalization and log1p\n",
    "    adata_for_dge = ad.AnnData(adata.raw.X.copy()) # type: ignore\n",
    "    adata_for_dge.obs_names = adata.obs_names # type: ignore\n",
    "    adata_for_dge.var_names = adata.raw.var_names # type: ignore\n",
    "\n",
    "    print(\"Normalizing total counts (target_sum=1e4)...\")\n",
    "    sc.pp.normalize_total(adata_for_dge, target_sum=1e4)\n",
    "\n",
    "    print(\"Applying log1p transformation...\")\n",
    "    sc.pp.log1p(adata_for_dge)\n",
    "\n",
    "    # Ensure the gene order matches the main adata object\n",
    "    adata_for_dge = adata_for_dge[:, adata.var_names].copy() # type: ignore\n",
    "\n",
    "    print(\"Storing result in adata.layers['for_DEGs']...\")\n",
    "    adata.layers['for_DEGs'] = adata_for_dge.X.copy() # type: ignore\n",
    "    print(\"'for_DEGs' layer created with shape:\", adata.layers['for_DEGs'].shape)\n",
    "else:\n",
    "    print(\"Warning: adata.raw.X not found. Cannot create 'for_DEGs' layer from raw counts.\")\n",
    "    print(\"DGE analysis will proceed using adata.X, which might be scaled.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "4fc8eb1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Metadata check:\n",
      "condition: ['Control', 'Mutant']\n",
      "genotype: ['Emx1', 'Nestin']\n",
      "mapmycells_second_layer: ['Astro-TE NN', 'L6 IT CTX Glut', 'CA3 Glut', 'Monocytes NN', 'L5 ET CTX Glut', 'L2/3 IT PPP Glut', 'CA1-ProS Glut', 'MEA Slc17a7 Glut', 'Astro-OLF NN', 'COAp Grxcr2 Glut', 'LA-BLA-BMA-PA Glut', 'DG Glut', 'Oligo NN', 'NTS Phox2b Glut', 'Sst Gaba', 'L2/3 IT PIR-ENTl Glut', 'SUB-ProS Glut', 'Lymphoid NN', 'L6b CTX Glut', 'L5 NP CTX Glut', 'L2/3 IT CTX Glut', 'L2/3 IT ENT Glut', 'HPF CR Glut', 'Microglia NN', 'Vip Gaba', 'L6b/CT ENT Glut', 'L2 IT ENT-po Glut', 'OB-in Frmd7 Gaba', 'CA2-FC-IG Glut', 'L6 CT CTX Glut', 'RHP-COA Ndnf Gaba', 'SC Lef1 Otx2 Gaba', 'DG-PIR Ex IMN', 'OPC NN', 'Sncg Gaba', 'L4/5 IT CTX Glut', 'IA Mgp Gaba', 'LSX Prdm12 Slit2 Gaba', 'IT AON-TT-DP Glut', 'Lamp5 Lhx6 Gaba', 'CEA-BST Ebf1 Pdyn Gaba', 'Lamp5 Gaba', 'IT EP-CLA Glut', 'Pvalb Gaba', 'VLMC NN', 'L6b EPd Glut', 'OB Meis2 Thsd7b Gaba', 'L5/6 IT TPE-ENT Glut', 'IC Six3 En2 Gaba', 'ENTmv-PA-COAp Glut', 'LDT Vsx2 Nkx6-1 Nfib Glut', 'Astro-NT NN', 'COAa-PAA-MEA Barhl2 Glut', 'L5 IT CTX Glut', 'SMC NN', 'Endo NN', 'ABC NN', 'CLA-EPd-CTX Car3 Glut', 'SPVI-SPVC Sall3 Nfib Gly-Gaba', 'CT SUB Glut', 'Peri NN', 'L2 IT PPP-APr Glut', 'Pvalb chandelier Gaba', 'Ependymal NN', 'NP SUB Glut', 'Tanycyte NN', 'OB-mi Frmd7 Gaba', 'OB Trdn Gaba', 'PVR Six3 Sox3 Gaba', 'MEA-BST Sox6 Gaba', 'STR D2 Gaba', 'CHOR NN', 'IC Tfap2d Maf Glut', 'NI-RPO Gata3 Nr4a2 Gaba', 'NLL Gata3 Gly-Gaba', 'LSX Otx2 Gaba', 'MH Tac2 Glut', 'DTN-LDT-IPN Otp Pax3 Gaba', 'OB Eomes Ms4a15 Glut', 'NTS-PARN Neurod2 Gly-Gaba', 'STR Prox1 Lhx6 Gaba', 'OB-STR-CTX Inh IMN', 'LSX Nkx2-1 Gaba', 'STR D1 Gaba', 'MY Lhx1 Gly-Gaba', 'TRS-BAC Sln Glut', 'CB Granule Glut', 'L2/3 IT RSP Glut', 'PRT Mecom Gaba', 'OB Dopa-Gaba', 'MDRNd Prox1 Pax6 Gly-Gaba', 'MEA-BST Lhx6 Nr2e1 Gaba', 'CB PLI Gly-Gaba', 'BAM NN', 'RT-ZI Gnb3 Gaba', 'SPVC Mafa Glut', 'TH Prkcd Grin2c Glut', 'SNc-VTA-RAmb Foxa1 Dopa', 'PGRN-PARN-MDRN Hoxb5 Glut', 'VCO Mafa Meis2 Glut', 'STR-PAL Chst9 Gaba', 'NP PPP Glut', 'SC-PAG Lef1 Emx2 Gaba', 'DC NN', 'LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba', 'SPVI-SPVC Tlx3 Ebf3 Glut', 'SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut', 'OB-out Frmd7 Gaba', 'CS-RPO Meis2 Gaba', 'AHN-SBPV-PVHd Pdrm12 Gaba', 'LSX Prdm12 Zeb2 Gaba', 'LDT-PCG-CS Gata3 Lhx1 Gaba', 'SCs Pax7 Nfia Gaba', 'ADP-MPO Trp73 Glut', 'Sst Chodl Gaba', 'OEC NN', 'NLOT Rho Glut', 'SCsg Gabrr2 Gaba', 'PAG-ND-PCG Onecut1 Gaba', 'MB-MY Tph2 Glut-Sero', 'PG-TRN-LRN Fat2 Glut', 'LSX Sall3 Lmo1 Gaba', 'PRC-PAG Pax6 Glut', 'SCsg Pde5a Glut', 'MEA-BST Lhx6 Nfib Gaba', 'MEA-BST Lhx6 Sp9 Gaba', 'CEA-BST Six3 Cyp26b1 Gaba', 'Bergmann NN', 'L5 PPP Glut', 'SCig Foxb1 Glut', 'SI-MA-LPO-LHA Skor1 Glut', 'MV-SPIV Zic4 Neurod2 Glut', 'LSX Sall3 Pax6 Gaba', 'NDB-SI-MA-STRv Lhx8 Gaba', 'MG-POL-SGN Nts Glut', 'LGv-ZI Otx2 Gaba', 'PH-SUM Foxa1 Glut', 'PH-LHA Foxb1 Glut', 'LDT-PCG St18 Gaba', 'SCH Six6 Cdc14a Gaba', 'L4 RSP-ACA Glut', 'PRT Tcf7l2 Gaba', 'PRP Otp Gly-Gaba', 'DMH Hmx2 Gaba', 'STN-PSTN Pitx2 Glut', 'OT D3 Folh1 Gaba', 'PAG-RN Nkx2-2 Otx1 Gaba', 'MRN-PPN-CUN Pax8 Gaba', 'SCs Lef1 Gli3 Gaba', 'PDTg-PCG Pax6 Gaba', 'PH-ant-LHA Otp Bsx Glut', 'ZI Pax6 Gaba', 'LDT-PCG Vsx2 Lhx4 Glut', 'CEA-AAA-BST Six3 Sp9 Gaba', 'PAG-SC Neurod2 Meis2 Glut', 'MPN-MPO-PVpo Hmx2 Glut', 'BST-MPN Six3 Nrgn Gaba', 'PAG-PPN Pax5 Sox21 Gaba', 'PRNc Otp Gly-Gaba', 'MPO-ADP Lhx8 Gaba', 'MS-SF Bsx Glut', 'PB Sst Gly-Gaba', 'SI-MA-ACB Ebf1 Bnc2 Gaba', 'NDB-SI-ant Prdm12 Gaba', 'SI-MPO-LPO Lhx8 Gaba', 'Pineal Crx Glut', 'PAG Pou4f2 Mesi2 Glut', 'LHA Barhl2 Glut', 'SBPV-PVa Six6 Satb2 Gaba', 'STR D1 Sema5a Gaba', 'SC Bnc2 Glut', 'SC Tnnt1 Gli3 Gaba', 'PVHd-SBPV Six3 Prox1 Gaba', 'SCm-PAG Cdh23 Gaba', 'SCop Sln Glut', 'CS-PRNr-DR En1 Sox2 Gaba', 'PAG-SC Pou4f1 Zic1 Glut', 'TU-ARH Otp Six6 Gaba', 'VMH Nr5a1 Glut', 'PARN-MDRNd-NTS Gbx2 Gly-Gaba', 'PVpo-VMPO-MPN Hmx2 Gaba', 'MY Prox1 Lmo7 Gly-Gaba', 'PVH-SO-PVa Otp Glut', 'PRC-PAG Tcf7l2 Irx2 Glut', 'IF-RL-CLI-PAG Foxa1 Glut', 'CU-ECU-SPVI Foxb1 Glut', 'STR Lhx8 Gaba', 'MEA-COA-BMA Ccdc42 Glut', 'CS-PRNr-PCG Tmem163 Otp Gaba', 'PVT-PT Ntrk1 Glut', 'PVHd-DMH Lhx6 Gaba', 'LDT Fgf7 Gaba', 'PPN-CUN-PCG Otp En1 Gaba', 'PH-an Pitx2 Glut', 'SCig-an-PPT Foxb1 Glut', 'PAG-MRN-RN Foxa2 Gaba', 'Astroependymal NN', 'MV-SPIV Phox2b Ebf3 Lbx1 Glut', 'SCs Dmbx1 Gaba', 'LH Pou4f1 Sox1 Glut', 'RPA Pax6 Hoxb5 Gly-Gaba', 'DCO Il22 Gly-Gaba', 'DMH-LHA Gsx1 Gaba', 'RN Spp1 Glut', 'APN C1ql4 Glut']\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nMetadata check:\")\n",
    "print(\"condition:\", list(adata.obs.condition.unique()))\n",
    "print(\"genotype:\", list(adata.obs.genotype.unique()))\n",
    "print(f\"{DEG_BY}:\", list(adata.obs[DEG_BY].unique()))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "657f4af3",
   "metadata": {},
   "source": [
    "# Differential Gene Expression Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0e463981",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running Overall DGE (grouped by mapmycells_second_layer)...\n",
      "\n",
      "Starting both genomes Differential Gene Expression analysis (Mutant vs Control)...\n",
      "  Running DGE for group: Astro-TE NN (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 101 cells (max of 3 absolute and 10.0% of 1005)...\n",
      "    Genes after filtering: 5798\n",
      "      Calculating mean expression for 5798 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Astro-TE_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Astro-TE_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Astro-TE_NN' is writable.\n",
      "    Saved 15 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Astro-TE_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 197 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Astro-TE_NN\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Astro-TE NN: ['Mir99ahg', 'Gm28175', 'Grik2', 'Gm29260', 'Nr2f1']\n",
      "---\n",
      "  Running DGE for group: L6 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 16 cells (max of 3 absolute and 10.0% of 159)...\n",
      "    Genes after filtering: 8362\n",
      "      Calculating mean expression for 8362 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_IT_CTX_Glut' is writable.\n",
      "    Saved 1136 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 27 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L6 IT CTX Glut: ['Meg3', 'Sptbn4', 'Slit3', 'Asic2', 'Mtus2']\n",
      "---\n",
      "  Running DGE for group: CA3 Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 155 cells (max of 3 absolute and 10.0% of 1546)...\n",
      "    Genes after filtering: 10200\n",
      "      Calculating mean expression for 10200 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA3_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA3_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA3_Glut' is writable.\n",
      "    Saved 3504 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA3_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 612 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA3_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group CA3 Glut: ['Meg3', 'Anks1b', 'Frmd4a', 'A830082K12Rik', 'Tenm4']\n",
      "---\n",
      "  Running DGE for group: Monocytes NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Monocytes NN: Not enough cells (< 50). Control: 3, Mutant: 1\n",
      "  Running DGE for group: L5 ET CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 26 cells (max of 3 absolute and 10.0% of 255)...\n",
      "    Genes after filtering: 10772\n",
      "      Calculating mean expression for 10772 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L5_ET_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L5_ET_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_ET_CTX_Glut' is writable.\n",
      "    Saved 2278 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_ET_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 48 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_ET_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L5 ET CTX Glut: ['Tenm2', 'Tenm3', 'Meg3', 'Sh3rf1', 'Tenm4']\n",
      "---\n",
      "  Running DGE for group: L2/3 IT PPP Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 105 cells (max of 3 absolute and 10.0% of 1041)...\n",
      "    Genes after filtering: 9771\n",
      "      Calculating mean expression for 9771 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_PPP_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_PPP_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Saved 82 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 120 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L2/3 IT PPP Glut: ['A830082K12Rik', 'Asic2', 'Meg3', 'Tenm3', '2210408I21Rik']\n",
      "---\n",
      "  Running DGE for group: CA1-ProS Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 338 cells (max of 3 absolute and 10.0% of 3379)...\n",
      "    Genes after filtering: 10036\n",
      "      Calculating mean expression for 10036 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA1-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA1-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA1-ProS_Glut' is writable.\n",
      "    Saved 1433 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA1-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1151 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA1-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group CA1-ProS Glut: ['Mir99ahg', 'Meg3', 'Anks1b', 'A830082K12Rik', 'Runx1t1']\n",
      "---\n",
      "  Running DGE for group: MEA Slc17a7 Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 83 cells (max of 3 absolute and 10.0% of 823)...\n",
      "    Genes after filtering: 8859\n",
      "      Calculating mean expression for 8859 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_MEA_Slc17a7_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_MEA_Slc17a7_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\MEA_Slc17a7_Glut' is writable.\n",
      "    Saved 4103 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\MEA_Slc17a7_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 111 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\MEA_Slc17a7_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group MEA Slc17a7 Glut: ['Tenm2', 'Mir99ahg', 'Asic2', 'Csmd1', 'Snhg11']\n",
      "---\n",
      "  Running DGE for group: Astro-OLF NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Astro-OLF NN: Not enough cells (< 50). Control: 8, Mutant: 2\n",
      "  Running DGE for group: COAp Grxcr2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group COAp Grxcr2 Glut: Not enough cells (< 50). Control: 30, Mutant: 26\n",
      "  Running DGE for group: LA-BLA-BMA-PA Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 64 cells (max of 3 absolute and 10.0% of 640)...\n",
      "    Genes after filtering: 8695\n",
      "      Calculating mean expression for 8695 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 2951 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 48 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group LA-BLA-BMA-PA Glut: ['Meg3', 'Snhg11', 'A830082K12Rik', 'Ulk4', 'B830012L14Rik']\n",
      "---\n",
      "  Running DGE for group: DG Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 373 cells (max of 3 absolute and 10.0% of 3724)...\n",
      "    Genes after filtering: 9299\n",
      "      Calculating mean expression for 9299 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_DG_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_DG_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\DG_Glut' is writable.\n",
      "    Saved 370 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\DG_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 2201 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\DG_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group DG Glut: ['Tenm2', 'Frmpd4', 'Grik4', 'Dab1', 'Mir99ahg']\n",
      "---\n",
      "  Running DGE for group: Oligo NN (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 183 cells (max of 3 absolute and 10.0% of 1828)...\n",
      "    Genes after filtering: 4890\n",
      "      Calculating mean expression for 4890 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Oligo_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Oligo_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Oligo_NN' is writable.\n",
      "    Saved 108 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Oligo_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 809 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Oligo_NN\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Oligo NN: ['Brinp3', 'Ralyl', 'Grik2', 'Cadm2', 'Dpp10']\n",
      "---\n",
      "  Running DGE for group: NTS Phox2b Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NTS Phox2b Glut: Not enough cells (< 50). Control: 14, Mutant: 4\n",
      "  Running DGE for group: Sst Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 71 cells (max of 3 absolute and 10.0% of 706)...\n",
      "    Genes after filtering: 9656\n",
      "      Calculating mean expression for 9656 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Sst_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Sst_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sst_Gaba' is writable.\n",
      "    Saved 49 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sst_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 99 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sst_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Sst Gaba: ['Meg3', 'C130071C03Rik', 'A830082K12Rik', 'Pam', 'Mir99ahg']\n",
      "---\n",
      "  Running DGE for group: L2/3 IT PIR-ENTl Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 70 cells (max of 3 absolute and 10.0% of 696)...\n",
      "    Genes after filtering: 9678\n",
      "      Calculating mean expression for 9678 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Saved 106 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 40 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L2/3 IT PIR-ENTl Glut: ['Meg3', 'A830082K12Rik', 'Frmd4a', 'Sfpq', 'Trrap']\n",
      "---\n",
      "  Running DGE for group: SUB-ProS Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 66 cells (max of 3 absolute and 10.0% of 658)...\n",
      "    Genes after filtering: 10903\n",
      "      Calculating mean expression for 10903 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_SUB-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_SUB-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\SUB-ProS_Glut' is writable.\n",
      "    Saved 189 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\SUB-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 234 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\SUB-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group SUB-ProS Glut: ['Meg3', 'A830082K12Rik', 'Runx1t1', 'Nr2f1', 'Mir99ahg']\n",
      "---\n",
      "  Running DGE for group: Lymphoid NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Lymphoid NN: Not enough cells (< 50). Control: 25, Mutant: 16\n",
      "  Running DGE for group: L6b CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L6b CTX Glut: Not enough cells (< 50). Control: 16, Mutant: 34\n",
      "  Running DGE for group: L5 NP CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L5 NP CTX Glut: Not enough cells (< 50). Control: 17, Mutant: 58\n",
      "  Running DGE for group: L2/3 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 85 cells (max of 3 absolute and 10.0% of 848)...\n",
      "    Genes after filtering: 9989\n",
      "      Calculating mean expression for 9989 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Saved 2363 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 112 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L2/3 IT CTX Glut: ['Meg3', 'Pdzrn3', 'Anks1b', 'Grm7', '4930578G10Rik']\n",
      "---\n",
      "  Running DGE for group: L2/3 IT ENT Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 124 cells (max of 3 absolute and 10.0% of 1231)...\n",
      "    Genes after filtering: 10894\n",
      "      Calculating mean expression for 10894 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_3_IT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Saved 424 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 817 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L2/3 IT ENT Glut: ['Meg3', 'Kalrn', 'Mir99ahg', 'Cacna2d3', 'C130071C03Rik']\n",
      "---\n",
      "  Running DGE for group: HPF CR Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group HPF CR Glut: Not enough cells (< 50). Control: 31, Mutant: 42\n",
      "  Running DGE for group: Microglia NN (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 94 cells (max of 3 absolute and 10.0% of 931)...\n",
      "    Genes after filtering: 3706\n",
      "      Calculating mean expression for 3706 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Microglia_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Microglia_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Microglia_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Microglia_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Microglia_NN' is writable.\n",
      "    Saved 84 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Microglia_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 322 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Microglia_NN\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Microglia NN: ['Naaladl2', 'Tenm2', 'Dpp10', 'Ptprd', 'Nrxn1']\n",
      "---\n",
      "  Running DGE for group: Vip Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 32 cells (max of 3 absolute and 10.0% of 315)...\n",
      "    Genes after filtering: 7408\n",
      "      Calculating mean expression for 7408 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Vip_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Vip_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Vip_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Vip_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Vip_Gaba' is writable.\n",
      "    Saved 17 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Vip_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 19 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Vip_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Vip Gaba: ['Mir99ahg', 'Meg3', 'Kcnmb2', 'Tcf4', 'Dlx6os1']\n",
      "---\n",
      "  Running DGE for group: L6b/CT ENT Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 55 cells (max of 3 absolute and 10.0% of 545)...\n",
      "    Genes after filtering: 9151\n",
      "      Calculating mean expression for 9151 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6b_CT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6b_CT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6b_CT_ENT_Glut' is writable.\n",
      "    Saved 17 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6b_CT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 21 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6b_CT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L6b/CT ENT Glut: ['Meg3', 'Naaladl2', 'Anks1b', 'A830082K12Rik', 'Pacsin1']\n",
      "---\n",
      "  Running DGE for group: L2 IT ENT-po Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 32 cells (max of 3 absolute and 10.0% of 316)...\n",
      "    Genes after filtering: 10580\n",
      "      Calculating mean expression for 10580 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_IT_ENT-po_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L2_IT_ENT-po_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_IT_ENT-po_Glut' is writable.\n",
      "    Saved 38 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_IT_ENT-po_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 26 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L2_IT_ENT-po_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L2 IT ENT-po Glut: ['Runx1t1', 'Meg3', 'Gabrg3', 'Cacna2d3', 'Satb2']\n",
      "---\n",
      "  Running DGE for group: OB-in Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB-in Frmd7 Gaba: Not enough cells (< 50). Control: 55, Mutant: 22\n",
      "  Running DGE for group: CA2-FC-IG Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 187)...\n",
      "    Genes after filtering: 9216\n",
      "      Calculating mean expression for 9216 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA2-FC-IG_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CA2-FC-IG_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA2-FC-IG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA2-FC-IG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA2-FC-IG_Glut' is writable.\n",
      "    Saved 696 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA2-FC-IG_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 18 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CA2-FC-IG_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group CA2-FC-IG Glut: ['Mir99ahg', 'Runx1t1', 'Meg3', 'Bach2', '4930578G10Rik']\n",
      "---\n",
      "  Running DGE for group: L6 CT CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 100 cells (max of 3 absolute and 10.0% of 991)...\n",
      "    Genes after filtering: 11712\n",
      "      Calculating mean expression for 11712 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6_CT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L6_CT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 1673 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_CT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 178 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L6_CT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L6 CT CTX Glut: ['Meg3', 'Mir99ahg', 'Tenm4', 'C130071C03Rik', 'A830082K12Rik']\n",
      "---\n",
      "  Running DGE for group: RHP-COA Ndnf Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 22 cells (max of 3 absolute and 10.0% of 211)...\n",
      "    Genes after filtering: 9663\n",
      "      Calculating mean expression for 9663 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_RHP-COA_Ndnf_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_RHP-COA_Ndnf_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\RHP-COA_Ndnf_Gaba' is writable.\n",
      "    Saved 10 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\RHP-COA_Ndnf_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 24 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\RHP-COA_Ndnf_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group RHP-COA Ndnf Gaba: ['Meg3', 'Tenm4', 'Vti1a', 'Srrm1', 'Dpp6']\n",
      "---\n",
      "  Running DGE for group: SC Lef1 Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SC Lef1 Otx2 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: DG-PIR Ex IMN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DG-PIR Ex IMN: Not enough cells (< 50). Control: 116, Mutant: 42\n",
      "  Running DGE for group: OPC NN (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 83 cells (max of 3 absolute and 10.0% of 822)...\n",
      "    Genes after filtering: 5862\n",
      "      Calculating mean expression for 5862 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_OPC_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_OPC_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\OPC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\OPC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\OPC_NN' is writable.\n",
      "    Saved 26 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\OPC_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1409 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\OPC_NN\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group OPC NN: ['Mir99ahg', 'Gm29260', 'Meis2', 'Naaladl2', 'Tenm2']\n",
      "---\n",
      "  Running DGE for group: Sncg Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 15 cells (max of 3 absolute and 10.0% of 149)...\n",
      "    Genes after filtering: 9594\n",
      "      Calculating mean expression for 9594 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Sncg_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Sncg_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sncg_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sncg_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sncg_Gaba' is writable.\n",
      "    Saved 1 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sncg_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Sncg_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Sncg Gaba: ['Snhg14', 'Chd8', 'Meg3', 'Ythdc2', 'Aak1']\n",
      "---\n",
      "  Running DGE for group: L4/5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 50 cells (max of 3 absolute and 10.0% of 498)...\n",
      "    Genes after filtering: 10321\n",
      "      Calculating mean expression for 10321 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L4_5_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L4_5_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Saved 322 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 165 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L4/5 IT CTX Glut: ['Meg3', 'Tenm3', 'Tenm4', 'Naaladl2', 'Tenm2']\n",
      "---\n",
      "  Running DGE for group: IA Mgp Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group IA Mgp Gaba: Not enough cells (< 50). Control: 6, Mutant: 3\n",
      "  Running DGE for group: LSX Prdm12 Slit2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Prdm12 Slit2 Gaba: Not enough cells (< 50). Control: 5, Mutant: 4\n",
      "  Running DGE for group: IT AON-TT-DP Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 21 cells (max of 3 absolute and 10.0% of 202)...\n",
      "    Genes after filtering: 6662\n",
      "      Calculating mean expression for 6662 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_IT_AON-TT-DP_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_IT_AON-TT-DP_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_AON-TT-DP_Glut' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_AON-TT-DP_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    No significant DOWN genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_AON-TT-DP_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group IT AON-TT-DP Glut: ['Kcnma1', 'Epha5', 'Pip4k2a', 'Csmd1', 'Marchf1']\n",
      "---\n",
      "  Running DGE for group: Lamp5 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 21 cells (max of 3 absolute and 10.0% of 206)...\n",
      "    Genes after filtering: 9460\n",
      "      Calculating mean expression for 9460 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Lamp5_Lhx6_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Lamp5_Lhx6_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Lhx6_Gaba' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Lhx6_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 2 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Lhx6_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Lamp5 Lhx6 Gaba: ['Naaladl2', 'Mir99ahg', 'Meg3', 'Dpp6', 'Tenm3']\n",
      "---\n",
      "  Running DGE for group: CEA-BST Ebf1 Pdyn Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CEA-BST Ebf1 Pdyn Gaba: Not enough cells (< 50). Control: 4, Mutant: 3\n",
      "  Running DGE for group: Lamp5 Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 184)...\n",
      "    Genes after filtering: 8817\n",
      "      Calculating mean expression for 8817 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Lamp5_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Lamp5_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Gaba' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Lamp5_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Lamp5 Gaba: ['C130071C03Rik', 'Mir99ahg', 'Nr2f1', 'Hivep2', 'Bcl11a']\n",
      "---\n",
      "  Running DGE for group: IT EP-CLA Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 13 cells (max of 3 absolute and 10.0% of 125)...\n",
      "    Genes after filtering: 10339\n",
      "      Calculating mean expression for 10339 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_IT_EP-CLA_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_IT_EP-CLA_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_EP-CLA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_EP-CLA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_EP-CLA_Glut' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_EP-CLA_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    No significant DOWN genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\IT_EP-CLA_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group IT EP-CLA Glut: ['Nr2f1', 'Gak', 'Kif13b', 'Ep400', 'Slc6a7']\n",
      "---\n",
      "  Running DGE for group: Pvalb Gaba (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 43 cells (max of 3 absolute and 10.0% of 427)...\n",
      "    Genes after filtering: 9258\n",
      "      Calculating mean expression for 9258 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Pvalb_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_Pvalb_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Pvalb_Gaba' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Pvalb_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 5 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\Pvalb_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group Pvalb Gaba: ['Naaladl2', 'Meg3', 'Nbeal1', 'Rbfox1', 'Hexb']\n",
      "---\n",
      "  Running DGE for group: VLMC NN (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 14 cells (max of 3 absolute and 10.0% of 137)...\n",
      "    Genes after filtering: 5199\n",
      "      Calculating mean expression for 5199 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_VLMC_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_VLMC_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\VLMC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\VLMC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\VLMC_NN' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\VLMC_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    No significant DOWN genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\VLMC_NN\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group VLMC NN: ['Ust', 'Pbx3', 'Sncaip', 'Slc6a20a', 'Rplp1']\n",
      "---\n",
      "  Running DGE for group: L6b EPd Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L6b EPd Glut: Not enough cells (< 50). Control: 13, Mutant: 8\n",
      "  Running DGE for group: OB Meis2 Thsd7b Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB Meis2 Thsd7b Gaba: Not enough cells (< 50). Control: 25, Mutant: 16\n",
      "  Running DGE for group: L5/6 IT TPE-ENT Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 28 cells (max of 3 absolute and 10.0% of 276)...\n",
      "    Genes after filtering: 10409\n",
      "      Calculating mean expression for 10409 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L5_6_IT_TPE-ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_L5_6_IT_TPE-ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut' is writable.\n",
      "    Saved 38 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 7 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group L5/6 IT TPE-ENT Glut: ['Meg3', 'Gucy1a2', 'Dpp10', 'Bcl11a', 'A830082K12Rik']\n",
      "---\n",
      "  Running DGE for group: IC Six3 En2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group IC Six3 En2 Gaba: Not enough cells (< 50). Control: 10, Mutant: 7\n",
      "  Running DGE for group: ENTmv-PA-COAp Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 18 cells (max of 3 absolute and 10.0% of 175)...\n",
      "    Genes after filtering: 10369\n",
      "      Calculating mean expression for 10369 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_ENTmv-PA-COAp_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_ENTmv-PA-COAp_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\ENTmv-PA-COAp_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\ENTmv-PA-COAp_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\ENTmv-PA-COAp_Glut' is writable.\n",
      "    Saved 20 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\ENTmv-PA-COAp_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 8 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\ENTmv-PA-COAp_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group ENTmv-PA-COAp Glut: ['A830082K12Rik', 'Nr2f1', 'Rasgef1b', 'Meg3', 'Jazf1']\n",
      "---\n",
      "  Running DGE for group: LDT Vsx2 Nkx6-1 Nfib Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: Astro-NT NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Astro-NT NN: Not enough cells (< 50). Control: 15, Mutant: 21\n",
      "  Running DGE for group: COAa-PAA-MEA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group COAa-PAA-MEA Barhl2 Glut: Not enough cells (< 50). Control: 23, Mutant: 13\n",
      "  Running DGE for group: L5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L5 IT CTX Glut: Not enough cells (< 50). Control: 26, Mutant: 37\n",
      "  Running DGE for group: SMC NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SMC NN: Not enough cells (< 50). Control: 9, Mutant: 6\n",
      "  Running DGE for group: Endo NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Endo NN: Not enough cells (< 50). Control: 73, Mutant: 49\n",
      "  Running DGE for group: ABC NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group ABC NN: Not enough cells (< 50). Control: 42, Mutant: 45\n",
      "  Running DGE for group: CLA-EPd-CTX Car3 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CLA-EPd-CTX Car3 Glut: Not enough cells (< 50). Control: 48, Mutant: 38\n",
      "  Running DGE for group: SPVI-SPVC Sall3 Nfib Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: CT SUB Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 24 cells (max of 3 absolute and 10.0% of 236)...\n",
      "    Genes after filtering: 8842\n",
      "      Calculating mean expression for 8842 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CT_SUB_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_CT_SUB_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CT_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CT_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CT_SUB_Glut' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CT_SUB_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    No significant DOWN genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\CT_SUB_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group CT SUB Glut: ['Samd5', 'Dlc1', '5730522E02Rik', 'Pex5l', 'Gm45159']\n",
      "---\n",
      "  Running DGE for group: Peri NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Peri NN: Not enough cells (< 50). Control: 27, Mutant: 19\n",
      "  Running DGE for group: L2 IT PPP-APr Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L2 IT PPP-APr Glut: Not enough cells (< 50). Control: 27, Mutant: 104\n",
      "  Running DGE for group: Pvalb chandelier Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Pvalb chandelier Gaba: Not enough cells (< 50). Control: 21, Mutant: 36\n",
      "  Running DGE for group: Ependymal NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Ependymal NN: Not enough cells (< 50). Control: 10, Mutant: 8\n",
      "  Running DGE for group: NP SUB Glut (using key 'mapmycells_second_layer')\n",
      "    Initial genes: 26870\n",
      "    Filtering genes: Requiring expression in >= 25 cells (max of 3 absolute and 10.0% of 246)...\n",
      "    Genes after filtering: 9636\n",
      "      Calculating mean expression for 9636 genes...\n",
      "      Using data from layer: for_DEGs\n",
      "Added mean expression columns.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_NP_SUB_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_both_genomes_NP_SUB_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\NP_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\NP_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\NP_SUB_Glut' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\NP_SUB_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 4 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\sig_deg_lists\\NP_SUB_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "    Top 5 genes upregulated in Mutant vs Control for group NP SUB Glut: ['Ralgapb', 'Meg3', 'Etl4', 'Xpo6', 'Snhg14']\n",
      "---\n",
      "  Running DGE for group: Tanycyte NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Tanycyte NN: Not enough cells (< 50). Control: 4, Mutant: 0\n",
      "  Running DGE for group: OB-mi Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB-mi Frmd7 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: OB Trdn Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB Trdn Gaba: Not enough cells (< 50). Control: 4, Mutant: 1\n",
      "  Running DGE for group: PVR Six3 Sox3 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVR Six3 Sox3 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: MEA-BST Sox6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MEA-BST Sox6 Gaba: Not enough cells (< 50). Control: 11, Mutant: 5\n",
      "  Running DGE for group: STR D2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR D2 Gaba: Not enough cells (< 50). Control: 24, Mutant: 15\n",
      "  Running DGE for group: CHOR NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CHOR NN: Not enough cells (< 50). Control: 2, Mutant: 8\n",
      "  Running DGE for group: IC Tfap2d Maf Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group IC Tfap2d Maf Glut: Not enough cells (< 50). Control: 7, Mutant: 10\n",
      "  Running DGE for group: NI-RPO Gata3 Nr4a2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NI-RPO Gata3 Nr4a2 Gaba: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: NLL Gata3 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NLL Gata3 Gly-Gaba: Not enough cells (< 50). Control: 13, Mutant: 7\n",
      "  Running DGE for group: LSX Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Otx2 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: MH Tac2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MH Tac2 Glut: Not enough cells (< 50). Control: 2, Mutant: 11\n",
      "  Running DGE for group: DTN-LDT-IPN Otp Pax3 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells (< 50). Control: 3, Mutant: 0\n",
      "  Running DGE for group: OB Eomes Ms4a15 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB Eomes Ms4a15 Glut: Not enough cells (< 50). Control: 14, Mutant: 3\n",
      "  Running DGE for group: NTS-PARN Neurod2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NTS-PARN Neurod2 Gly-Gaba: Not enough cells (< 50). Control: 2, Mutant: 4\n",
      "  Running DGE for group: STR Prox1 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR Prox1 Lhx6 Gaba: Not enough cells (< 50). Control: 5, Mutant: 5\n",
      "  Running DGE for group: OB-STR-CTX Inh IMN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB-STR-CTX Inh IMN: Not enough cells (< 50). Control: 31, Mutant: 28\n",
      "  Running DGE for group: LSX Nkx2-1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Nkx2-1 Gaba: Not enough cells (< 50). Control: 8, Mutant: 4\n",
      "  Running DGE for group: STR D1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR D1 Gaba: Not enough cells (< 50). Control: 10, Mutant: 9\n",
      "  Running DGE for group: MY Lhx1 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MY Lhx1 Gly-Gaba: Not enough cells (< 50). Control: 3, Mutant: 3\n",
      "  Running DGE for group: TRS-BAC Sln Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group TRS-BAC Sln Glut: Not enough cells (< 50). Control: 1, Mutant: 4\n",
      "  Running DGE for group: CB Granule Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CB Granule Glut: Not enough cells (< 50). Control: 5, Mutant: 4\n",
      "  Running DGE for group: L2/3 IT RSP Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L2/3 IT RSP Glut: Not enough cells (< 50). Control: 11, Mutant: 14\n",
      "  Running DGE for group: PRT Mecom Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRT Mecom Gaba: Not enough cells (< 50). Control: 3, Mutant: 5\n",
      "  Running DGE for group: OB Dopa-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB Dopa-Gaba: Not enough cells (< 50). Control: 21, Mutant: 4\n",
      "  Running DGE for group: MDRNd Prox1 Pax6 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: MEA-BST Lhx6 Nr2e1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells (< 50). Control: 2, Mutant: 3\n",
      "  Running DGE for group: CB PLI Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CB PLI Gly-Gaba: Not enough cells (< 50). Control: 3, Mutant: 0\n",
      "  Running DGE for group: BAM NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group BAM NN: Not enough cells (< 50). Control: 17, Mutant: 10\n",
      "  Running DGE for group: RT-ZI Gnb3 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group RT-ZI Gnb3 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: SPVC Mafa Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SPVC Mafa Glut: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: TH Prkcd Grin2c Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group TH Prkcd Grin2c Glut: Not enough cells (< 50). Control: 1, Mutant: 8\n",
      "  Running DGE for group: SNc-VTA-RAmb Foxa1 Dopa (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SNc-VTA-RAmb Foxa1 Dopa: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: PGRN-PARN-MDRN Hoxb5 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells (< 50). Control: 10, Mutant: 7\n",
      "  Running DGE for group: VCO Mafa Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group VCO Mafa Meis2 Glut: Not enough cells (< 50). Control: 3, Mutant: 0\n",
      "  Running DGE for group: STR-PAL Chst9 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR-PAL Chst9 Gaba: Not enough cells (< 50). Control: 9, Mutant: 20\n",
      "  Running DGE for group: NP PPP Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NP PPP Glut: Not enough cells (< 50). Control: 4, Mutant: 20\n",
      "  Running DGE for group: SC-PAG Lef1 Emx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SC-PAG Lef1 Emx2 Gaba: Not enough cells (< 50). Control: 2, Mutant: 5\n",
      "  Running DGE for group: DC NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DC NN: Not enough cells (< 50). Control: 6, Mutant: 7\n",
      "  Running DGE for group: LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells (< 50). Control: 5, Mutant: 8\n",
      "  Running DGE for group: SPVI-SPVC Tlx3 Ebf3 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells (< 50). Control: 4, Mutant: 4\n",
      "  Running DGE for group: SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells (< 50). Control: 9, Mutant: 7\n",
      "  Running DGE for group: OB-out Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OB-out Frmd7 Gaba: Not enough cells (< 50). Control: 49, Mutant: 6\n",
      "  Running DGE for group: CS-RPO Meis2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CS-RPO Meis2 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: AHN-SBPV-PVHd Pdrm12 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: LSX Prdm12 Zeb2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Prdm12 Zeb2 Gaba: Not enough cells (< 50). Control: 3, Mutant: 4\n",
      "  Running DGE for group: LDT-PCG-CS Gata3 Lhx1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells (< 50). Control: 3, Mutant: 2\n",
      "  Running DGE for group: SCs Pax7 Nfia Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCs Pax7 Nfia Gaba: Not enough cells (< 50). Control: 5, Mutant: 4\n",
      "  Running DGE for group: ADP-MPO Trp73 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group ADP-MPO Trp73 Glut: Not enough cells (< 50). Control: 2, Mutant: 2\n",
      "  Running DGE for group: Sst Chodl Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Sst Chodl Gaba: Not enough cells (< 50). Control: 6, Mutant: 14\n",
      "  Running DGE for group: OEC NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OEC NN: Not enough cells (< 50). Control: 40, Mutant: 0\n",
      "  Running DGE for group: NLOT Rho Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NLOT Rho Glut: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: SCsg Gabrr2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCsg Gabrr2 Gaba: Not enough cells (< 50). Control: 8, Mutant: 7\n",
      "  Running DGE for group: PAG-ND-PCG Onecut1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-ND-PCG Onecut1 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: MB-MY Tph2 Glut-Sero (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MB-MY Tph2 Glut-Sero: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: PG-TRN-LRN Fat2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PG-TRN-LRN Fat2 Glut: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: LSX Sall3 Lmo1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Sall3 Lmo1 Gaba: Not enough cells (< 50). Control: 1, Mutant: 10\n",
      "  Running DGE for group: PRC-PAG Pax6 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRC-PAG Pax6 Glut: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: SCsg Pde5a Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCsg Pde5a Glut: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: MEA-BST Lhx6 Nfib Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MEA-BST Lhx6 Nfib Gaba: Not enough cells (< 50). Control: 4, Mutant: 4\n",
      "  Running DGE for group: MEA-BST Lhx6 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MEA-BST Lhx6 Sp9 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: CEA-BST Six3 Cyp26b1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CEA-BST Six3 Cyp26b1 Gaba: Not enough cells (< 50). Control: 5, Mutant: 7\n",
      "  Running DGE for group: Bergmann NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Bergmann NN: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: L5 PPP Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L5 PPP Glut: Not enough cells (< 50). Control: 2, Mutant: 11\n",
      "  Running DGE for group: SCig Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCig Foxb1 Glut: Not enough cells (< 50). Control: 6, Mutant: 9\n",
      "  Running DGE for group: SI-MA-LPO-LHA Skor1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SI-MA-LPO-LHA Skor1 Glut: Not enough cells (< 50). Control: 7, Mutant: 11\n",
      "  Running DGE for group: MV-SPIV Zic4 Neurod2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MV-SPIV Zic4 Neurod2 Glut: Not enough cells (< 50). Control: 1, Mutant: 2\n",
      "  Running DGE for group: LSX Sall3 Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LSX Sall3 Pax6 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: NDB-SI-MA-STRv Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells (< 50). Control: 2, Mutant: 3\n",
      "  Running DGE for group: MG-POL-SGN Nts Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MG-POL-SGN Nts Glut: Not enough cells (< 50). Control: 0, Mutant: 2\n",
      "  Running DGE for group: LGv-ZI Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LGv-ZI Otx2 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: PH-SUM Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PH-SUM Foxa1 Glut: Not enough cells (< 50). Control: 2, Mutant: 2\n",
      "  Running DGE for group: PH-LHA Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PH-LHA Foxb1 Glut: Not enough cells (< 50). Control: 1, Mutant: 2\n",
      "  Running DGE for group: LDT-PCG St18 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LDT-PCG St18 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: SCH Six6 Cdc14a Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCH Six6 Cdc14a Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: L4 RSP-ACA Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group L4 RSP-ACA Glut: Not enough cells (< 50). Control: 3, Mutant: 5\n",
      "  Running DGE for group: PRT Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRT Tcf7l2 Gaba: Not enough cells (< 50). Control: 6, Mutant: 11\n",
      "  Running DGE for group: PRP Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRP Otp Gly-Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: DMH Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DMH Hmx2 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: STN-PSTN Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STN-PSTN Pitx2 Glut: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: OT D3 Folh1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group OT D3 Folh1 Gaba: Not enough cells (< 50). Control: 0, Mutant: 2\n",
      "  Running DGE for group: PAG-RN Nkx2-2 Otx1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells (< 50). Control: 1, Mutant: 4\n",
      "  Running DGE for group: MRN-PPN-CUN Pax8 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MRN-PPN-CUN Pax8 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: SCs Lef1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCs Lef1 Gli3 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: PDTg-PCG Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PDTg-PCG Pax6 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: PH-ant-LHA Otp Bsx Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PH-ant-LHA Otp Bsx Glut: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: ZI Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group ZI Pax6 Gaba: Not enough cells (< 50). Control: 0, Mutant: 3\n",
      "  Running DGE for group: LDT-PCG Vsx2 Lhx4 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LDT-PCG Vsx2 Lhx4 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: CEA-AAA-BST Six3 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: PAG-SC Neurod2 Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-SC Neurod2 Meis2 Glut: Not enough cells (< 50). Control: 4, Mutant: 2\n",
      "  Running DGE for group: MPN-MPO-PVpo Hmx2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MPN-MPO-PVpo Hmx2 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: BST-MPN Six3 Nrgn Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group BST-MPN Six3 Nrgn Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: PAG-PPN Pax5 Sox21 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-PPN Pax5 Sox21 Gaba: Not enough cells (< 50). Control: 3, Mutant: 0\n",
      "  Running DGE for group: PRNc Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRNc Otp Gly-Gaba: Not enough cells (< 50). Control: 5, Mutant: 3\n",
      "  Running DGE for group: MPO-ADP Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MPO-ADP Lhx8 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: MS-SF Bsx Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MS-SF Bsx Glut: Not enough cells (< 50). Control: 2, Mutant: 2\n",
      "  Running DGE for group: PB Sst Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PB Sst Gly-Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: SI-MA-ACB Ebf1 Bnc2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: NDB-SI-ant Prdm12 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group NDB-SI-ant Prdm12 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: SI-MPO-LPO Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SI-MPO-LPO Lhx8 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: Pineal Crx Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Pineal Crx Glut: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: PAG Pou4f2 Mesi2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG Pou4f2 Mesi2 Glut: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: LHA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LHA Barhl2 Glut: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: SBPV-PVa Six6 Satb2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SBPV-PVa Six6 Satb2 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: STR D1 Sema5a Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR D1 Sema5a Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: SC Bnc2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SC Bnc2 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: SC Tnnt1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SC Tnnt1 Gli3 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: PVHd-SBPV Six3 Prox1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVHd-SBPV Six3 Prox1 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: SCm-PAG Cdh23 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCm-PAG Cdh23 Gaba: Not enough cells (< 50). Control: 2, Mutant: 1\n",
      "  Running DGE for group: SCop Sln Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCop Sln Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: CS-PRNr-DR En1 Sox2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CS-PRNr-DR En1 Sox2 Gaba: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: PAG-SC Pou4f1 Zic1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-SC Pou4f1 Zic1 Glut: Not enough cells (< 50). Control: 1, Mutant: 2\n",
      "  Running DGE for group: TU-ARH Otp Six6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group TU-ARH Otp Six6 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: VMH Nr5a1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group VMH Nr5a1 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: PARN-MDRNd-NTS Gbx2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: PVpo-VMPO-MPN Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells (< 50). Control: 2, Mutant: 0\n",
      "  Running DGE for group: MY Prox1 Lmo7 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MY Prox1 Lmo7 Gly-Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: PVH-SO-PVa Otp Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVH-SO-PVa Otp Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: PRC-PAG Tcf7l2 Irx2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells (< 50). Control: 1, Mutant: 2\n",
      "  Running DGE for group: IF-RL-CLI-PAG Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group IF-RL-CLI-PAG Foxa1 Glut: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: CU-ECU-SPVI Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CU-ECU-SPVI Foxb1 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: STR Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group STR Lhx8 Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: MEA-COA-BMA Ccdc42 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MEA-COA-BMA Ccdc42 Glut: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: CS-PRNr-PCG Tmem163 Otp Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells (< 50). Control: 1, Mutant: 0\n",
      "  Running DGE for group: PVT-PT Ntrk1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVT-PT Ntrk1 Glut: Not enough cells (< 50). Control: 1, Mutant: 1\n",
      "  Running DGE for group: PVHd-DMH Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PVHd-DMH Lhx6 Gaba: Not enough cells (< 50). Control: 0, Mutant: 4\n",
      "  Running DGE for group: LDT Fgf7 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LDT Fgf7 Gaba: Not enough cells (< 50). Control: 0, Mutant: 2\n",
      "  Running DGE for group: PPN-CUN-PCG Otp En1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PPN-CUN-PCG Otp En1 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: PH-an Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PH-an Pitx2 Glut: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: SCig-an-PPT Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCig-an-PPT Foxb1 Glut: Not enough cells (< 50). Control: 0, Mutant: 2\n",
      "  Running DGE for group: PAG-MRN-RN Foxa2 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group PAG-MRN-RN Foxa2 Gaba: Not enough cells (< 50). Control: 0, Mutant: 2\n",
      "  Running DGE for group: Astroependymal NN (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group Astroependymal NN: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: MV-SPIV Phox2b Ebf3 Lbx1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: SCs Dmbx1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group SCs Dmbx1 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: LH Pou4f1 Sox1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group LH Pou4f1 Sox1 Glut: Not enough cells (< 50). Control: 0, Mutant: 4\n",
      "  Running DGE for group: RPA Pax6 Hoxb5 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: DCO Il22 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DCO Il22 Gly-Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: DMH-LHA Gsx1 Gaba (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group DMH-LHA Gsx1 Gaba: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: RN Spp1 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group RN Spp1 Glut: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "  Running DGE for group: APN C1ql4 Glut (using key 'mapmycells_second_layer')\n",
      "    Skipping DGE for group APN C1ql4 Glut: Not enough cells (< 50). Control: 0, Mutant: 1\n",
      "\n",
      "Summary of skipped groups (using key 'mapmycells_second_layer') saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\both_geno_cond_comp\\dge_overall_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "Both genomes DGE analysis finished.\n",
      "\n",
      "Generating Volcano Plots for Overall DGE results...\n",
      "  Generating volcano plot for: Both genomes - Astro-TE NN (Counts: Control=626, Mutant=379) -> dge_conditions_comparison_Astro-TE_NN_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L6 IT CTX Glut (Counts: Control=67, Mutant=92) -> dge_conditions_comparison_L6_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - CA3 Glut (Counts: Control=666, Mutant=880) -> dge_conditions_comparison_CA3_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - Monocytes NN: Not enough cells in one or both groups (< 50). Control=3, Mutant=1.\n",
      "  Generating volcano plot for: Both genomes - L5 ET CTX Glut (Counts: Control=96, Mutant=159) -> dge_conditions_comparison_L5_ET_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L2/3 IT PPP Glut (Counts: Control=189, Mutant=852) -> dge_conditions_comparison_L2_3_IT_PPP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - CA1-ProS Glut (Counts: Control=1467, Mutant=1912) -> dge_conditions_comparison_CA1-ProS_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - MEA Slc17a7 Glut (Counts: Control=532, Mutant=291) -> dge_conditions_comparison_MEA_Slc17a7_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - Astro-OLF NN: Not enough cells in one or both groups (< 50). Control=8, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - COAp Grxcr2 Glut: Not enough cells in one or both groups (< 50). Control=30, Mutant=26.\n",
      "  Generating volcano plot for: Both genomes - LA-BLA-BMA-PA Glut (Counts: Control=352, Mutant=288) -> dge_conditions_comparison_LA-BLA-BMA-PA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - DG Glut (Counts: Control=2400, Mutant=1324) -> dge_conditions_comparison_DG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - Oligo NN (Counts: Control=933, Mutant=895) -> dge_conditions_comparison_Oligo_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - NTS Phox2b Glut: Not enough cells in one or both groups (< 50). Control=14, Mutant=4.\n",
      "  Generating volcano plot for: Both genomes - Sst Gaba (Counts: Control=250, Mutant=456) -> dge_conditions_comparison_Sst_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L2/3 IT PIR-ENTl Glut (Counts: Control=371, Mutant=325) -> dge_conditions_comparison_L2_3_IT_PIR-ENTl_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - SUB-ProS Glut (Counts: Control=267, Mutant=391) -> dge_conditions_comparison_SUB-ProS_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - Lymphoid NN: Not enough cells in one or both groups (< 50). Control=25, Mutant=16.\n",
      "  Skipping volcano plot for Both genomes - L6b CTX Glut: Not enough cells in one or both groups (< 50). Control=16, Mutant=34.\n",
      "  Skipping volcano plot for Both genomes - L5 NP CTX Glut: Not enough cells in one or both groups (< 50). Control=17, Mutant=58.\n",
      "  Generating volcano plot for: Both genomes - L2/3 IT CTX Glut (Counts: Control=297, Mutant=551) -> dge_conditions_comparison_L2_3_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L2/3 IT ENT Glut (Counts: Control=389, Mutant=842) -> dge_conditions_comparison_L2_3_IT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - HPF CR Glut: Not enough cells in one or both groups (< 50). Control=31, Mutant=42.\n",
      "  Generating volcano plot for: Both genomes - Microglia NN (Counts: Control=408, Mutant=523) -> dge_conditions_comparison_Microglia_NN_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - Vip Gaba (Counts: Control=101, Mutant=214) -> dge_conditions_comparison_Vip_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L6b/CT ENT Glut (Counts: Control=190, Mutant=355) -> dge_conditions_comparison_L6b_CT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L2 IT ENT-po Glut (Counts: Control=74, Mutant=242) -> dge_conditions_comparison_L2_IT_ENT-po_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - OB-in Frmd7 Gaba: Not enough cells in one or both groups (< 50). Control=55, Mutant=22.\n",
      "  Generating volcano plot for: Both genomes - CA2-FC-IG Glut (Counts: Control=93, Mutant=94) -> dge_conditions_comparison_CA2-FC-IG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L6 CT CTX Glut (Counts: Control=336, Mutant=655) -> dge_conditions_comparison_L6_CT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - RHP-COA Ndnf Gaba (Counts: Control=79, Mutant=132) -> dge_conditions_comparison_RHP-COA_Ndnf_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - SC Lef1 Otx2 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - DG-PIR Ex IMN: Not enough cells in one or both groups (< 50). Control=116, Mutant=42.\n",
      "  Generating volcano plot for: Both genomes - OPC NN (Counts: Control=334, Mutant=488) -> dge_conditions_comparison_OPC_NN_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - Sncg Gaba (Counts: Control=57, Mutant=92) -> dge_conditions_comparison_Sncg_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - L4/5 IT CTX Glut (Counts: Control=142, Mutant=356) -> dge_conditions_comparison_L4_5_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - IA Mgp Gaba: Not enough cells in one or both groups (< 50). Control=6, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - LSX Prdm12 Slit2 Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=4.\n",
      "  Generating volcano plot for: Both genomes - IT AON-TT-DP Glut (Counts: Control=152, Mutant=50) -> dge_conditions_comparison_IT_AON-TT-DP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - Lamp5 Lhx6 Gaba (Counts: Control=63, Mutant=143) -> dge_conditions_comparison_Lamp5_Lhx6_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - CEA-BST Ebf1 Pdyn Gaba: Not enough cells in one or both groups (< 50). Control=4, Mutant=3.\n",
      "  Generating volcano plot for: Both genomes - Lamp5 Gaba (Counts: Control=66, Mutant=118) -> dge_conditions_comparison_Lamp5_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - IT EP-CLA Glut (Counts: Control=53, Mutant=72) -> dge_conditions_comparison_IT_EP-CLA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - Pvalb Gaba (Counts: Control=144, Mutant=283) -> dge_conditions_comparison_Pvalb_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Both genomes - VLMC NN (Counts: Control=70, Mutant=67) -> dge_conditions_comparison_VLMC_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - L6b EPd Glut: Not enough cells in one or both groups (< 50). Control=13, Mutant=8.\n",
      "  Skipping volcano plot for Both genomes - OB Meis2 Thsd7b Gaba: Not enough cells in one or both groups (< 50). Control=25, Mutant=16.\n",
      "  Generating volcano plot for: Both genomes - L5/6 IT TPE-ENT Glut (Counts: Control=93, Mutant=183) -> dge_conditions_comparison_L5_6_IT_TPE-ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - IC Six3 En2 Gaba: Not enough cells in one or both groups (< 50). Control=10, Mutant=7.\n",
      "  Generating volcano plot for: Both genomes - ENTmv-PA-COAp Glut (Counts: Control=76, Mutant=99) -> dge_conditions_comparison_ENTmv-PA-COAp_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - Astro-NT NN: Not enough cells in one or both groups (< 50). Control=15, Mutant=21.\n",
      "  Skipping volcano plot for Both genomes - COAa-PAA-MEA Barhl2 Glut: Not enough cells in one or both groups (< 50). Control=23, Mutant=13.\n",
      "  Skipping volcano plot for Both genomes - L5 IT CTX Glut: Not enough cells in one or both groups (< 50). Control=26, Mutant=37.\n",
      "  Skipping volcano plot for Both genomes - SMC NN: Not enough cells in one or both groups (< 50). Control=9, Mutant=6.\n",
      "  Skipping volcano plot for Both genomes - Endo NN: Not enough cells in one or both groups (< 50). Control=73, Mutant=49.\n",
      "  Skipping volcano plot for Both genomes - ABC NN: Not enough cells in one or both groups (< 50). Control=42, Mutant=45.\n",
      "  Skipping volcano plot for Both genomes - CLA-EPd-CTX Car3 Glut: Not enough cells in one or both groups (< 50). Control=48, Mutant=38.\n",
      "  Skipping volcano plot for Both genomes - SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Generating volcano plot for: Both genomes - CT SUB Glut (Counts: Control=76, Mutant=160) -> dge_conditions_comparison_CT_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - Peri NN: Not enough cells in one or both groups (< 50). Control=27, Mutant=19.\n",
      "  Skipping volcano plot for Both genomes - L2 IT PPP-APr Glut: Not enough cells in one or both groups (< 50). Control=27, Mutant=104.\n",
      "  Skipping volcano plot for Both genomes - Pvalb chandelier Gaba: Not enough cells in one or both groups (< 50). Control=21, Mutant=36.\n",
      "  Skipping volcano plot for Both genomes - Ependymal NN: Not enough cells in one or both groups (< 50). Control=10, Mutant=8.\n",
      "  Generating volcano plot for: Both genomes - NP SUB Glut (Counts: Control=105, Mutant=141) -> dge_conditions_comparison_NP_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Both genomes - Tanycyte NN: Not enough cells in one or both groups (< 50). Control=4, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - OB-mi Frmd7 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - OB Trdn Gaba: Not enough cells in one or both groups (< 50). Control=4, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PVR Six3 Sox3 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MEA-BST Sox6 Gaba: Not enough cells in one or both groups (< 50). Control=11, Mutant=5.\n",
      "  Skipping volcano plot for Both genomes - STR D2 Gaba: Not enough cells in one or both groups (< 50). Control=24, Mutant=15.\n",
      "  Skipping volcano plot for Both genomes - CHOR NN: Not enough cells in one or both groups (< 50). Control=2, Mutant=8.\n",
      "  Skipping volcano plot for Both genomes - IC Tfap2d Maf Glut: Not enough cells in one or both groups (< 50). Control=7, Mutant=10.\n",
      "  Skipping volcano plot for Both genomes - NI-RPO Gata3 Nr4a2 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - NLL Gata3 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=13, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - LSX Otx2 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MH Tac2 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=11.\n",
      "  Skipping volcano plot for Both genomes - DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - OB Eomes Ms4a15 Glut: Not enough cells in one or both groups (< 50). Control=14, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - NTS-PARN Neurod2 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - STR Prox1 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=5.\n",
      "  Skipping volcano plot for Both genomes - OB-STR-CTX Inh IMN: Not enough cells in one or both groups (< 50). Control=31, Mutant=28.\n",
      "  Skipping volcano plot for Both genomes - LSX Nkx2-1 Gaba: Not enough cells in one or both groups (< 50). Control=8, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - STR D1 Gaba: Not enough cells in one or both groups (< 50). Control=10, Mutant=9.\n",
      "  Skipping volcano plot for Both genomes - MY Lhx1 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - TRS-BAC Sln Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - CB Granule Glut: Not enough cells in one or both groups (< 50). Control=5, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - L2/3 IT RSP Glut: Not enough cells in one or both groups (< 50). Control=11, Mutant=14.\n",
      "  Skipping volcano plot for Both genomes - PRT Mecom Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=5.\n",
      "  Skipping volcano plot for Both genomes - OB Dopa-Gaba: Not enough cells in one or both groups (< 50). Control=21, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - CB PLI Gly-Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - BAM NN: Not enough cells in one or both groups (< 50). Control=17, Mutant=10.\n",
      "  Skipping volcano plot for Both genomes - RT-ZI Gnb3 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SPVC Mafa Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - TH Prkcd Grin2c Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=8.\n",
      "  Skipping volcano plot for Both genomes - SNc-VTA-RAmb Foxa1 Dopa: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells in one or both groups (< 50). Control=10, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - VCO Mafa Meis2 Glut: Not enough cells in one or both groups (< 50). Control=3, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - STR-PAL Chst9 Gaba: Not enough cells in one or both groups (< 50). Control=9, Mutant=20.\n",
      "  Skipping volcano plot for Both genomes - NP PPP Glut: Not enough cells in one or both groups (< 50). Control=4, Mutant=20.\n",
      "  Skipping volcano plot for Both genomes - SC-PAG Lef1 Emx2 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=5.\n",
      "  Skipping volcano plot for Both genomes - DC NN: Not enough cells in one or both groups (< 50). Control=6, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=8.\n",
      "  Skipping volcano plot for Both genomes - SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells in one or both groups (< 50). Control=4, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells in one or both groups (< 50). Control=9, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - OB-out Frmd7 Gaba: Not enough cells in one or both groups (< 50). Control=49, Mutant=6.\n",
      "  Skipping volcano plot for Both genomes - CS-RPO Meis2 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - LSX Prdm12 Zeb2 Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - SCs Pax7 Nfia Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - ADP-MPO Trp73 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - Sst Chodl Gaba: Not enough cells in one or both groups (< 50). Control=6, Mutant=14.\n",
      "  Skipping volcano plot for Both genomes - OEC NN: Not enough cells in one or both groups (< 50). Control=40, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - NLOT Rho Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SCsg Gabrr2 Gaba: Not enough cells in one or both groups (< 50). Control=8, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - PAG-ND-PCG Onecut1 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MB-MY Tph2 Glut-Sero: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PG-TRN-LRN Fat2 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - LSX Sall3 Lmo1 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=10.\n",
      "  Skipping volcano plot for Both genomes - PRC-PAG Pax6 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCsg Pde5a Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - MEA-BST Lhx6 Nfib Gaba: Not enough cells in one or both groups (< 50). Control=4, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - MEA-BST Lhx6 Sp9 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - CEA-BST Six3 Cyp26b1 Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=7.\n",
      "  Skipping volcano plot for Both genomes - Bergmann NN: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - L5 PPP Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=11.\n",
      "  Skipping volcano plot for Both genomes - SCig Foxb1 Glut: Not enough cells in one or both groups (< 50). Control=6, Mutant=9.\n",
      "  Skipping volcano plot for Both genomes - SI-MA-LPO-LHA Skor1 Glut: Not enough cells in one or both groups (< 50). Control=7, Mutant=11.\n",
      "  Skipping volcano plot for Both genomes - MV-SPIV Zic4 Neurod2 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - LSX Sall3 Pax6 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - MG-POL-SGN Nts Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - LGv-ZI Otx2 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PH-SUM Foxa1 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - PH-LHA Foxb1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - LDT-PCG St18 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCH Six6 Cdc14a Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - L4 RSP-ACA Glut: Not enough cells in one or both groups (< 50). Control=3, Mutant=5.\n",
      "  Skipping volcano plot for Both genomes - PRT Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Control=6, Mutant=11.\n",
      "  Skipping volcano plot for Both genomes - PRP Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - DMH Hmx2 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - STN-PSTN Pitx2 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - OT D3 Folh1 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - MRN-PPN-CUN Pax8 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCs Lef1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PDTg-PCG Pax6 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PH-ant-LHA Otp Bsx Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - ZI Pax6 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - LDT-PCG Vsx2 Lhx4 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PAG-SC Neurod2 Meis2 Glut: Not enough cells in one or both groups (< 50). Control=4, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - MPN-MPO-PVpo Hmx2 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - BST-MPN Six3 Nrgn Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PAG-PPN Pax5 Sox21 Gaba: Not enough cells in one or both groups (< 50). Control=3, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PRNc Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Control=5, Mutant=3.\n",
      "  Skipping volcano plot for Both genomes - MPO-ADP Lhx8 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MS-SF Bsx Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - PB Sst Gly-Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - NDB-SI-ant Prdm12 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SI-MPO-LPO Lhx8 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - Pineal Crx Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PAG Pou4f2 Mesi2 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - LHA Barhl2 Glut: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SBPV-PVa Six6 Satb2 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - STR D1 Sema5a Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SC Bnc2 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SC Tnnt1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PVHd-SBPV Six3 Prox1 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - SCm-PAG Cdh23 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCop Sln Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - CS-PRNr-DR En1 Sox2 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PAG-SC Pou4f1 Zic1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - TU-ARH Otp Six6 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - VMH Nr5a1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells in one or both groups (< 50). Control=2, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - MY Prox1 Lmo7 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PVH-SO-PVa Otp Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - IF-RL-CLI-PAG Foxa1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - CU-ECU-SPVI Foxb1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - STR Lhx8 Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - MEA-COA-BMA Ccdc42 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells in one or both groups (< 50). Control=1, Mutant=0.\n",
      "  Skipping volcano plot for Both genomes - PVT-PT Ntrk1 Glut: Not enough cells in one or both groups (< 50). Control=1, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PVHd-DMH Lhx6 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - LDT Fgf7 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - PPN-CUN-PCG Otp En1 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - PH-an Pitx2 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCig-an-PPT Foxb1 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - PAG-MRN-RN Foxa2 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=2.\n",
      "  Skipping volcano plot for Both genomes - Astroependymal NN: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - SCs Dmbx1 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - LH Pou4f1 Sox1 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=4.\n",
      "  Skipping volcano plot for Both genomes - RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - DCO Il22 Gly-Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - DMH-LHA Gsx1 Gaba: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - RN Spp1 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "  Skipping volcano plot for Both genomes - APN C1ql4 Glut: Not enough cells in one or both groups (< 50). Control=0, Mutant=1.\n",
      "\n",
      "Both genomes DGE visualization finished.\n"
     ]
    }
   ],
   "source": [
    "# Overall DGE (Mutant vs Control) using the 'for_DEGs' layer, grouped by the DEG_BY\n",
    "print(f\"\\nRunning Overall DGE (grouped by {DEG_BY})...\")\n",
    "dge_results = functions_degs.run_overall_dge(\n",
    "    adata,\n",
    "    grouping_key=DEG_BY,\n",
    "    dge_output_dir=DGE_OUTPUT_DIR,\n",
    "    plot_output_dir=PLOT_OUTPUT_DIR,\n",
    "    layer='for_DEGs' if 'for_DEGs' in adata.layers else None\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a2daa24a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running Genotype-Specific DGE (grouped by mapmycells_second_layer)...\n",
      "\n",
      "Starting Genotype-Specific DGE analysis...\n",
      "Created 'genotype_condition' column with categories: ['Emx1_Control', 'Emx1_Mutant', 'Nestin_Control', 'Nestin_Mutant']\n",
      "\n",
      "Processing group: Astro-TE NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 38 cells (max of 3 absolute and 10.0% of 372)...\n",
      "    Genes after filtering: 5030\n",
      "      Calculating mean expression for 5030 genes (Emx1, group Astro-TE NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group Astro-TE NN).\n",
      "    Finished DGE for Emx1, group Astro-TE NN. Found 5030 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Astro-TE_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Astro-TE_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Astro-TE_NN' is writable.\n",
      "    Saved 56 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Astro-TE_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 26 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Astro-TE_NN\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 64 cells (max of 3 absolute and 10.0% of 633)...\n",
      "    Genes after filtering: 6067\n",
      "      Calculating mean expression for 6067 genes (Nestin, group Astro-TE NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Astro-TE NN).\n",
      "    Finished DGE for Nestin, group Astro-TE NN. Found 6067 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Astro-TE_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Astro-TE_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Astro-TE_NN' is writable.\n",
      "    Saved 47 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Astro-TE_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 329 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Astro-TE_NN\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L6 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L6 IT CTX Glut: Not enough cells (< 50). Emx1_Control: 36, Emx1_Mutant: 37\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L6 IT CTX Glut: Not enough cells (< 50). Nestin_Control: 31, Nestin_Mutant: 55\n",
      "\n",
      "Processing group: CA3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 85 cells (max of 3 absolute and 10.0% of 842)...\n",
      "    Genes after filtering: 10070\n",
      "      Calculating mean expression for 10070 genes (Emx1, group CA3 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group CA3 Glut).\n",
      "    Finished DGE for Emx1, group CA3 Glut. Found 10070 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_CA3_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_CA3_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA3_Glut' is writable.\n",
      "    Saved 3517 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA3_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 461 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA3_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 71 cells (max of 3 absolute and 10.0% of 704)...\n",
      "    Genes after filtering: 10260\n",
      "      Calculating mean expression for 10260 genes (Nestin, group CA3 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group CA3 Glut).\n",
      "    Finished DGE for Nestin, group CA3 Glut. Found 10260 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CA3_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CA3_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA3_Glut' is writable.\n",
      "    Saved 1820 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA3_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 356 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA3_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Monocytes NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Monocytes NN: Not enough cells (< 50). Emx1_Control: 3, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Monocytes NN: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: L5 ET CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L5 ET CTX Glut: Not enough cells (< 50). Emx1_Control: 46, Emx1_Mutant: 45\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 17 cells (max of 3 absolute and 10.0% of 164)...\n",
      "    Genes after filtering: 10836\n",
      "      Calculating mean expression for 10836 genes (Nestin, group L5 ET CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L5 ET CTX Glut).\n",
      "    Finished DGE for Nestin, group L5 ET CTX Glut. Found 10836 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L5_ET_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L5_ET_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_ET_CTX_Glut' is writable.\n",
      "    Saved 1329 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_ET_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 17 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_ET_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L2/3 IT PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 47 cells (max of 3 absolute and 10.0% of 464)...\n",
      "    Genes after filtering: 9760\n",
      "      Calculating mean expression for 9760 genes (Emx1, group L2/3 IT PPP Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L2/3 IT PPP Glut).\n",
      "    Finished DGE for Emx1, group L2/3 IT PPP Glut. Found 9760 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_PPP_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_PPP_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Saved 31 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 42 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 58 cells (max of 3 absolute and 10.0% of 577)...\n",
      "    Genes after filtering: 9802\n",
      "      Calculating mean expression for 9802 genes (Nestin, group L2/3 IT PPP Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L2/3 IT PPP Glut).\n",
      "    Finished DGE for Nestin, group L2/3 IT PPP Glut. Found 9802 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_PPP_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_PPP_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Saved 118 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 200 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PPP_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: CA1-ProS Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 142 cells (max of 3 absolute and 10.0% of 1415)...\n",
      "    Genes after filtering: 10012\n",
      "      Calculating mean expression for 10012 genes (Emx1, group CA1-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group CA1-ProS Glut).\n",
      "    Finished DGE for Emx1, group CA1-ProS Glut. Found 10012 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_CA1-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_CA1-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA1-ProS_Glut' is writable.\n",
      "    Saved 1113 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA1-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 726 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\CA1-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 197 cells (max of 3 absolute and 10.0% of 1964)...\n",
      "    Genes after filtering: 9980\n",
      "      Calculating mean expression for 9980 genes (Nestin, group CA1-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group CA1-ProS Glut).\n",
      "    Finished DGE for Nestin, group CA1-ProS Glut. Found 9980 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CA1-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CA1-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA1-ProS_Glut' is writable.\n",
      "    Saved 1288 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA1-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 697 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CA1-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: MEA Slc17a7 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 18 cells (max of 3 absolute and 10.0% of 171)...\n",
      "    Genes after filtering: 9973\n",
      "      Calculating mean expression for 9973 genes (Emx1, group MEA Slc17a7 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group MEA Slc17a7 Glut).\n",
      "    Finished DGE for Emx1, group MEA Slc17a7 Glut. Found 9973 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_MEA_Slc17a7_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_MEA_Slc17a7_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\MEA_Slc17a7_Glut' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\MEA_Slc17a7_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 8 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\MEA_Slc17a7_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 66 cells (max of 3 absolute and 10.0% of 652)...\n",
      "    Genes after filtering: 8454\n",
      "      Calculating mean expression for 8454 genes (Nestin, group MEA Slc17a7 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group MEA Slc17a7 Glut).\n",
      "    Finished DGE for Nestin, group MEA Slc17a7 Glut. Found 8454 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_MEA_Slc17a7_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_MEA_Slc17a7_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\MEA_Slc17a7_Glut' is writable.\n",
      "    Saved 4236 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\MEA_Slc17a7_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 95 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\MEA_Slc17a7_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Astro-OLF NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Astro-OLF NN: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Astro-OLF NN: Not enough cells (< 50). Nestin_Control: 6, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: COAp Grxcr2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group COAp Grxcr2 Glut: Not enough cells (< 50). Emx1_Control: 23, Emx1_Mutant: 25\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group COAp Grxcr2 Glut: Not enough cells (< 50). Nestin_Control: 7, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LA-BLA-BMA-PA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 33 cells (max of 3 absolute and 10.0% of 321)...\n",
      "    Genes after filtering: 9074\n",
      "      Calculating mean expression for 9074 genes (Emx1, group LA-BLA-BMA-PA Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group LA-BLA-BMA-PA Glut).\n",
      "    Finished DGE for Emx1, group LA-BLA-BMA-PA Glut. Found 9074 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 228 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 36 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 32 cells (max of 3 absolute and 10.0% of 319)...\n",
      "    Genes after filtering: 8202\n",
      "      Calculating mean expression for 8202 genes (Nestin, group LA-BLA-BMA-PA Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group LA-BLA-BMA-PA Glut).\n",
      "    Finished DGE for Nestin, group LA-BLA-BMA-PA Glut. Found 8202 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_LA-BLA-BMA-PA_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 1298 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 69 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\LA-BLA-BMA-PA_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: DG Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 125 cells (max of 3 absolute and 10.0% of 1248)...\n",
      "    Genes after filtering: 9512\n",
      "      Calculating mean expression for 9512 genes (Emx1, group DG Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group DG Glut).\n",
      "    Finished DGE for Emx1, group DG Glut. Found 9512 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_DG_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_DG_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\DG_Glut' is writable.\n",
      "    Saved 503 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\DG_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1048 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\DG_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 248 cells (max of 3 absolute and 10.0% of 2476)...\n",
      "    Genes after filtering: 9166\n",
      "      Calculating mean expression for 9166 genes (Nestin, group DG Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group DG Glut).\n",
      "    Finished DGE for Nestin, group DG Glut. Found 9166 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_DG_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_DG_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\DG_Glut' is writable.\n",
      "    Saved 335 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\DG_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 2051 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\DG_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Oligo NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 28 cells (max of 3 absolute and 10.0% of 275)...\n",
      "    Genes after filtering: 4515\n",
      "      Calculating mean expression for 4515 genes (Emx1, group Oligo NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group Oligo NN).\n",
      "    Finished DGE for Emx1, group Oligo NN. Found 4515 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Oligo_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Oligo_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Oligo_NN' is writable.\n",
      "    Saved 7 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Oligo_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 4 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Oligo_NN\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 156 cells (max of 3 absolute and 10.0% of 1553)...\n",
      "    Genes after filtering: 4890\n",
      "      Calculating mean expression for 4890 genes (Nestin, group Oligo NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Oligo NN).\n",
      "    Finished DGE for Nestin, group Oligo NN. Found 4890 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Oligo_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Oligo_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Oligo_NN' is writable.\n",
      "    Saved 109 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Oligo_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 927 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Oligo_NN\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: NTS Phox2b Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NTS Phox2b Glut: Not enough cells (< 50). Emx1_Control: 6, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NTS Phox2b Glut: Not enough cells (< 50). Nestin_Control: 8, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Sst Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 31 cells (max of 3 absolute and 10.0% of 301)...\n",
      "    Genes after filtering: 9685\n",
      "      Calculating mean expression for 9685 genes (Emx1, group Sst Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group Sst Gaba).\n",
      "    Finished DGE for Emx1, group Sst Gaba. Found 9685 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Sst_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Sst_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Sst_Gaba' is writable.\n",
      "    Saved 34 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Sst_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 55 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Sst_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 41 cells (max of 3 absolute and 10.0% of 405)...\n",
      "    Genes after filtering: 9530\n",
      "      Calculating mean expression for 9530 genes (Nestin, group Sst Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Sst Gaba).\n",
      "    Finished DGE for Nestin, group Sst Gaba. Found 9530 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Sst_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Sst_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Sst_Gaba' is writable.\n",
      "    Saved 45 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Sst_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 50 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Sst_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L2/3 IT PIR-ENTl Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 31 cells (max of 3 absolute and 10.0% of 308)...\n",
      "    Genes after filtering: 10131\n",
      "      Calculating mean expression for 10131 genes (Emx1, group L2/3 IT PIR-ENTl Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L2/3 IT PIR-ENTl Glut).\n",
      "    Finished DGE for Emx1, group L2/3 IT PIR-ENTl Glut. Found 10131 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Saved 16 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 81 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 39 cells (max of 3 absolute and 10.0% of 388)...\n",
      "    Genes after filtering: 9288\n",
      "      Calculating mean expression for 9288 genes (Nestin, group L2/3 IT PIR-ENTl Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L2/3 IT PIR-ENTl Glut).\n",
      "    Finished DGE for Nestin, group L2/3 IT PIR-ENTl Glut. Found 9288 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_PIR-ENTl_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Saved 330 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 59 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_PIR-ENTl_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: SUB-ProS Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 29 cells (max of 3 absolute and 10.0% of 287)...\n",
      "    Genes after filtering: 10548\n",
      "      Calculating mean expression for 10548 genes (Emx1, group SUB-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group SUB-ProS Glut).\n",
      "    Finished DGE for Emx1, group SUB-ProS Glut. Found 10548 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_SUB-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_SUB-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\SUB-ProS_Glut' is writable.\n",
      "    Saved 228 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\SUB-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 87 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\SUB-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 38 cells (max of 3 absolute and 10.0% of 371)...\n",
      "    Genes after filtering: 11102\n",
      "      Calculating mean expression for 11102 genes (Nestin, group SUB-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group SUB-ProS Glut).\n",
      "    Finished DGE for Nestin, group SUB-ProS Glut. Found 11102 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_SUB-ProS_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_SUB-ProS_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\SUB-ProS_Glut' is writable.\n",
      "    Saved 95 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\SUB-ProS_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 201 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\SUB-ProS_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Lymphoid NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Lymphoid NN: Not enough cells (< 50). Emx1_Control: 11, Emx1_Mutant: 10\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Lymphoid NN: Not enough cells (< 50). Nestin_Control: 14, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: L6b CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L6b CTX Glut: Not enough cells (< 50). Emx1_Control: 12, Emx1_Mutant: 13\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L6b CTX Glut: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: L5 NP CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L5 NP CTX Glut: Not enough cells (< 50). Emx1_Control: 8, Emx1_Mutant: 13\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L5 NP CTX Glut: Not enough cells (< 50). Nestin_Control: 9, Nestin_Mutant: 45\n",
      "\n",
      "Processing group: L2/3 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 28 cells (max of 3 absolute and 10.0% of 278)...\n",
      "    Genes after filtering: 10216\n",
      "      Calculating mean expression for 10216 genes (Emx1, group L2/3 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L2/3 IT CTX Glut).\n",
      "    Finished DGE for Emx1, group L2/3 IT CTX Glut. Found 10216 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Saved 166 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 16 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 57 cells (max of 3 absolute and 10.0% of 570)...\n",
      "    Genes after filtering: 9886\n",
      "      Calculating mean expression for 9886 genes (Nestin, group L2/3 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L2/3 IT CTX Glut).\n",
      "    Finished DGE for Nestin, group L2/3 IT CTX Glut. Found 9886 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Saved 2157 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 173 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L2/3 IT ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 57 cells (max of 3 absolute and 10.0% of 568)...\n",
      "    Genes after filtering: 10892\n",
      "      Calculating mean expression for 10892 genes (Emx1, group L2/3 IT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L2/3 IT ENT Glut).\n",
      "    Finished DGE for Emx1, group L2/3 IT ENT Glut. Found 10892 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L2_3_IT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Saved 687 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 547 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 67 cells (max of 3 absolute and 10.0% of 663)...\n",
      "    Genes after filtering: 10901\n",
      "      Calculating mean expression for 10901 genes (Nestin, group L2/3 IT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L2/3 IT ENT Glut).\n",
      "    Finished DGE for Nestin, group L2/3 IT ENT Glut. Found 10901 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L2_3_IT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Saved 154 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 360 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L2_3_IT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: HPF CR Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group HPF CR Glut: Not enough cells (< 50). Emx1_Control: 6, Emx1_Mutant: 5\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group HPF CR Glut: Not enough cells (< 50). Nestin_Control: 25, Nestin_Mutant: 37\n",
      "\n",
      "Processing group: Microglia NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Microglia NN: Not enough cells (< 50). Emx1_Control: 53, Emx1_Mutant: 27\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 86 cells (max of 3 absolute and 10.0% of 851)...\n",
      "    Genes after filtering: 3726\n",
      "      Calculating mean expression for 3726 genes (Nestin, group Microglia NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Microglia NN).\n",
      "    Finished DGE for Nestin, group Microglia NN. Found 3726 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Microglia_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Microglia_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Microglia_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Microglia_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Microglia_NN' is writable.\n",
      "    Saved 66 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Microglia_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 382 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Microglia_NN\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Vip Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Vip Gaba: Not enough cells (< 50). Emx1_Control: 39, Emx1_Mutant: 67\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 21 cells (max of 3 absolute and 10.0% of 209)...\n",
      "    Genes after filtering: 7334\n",
      "      Calculating mean expression for 7334 genes (Nestin, group Vip Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Vip Gaba).\n",
      "    Finished DGE for Nestin, group Vip Gaba. Found 7334 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Vip_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Vip_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Vip_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Vip_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Vip_Gaba' is writable.\n",
      "    Saved 29 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Vip_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 15 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Vip_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L6b/CT ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 182)...\n",
      "    Genes after filtering: 8914\n",
      "      Calculating mean expression for 8914 genes (Emx1, group L6b/CT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L6b/CT ENT Glut).\n",
      "    Finished DGE for Emx1, group L6b/CT ENT Glut. Found 8914 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L6b_CT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L6b_CT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6b_CT_ENT_Glut' is writable.\n",
      "    Saved 68 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6b_CT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 5 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6b_CT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 37 cells (max of 3 absolute and 10.0% of 363)...\n",
      "    Genes after filtering: 9165\n",
      "      Calculating mean expression for 9165 genes (Nestin, group L6b/CT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L6b/CT ENT Glut).\n",
      "    Finished DGE for Nestin, group L6b/CT ENT Glut. Found 9165 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L6b_CT_ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L6b_CT_ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6b_CT_ENT_Glut' is writable.\n",
      "    Saved 23 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6b_CT_ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 263 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6b_CT_ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: L2 IT ENT-po Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L2 IT ENT-po Glut: Not enough cells (< 50). Emx1_Control: 30, Emx1_Mutant: 126\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L2 IT ENT-po Glut: Not enough cells (< 50). Nestin_Control: 44, Nestin_Mutant: 116\n",
      "\n",
      "Processing group: OB-in Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB-in Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 5, Emx1_Mutant: 11\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB-in Frmd7 Gaba: Not enough cells (< 50). Nestin_Control: 50, Nestin_Mutant: 11\n",
      "\n",
      "Processing group: CA2-FC-IG Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CA2-FC-IG Glut: Not enough cells (< 50). Emx1_Control: 46, Emx1_Mutant: 55\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CA2-FC-IG Glut: Not enough cells (< 50). Nestin_Control: 47, Nestin_Mutant: 39\n",
      "\n",
      "Processing group: L6 CT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 25 cells (max of 3 absolute and 10.0% of 243)...\n",
      "    Genes after filtering: 12068\n",
      "      Calculating mean expression for 12068 genes (Emx1, group L6 CT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L6 CT CTX Glut).\n",
      "    Finished DGE for Emx1, group L6 CT CTX Glut. Found 12068 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L6_CT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L6_CT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 437 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6_CT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 162 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L6_CT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 75 cells (max of 3 absolute and 10.0% of 748)...\n",
      "    Genes after filtering: 11578\n",
      "      Calculating mean expression for 11578 genes (Nestin, group L6 CT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L6 CT CTX Glut).\n",
      "    Finished DGE for Nestin, group L6 CT CTX Glut. Found 11578 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L6_CT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L6_CT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 1240 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6_CT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 161 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L6_CT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: RHP-COA Ndnf Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group RHP-COA Ndnf Gaba: Not enough cells (< 50). Emx1_Control: 48, Emx1_Mutant: 62\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group RHP-COA Ndnf Gaba: Not enough cells (< 50). Nestin_Control: 31, Nestin_Mutant: 70\n",
      "\n",
      "Processing group: SC Lef1 Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SC Lef1 Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SC Lef1 Otx2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DG-PIR Ex IMN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DG-PIR Ex IMN: Not enough cells (< 50). Emx1_Control: 15, Emx1_Mutant: 5\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DG-PIR Ex IMN: Not enough cells (< 50). Nestin_Control: 101, Nestin_Mutant: 37\n",
      "\n",
      "Processing group: OPC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OPC NN: Not enough cells (< 50). Emx1_Control: 40, Emx1_Mutant: 84\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 70 cells (max of 3 absolute and 10.0% of 698)...\n",
      "    Genes after filtering: 5890\n",
      "      Calculating mean expression for 5890 genes (Nestin, group OPC NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group OPC NN).\n",
      "    Finished DGE for Nestin, group OPC NN. Found 5890 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_OPC_NN_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_OPC_NN_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\OPC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\OPC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\OPC_NN' is writable.\n",
      "    Saved 37 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\OPC_NN\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 1549 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\OPC_NN\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Sncg Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Sncg Gaba: Not enough cells (< 50). Emx1_Control: 36, Emx1_Mutant: 71\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Sncg Gaba: Not enough cells (< 50). Nestin_Control: 21, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: L4/5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 20 cells (max of 3 absolute and 10.0% of 195)...\n",
      "    Genes after filtering: 10316\n",
      "      Calculating mean expression for 10316 genes (Emx1, group L4/5 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group L4/5 IT CTX Glut).\n",
      "    Finished DGE for Emx1, group L4/5 IT CTX Glut. Found 10316 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L4_5_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_L4_5_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Saved 8 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 10 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 31 cells (max of 3 absolute and 10.0% of 303)...\n",
      "    Genes after filtering: 10253\n",
      "      Calculating mean expression for 10253 genes (Nestin, group L4/5 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L4/5 IT CTX Glut).\n",
      "    Finished DGE for Nestin, group L4/5 IT CTX Glut. Found 10253 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L4_5_IT_CTX_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L4_5_IT_CTX_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Saved 85 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 20 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L4_5_IT_CTX_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: IA Mgp Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IA Mgp Gaba: Not enough cells (< 50). Emx1_Control: 3, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IA Mgp Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LSX Prdm12 Slit2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Prdm12 Slit2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 4\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Prdm12 Slit2 Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: IT AON-TT-DP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IT AON-TT-DP Glut: Not enough cells (< 50). Emx1_Control: 55, Emx1_Mutant: 23\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IT AON-TT-DP Glut: Not enough cells (< 50). Nestin_Control: 97, Nestin_Mutant: 27\n",
      "\n",
      "Processing group: Lamp5 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Lamp5 Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 31, Emx1_Mutant: 60\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Lamp5 Lhx6 Gaba: Not enough cells (< 50). Nestin_Control: 32, Nestin_Mutant: 83\n",
      "\n",
      "Processing group: CEA-BST Ebf1 Pdyn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CEA-BST Ebf1 Pdyn Gaba: Not enough cells (< 50). Emx1_Control: 3, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CEA-BST Ebf1 Pdyn Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Lamp5 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Lamp5 Gaba: Not enough cells (< 50). Emx1_Control: 21, Emx1_Mutant: 55\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Lamp5 Gaba: Not enough cells (< 50). Nestin_Control: 45, Nestin_Mutant: 63\n",
      "\n",
      "Processing group: IT EP-CLA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IT EP-CLA Glut: Not enough cells (< 50). Emx1_Control: 20, Emx1_Mutant: 36\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IT EP-CLA Glut: Not enough cells (< 50). Nestin_Control: 33, Nestin_Mutant: 36\n",
      "\n",
      "Processing group: Pvalb Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Filtering genes for Emx1 (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 21 cells (max of 3 absolute and 10.0% of 206)...\n",
      "    Genes after filtering: 9183\n",
      "      Calculating mean expression for 9183 genes (Emx1, group Pvalb Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Emx1, group Pvalb Gaba).\n",
      "    Finished DGE for Emx1, group Pvalb Gaba. Found 9183 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Pvalb_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_Pvalb_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Pvalb_Gaba' is writable.\n",
      "    Saved 11 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Pvalb_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 5 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Emx1\\Pvalb_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 23 cells (max of 3 absolute and 10.0% of 221)...\n",
      "    Genes after filtering: 9184\n",
      "      Calculating mean expression for 9184 genes (Nestin, group Pvalb Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group Pvalb Gaba).\n",
      "    Finished DGE for Nestin, group Pvalb Gaba. Found 9184 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Pvalb_Gaba_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_Pvalb_Gaba_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Pvalb_Gaba' is writable.\n",
      "    Saved 3 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Pvalb_Gaba\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 9 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\Pvalb_Gaba\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: VLMC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group VLMC NN: Not enough cells (< 50). Emx1_Control: 30, Emx1_Mutant: 53\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group VLMC NN: Not enough cells (< 50). Nestin_Control: 40, Nestin_Mutant: 14\n",
      "\n",
      "Processing group: L6b EPd Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L6b EPd Glut: Not enough cells (< 50). Emx1_Control: 6, Emx1_Mutant: 6\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L6b EPd Glut: Not enough cells (< 50). Nestin_Control: 7, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: OB Meis2 Thsd7b Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB Meis2 Thsd7b Gaba: Not enough cells (< 50). Emx1_Control: 11, Emx1_Mutant: 9\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB Meis2 Thsd7b Gaba: Not enough cells (< 50). Nestin_Control: 14, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: L5/6 IT TPE-ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L5/6 IT TPE-ENT Glut: Not enough cells (< 50). Emx1_Control: 41, Emx1_Mutant: 101\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 14 cells (max of 3 absolute and 10.0% of 134)...\n",
      "    Genes after filtering: 10656\n",
      "      Calculating mean expression for 10656 genes (Nestin, group L5/6 IT TPE-ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group L5/6 IT TPE-ENT Glut).\n",
      "    Finished DGE for Nestin, group L5/6 IT TPE-ENT Glut. Found 10656 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L5_6_IT_TPE-ENT_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_L5_6_IT_TPE-ENT_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_6_IT_TPE-ENT_Glut' is writable.\n",
      "    Saved 23 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_6_IT_TPE-ENT_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 15 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\L5_6_IT_TPE-ENT_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: IC Six3 En2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IC Six3 En2 Gaba: Not enough cells (< 50). Emx1_Control: 4, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IC Six3 En2 Gaba: Not enough cells (< 50). Nestin_Control: 6, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: ENTmv-PA-COAp Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group ENTmv-PA-COAp Glut: Not enough cells (< 50). Emx1_Control: 32, Emx1_Mutant: 35\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group ENTmv-PA-COAp Glut: Not enough cells (< 50). Nestin_Control: 44, Nestin_Mutant: 64\n",
      "\n",
      "Processing group: LDT Vsx2 Nkx6-1 Nfib Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: Astro-NT NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Astro-NT NN: Not enough cells (< 50). Emx1_Control: 7, Emx1_Mutant: 12\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Astro-NT NN: Not enough cells (< 50). Nestin_Control: 8, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: COAa-PAA-MEA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group COAa-PAA-MEA Barhl2 Glut: Not enough cells (< 50). Emx1_Control: 7, Emx1_Mutant: 10\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group COAa-PAA-MEA Barhl2 Glut: Not enough cells (< 50). Nestin_Control: 16, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: L5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L5 IT CTX Glut: Not enough cells (< 50). Emx1_Control: 9, Emx1_Mutant: 11\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L5 IT CTX Glut: Not enough cells (< 50). Nestin_Control: 17, Nestin_Mutant: 26\n",
      "\n",
      "Processing group: SMC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SMC NN: Not enough cells (< 50). Emx1_Control: 6, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SMC NN: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: Endo NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Endo NN: Not enough cells (< 50). Emx1_Control: 26, Emx1_Mutant: 41\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Endo NN: Not enough cells (< 50). Nestin_Control: 47, Nestin_Mutant: 8\n",
      "\n",
      "Processing group: ABC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group ABC NN: Not enough cells (< 50). Emx1_Control: 10, Emx1_Mutant: 32\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group ABC NN: Not enough cells (< 50). Nestin_Control: 32, Nestin_Mutant: 13\n",
      "\n",
      "Processing group: CLA-EPd-CTX Car3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CLA-EPd-CTX Car3 Glut: Not enough cells (< 50). Emx1_Control: 21, Emx1_Mutant: 11\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CLA-EPd-CTX Car3 Glut: Not enough cells (< 50). Nestin_Control: 27, Nestin_Mutant: 27\n",
      "\n",
      "Processing group: SPVI-SPVC Sall3 Nfib Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CT SUB Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CT SUB Glut: Not enough cells (< 50). Emx1_Control: 24, Emx1_Mutant: 74\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 14 cells (max of 3 absolute and 10.0% of 138)...\n",
      "    Genes after filtering: 9021\n",
      "      Calculating mean expression for 9021 genes (Nestin, group CT SUB Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group CT SUB Glut).\n",
      "    Finished DGE for Nestin, group CT SUB Glut. Found 9021 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CT_SUB_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_CT_SUB_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CT_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CT_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CT_SUB_Glut' is writable.\n",
      "    No significant UP genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CT_SUB_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    No significant DOWN genes found for Mutant_vs_Control. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\CT_SUB_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Peri NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Peri NN: Not enough cells (< 50). Emx1_Control: 9, Emx1_Mutant: 9\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Peri NN: Not enough cells (< 50). Nestin_Control: 18, Nestin_Mutant: 10\n",
      "\n",
      "Processing group: L2 IT PPP-APr Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L2 IT PPP-APr Glut: Not enough cells (< 50). Emx1_Control: 8, Emx1_Mutant: 24\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L2 IT PPP-APr Glut: Not enough cells (< 50). Nestin_Control: 19, Nestin_Mutant: 80\n",
      "\n",
      "Processing group: Pvalb chandelier Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Pvalb chandelier Gaba: Not enough cells (< 50). Emx1_Control: 12, Emx1_Mutant: 15\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Pvalb chandelier Gaba: Not enough cells (< 50). Nestin_Control: 9, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: Ependymal NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Ependymal NN: Not enough cells (< 50). Emx1_Control: 7, Emx1_Mutant: 6\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Ependymal NN: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: NP SUB Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NP SUB Glut: Not enough cells (< 50). Emx1_Control: 25, Emx1_Mutant: 55\n",
      "  Running DGE for Nestin genotype...\n",
      "    Filtering genes for Nestin (Initial: 26870)\n",
      "    Filtering genes: Requiring expression in >= 17 cells (max of 3 absolute and 10.0% of 166)...\n",
      "    Genes after filtering: 9637\n",
      "      Calculating mean expression for 9637 genes (Nestin, group NP SUB Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Nestin, group NP SUB Glut).\n",
      "    Finished DGE for Nestin, group NP SUB Glut. Found 9637 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_NP_SUB_Glut_mut_vs_ctrl.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_NP_SUB_Glut_mut_vs_ctrl.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\NP_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\NP_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\NP_SUB_Glut' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\NP_SUB_Glut\\Mutant_vs_Control_up_significant.csv\n",
      "    Saved 13 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\sig_deg_lists\\Nestin\\NP_SUB_Glut\\Mutant_vs_Control_down_significant.csv\n",
      "\n",
      "Processing group: Tanycyte NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Tanycyte NN: Not enough cells (< 50). Emx1_Control: 4, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Tanycyte NN: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB-mi Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB-mi Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB-mi Frmd7 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB Trdn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB Trdn Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB Trdn Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVR Six3 Sox3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVR Six3 Sox3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVR Six3 Sox3 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-BST Sox6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MEA-BST Sox6 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 4\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MEA-BST Sox6 Gaba: Not enough cells (< 50). Nestin_Control: 9, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: STR D2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR D2 Gaba: Not enough cells (< 50). Emx1_Control: 5, Emx1_Mutant: 13\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR D2 Gaba: Not enough cells (< 50). Nestin_Control: 19, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: CHOR NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CHOR NN: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CHOR NN: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 8\n",
      "\n",
      "Processing group: IC Tfap2d Maf Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IC Tfap2d Maf Glut: Not enough cells (< 50). Emx1_Control: 3, Emx1_Mutant: 4\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IC Tfap2d Maf Glut: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: NI-RPO Gata3 Nr4a2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NI-RPO Gata3 Nr4a2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NI-RPO Gata3 Nr4a2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NLL Gata3 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NLL Gata3 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NLL Gata3 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 12, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: LSX Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Otx2 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MH Tac2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MH Tac2 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MH Tac2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: DTN-LDT-IPN Otp Pax3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB Eomes Ms4a15 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB Eomes Ms4a15 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB Eomes Ms4a15 Glut: Not enough cells (< 50). Nestin_Control: 13, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NTS-PARN Neurod2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NTS-PARN Neurod2 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NTS-PARN Neurod2 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: STR Prox1 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR Prox1 Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR Prox1 Lhx6 Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: OB-STR-CTX Inh IMN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB-STR-CTX Inh IMN: Not enough cells (< 50). Emx1_Control: 14, Emx1_Mutant: 13\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB-STR-CTX Inh IMN: Not enough cells (< 50). Nestin_Control: 17, Nestin_Mutant: 15\n",
      "\n",
      "Processing group: LSX Nkx2-1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Nkx2-1 Gaba: Not enough cells (< 50). Emx1_Control: 5, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Nkx2-1 Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: STR D1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR D1 Gaba: Not enough cells (< 50). Emx1_Control: 5, Emx1_Mutant: 7\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR D1 Gaba: Not enough cells (< 50). Nestin_Control: 5, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MY Lhx1 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MY Lhx1 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MY Lhx1 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: TRS-BAC Sln Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group TRS-BAC Sln Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 4\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group TRS-BAC Sln Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CB Granule Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CB Granule Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CB Granule Glut: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: L2/3 IT RSP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L2/3 IT RSP Glut: Not enough cells (< 50). Emx1_Control: 5, Emx1_Mutant: 5\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L2/3 IT RSP Glut: Not enough cells (< 50). Nestin_Control: 6, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: PRT Mecom Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRT Mecom Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRT Mecom Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: OB Dopa-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB Dopa-Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB Dopa-Gaba: Not enough cells (< 50). Nestin_Control: 19, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MDRNd Prox1 Pax6 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Nr2e1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: CB PLI Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CB PLI Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CB PLI Gly-Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: BAM NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group BAM NN: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group BAM NN: Not enough cells (< 50). Nestin_Control: 16, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: RT-ZI Gnb3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group RT-ZI Gnb3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group RT-ZI Gnb3 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SPVC Mafa Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SPVC Mafa Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SPVC Mafa Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: TH Prkcd Grin2c Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group TH Prkcd Grin2c Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 8\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group TH Prkcd Grin2c Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SNc-VTA-RAmb Foxa1 Dopa (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SNc-VTA-RAmb Foxa1 Dopa: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SNc-VTA-RAmb Foxa1 Dopa: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PGRN-PARN-MDRN Hoxb5 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells (< 50). Nestin_Control: 9, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: VCO Mafa Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group VCO Mafa Meis2 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group VCO Mafa Meis2 Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR-PAL Chst9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR-PAL Chst9 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 15\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR-PAL Chst9 Gaba: Not enough cells (< 50). Nestin_Control: 7, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: NP PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NP PPP Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 6\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NP PPP Glut: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 14\n",
      "\n",
      "Processing group: SC-PAG Lef1 Emx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SC-PAG Lef1 Emx2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SC-PAG Lef1 Emx2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: DC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DC NN: Not enough cells (< 50). Emx1_Control: 4, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DC NN: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: SPVI-SPVC Tlx3 Ebf3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells (< 50). Nestin_Control: 8, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: OB-out Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OB-out Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OB-out Frmd7 Gaba: Not enough cells (< 50). Nestin_Control: 48, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: CS-RPO Meis2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CS-RPO Meis2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CS-RPO Meis2 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: AHN-SBPV-PVHd Pdrm12 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: LSX Prdm12 Zeb2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Prdm12 Zeb2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Prdm12 Zeb2 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LDT-PCG-CS Gata3 Lhx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCs Pax7 Nfia Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCs Pax7 Nfia Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCs Pax7 Nfia Gaba: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: ADP-MPO Trp73 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group ADP-MPO Trp73 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group ADP-MPO Trp73 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Sst Chodl Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Sst Chodl Gaba: Not enough cells (< 50). Emx1_Control: 2, Emx1_Mutant: 7\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Sst Chodl Gaba: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: OEC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OEC NN: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OEC NN: Not enough cells (< 50). Nestin_Control: 39, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NLOT Rho Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NLOT Rho Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NLOT Rho Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCsg Gabrr2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCsg Gabrr2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCsg Gabrr2 Gaba: Not enough cells (< 50). Nestin_Control: 7, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: PAG-ND-PCG Onecut1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-ND-PCG Onecut1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-ND-PCG Onecut1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MB-MY Tph2 Glut-Sero (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MB-MY Tph2 Glut-Sero: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MB-MY Tph2 Glut-Sero: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PG-TRN-LRN Fat2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PG-TRN-LRN Fat2 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PG-TRN-LRN Fat2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LSX Sall3 Lmo1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Sall3 Lmo1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 9\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Sall3 Lmo1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PRC-PAG Pax6 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRC-PAG Pax6 Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRC-PAG Pax6 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCsg Pde5a Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCsg Pde5a Glut: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCsg Pde5a Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Nfib Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MEA-BST Lhx6 Nfib Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MEA-BST Lhx6 Nfib Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MEA-BST Lhx6 Sp9 Gaba: Not enough cells (< 50). Emx1_Control: 1, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MEA-BST Lhx6 Sp9 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CEA-BST Six3 Cyp26b1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CEA-BST Six3 Cyp26b1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 5\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CEA-BST Six3 Cyp26b1 Gaba: Not enough cells (< 50). Nestin_Control: 5, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: Bergmann NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Bergmann NN: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Bergmann NN: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: L5 PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L5 PPP Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 6\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L5 PPP Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: SCig Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCig Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCig Foxb1 Glut: Not enough cells (< 50). Nestin_Control: 6, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: SI-MA-LPO-LHA Skor1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SI-MA-LPO-LHA Skor1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 6\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SI-MA-LPO-LHA Skor1 Glut: Not enough cells (< 50). Nestin_Control: 7, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: MV-SPIV Zic4 Neurod2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MV-SPIV Zic4 Neurod2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MV-SPIV Zic4 Neurod2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LSX Sall3 Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LSX Sall3 Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LSX Sall3 Pax6 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NDB-SI-MA-STRv Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MG-POL-SGN Nts Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MG-POL-SGN Nts Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MG-POL-SGN Nts Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: LGv-ZI Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LGv-ZI Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LGv-ZI Otx2 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PH-SUM Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PH-SUM Foxa1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PH-SUM Foxa1 Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PH-LHA Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PH-LHA Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PH-LHA Foxb1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LDT-PCG St18 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LDT-PCG St18 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LDT-PCG St18 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCH Six6 Cdc14a Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCH Six6 Cdc14a Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCH Six6 Cdc14a Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: L4 RSP-ACA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group L4 RSP-ACA Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 3\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group L4 RSP-ACA Glut: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PRT Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRT Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRT Tcf7l2 Gaba: Not enough cells (< 50). Nestin_Control: 6, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: PRP Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRP Otp Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRP Otp Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: DMH Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DMH Hmx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DMH Hmx2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STN-PSTN Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STN-PSTN Pitx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STN-PSTN Pitx2 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OT D3 Folh1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group OT D3 Folh1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group OT D3 Folh1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-RN Nkx2-2 Otx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 2\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MRN-PPN-CUN Pax8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MRN-PPN-CUN Pax8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MRN-PPN-CUN Pax8 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCs Lef1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCs Lef1 Gli3 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCs Lef1 Gli3 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PDTg-PCG Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PDTg-PCG Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PDTg-PCG Pax6 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PH-ant-LHA Otp Bsx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PH-ant-LHA Otp Bsx Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PH-ant-LHA Otp Bsx Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: ZI Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group ZI Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 1\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group ZI Pax6 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LDT-PCG Vsx2 Lhx4 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LDT-PCG Vsx2 Lhx4 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LDT-PCG Vsx2 Lhx4 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CEA-AAA-BST Six3 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PAG-SC Neurod2 Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-SC Neurod2 Meis2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-SC Neurod2 Meis2 Glut: Not enough cells (< 50). Nestin_Control: 4, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MPN-MPO-PVpo Hmx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MPN-MPO-PVpo Hmx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MPN-MPO-PVpo Hmx2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: BST-MPN Six3 Nrgn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group BST-MPN Six3 Nrgn Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group BST-MPN Six3 Nrgn Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-PPN Pax5 Sox21 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-PPN Pax5 Sox21 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-PPN Pax5 Sox21 Gaba: Not enough cells (< 50). Nestin_Control: 3, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PRNc Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRNc Otp Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRNc Otp Gly-Gaba: Not enough cells (< 50). Nestin_Control: 5, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: MPO-ADP Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MPO-ADP Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MPO-ADP Lhx8 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MS-SF Bsx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MS-SF Bsx Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MS-SF Bsx Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PB Sst Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PB Sst Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PB Sst Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SI-MA-ACB Ebf1 Bnc2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NDB-SI-ant Prdm12 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group NDB-SI-ant Prdm12 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group NDB-SI-ant Prdm12 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SI-MPO-LPO Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SI-MPO-LPO Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SI-MPO-LPO Lhx8 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: Pineal Crx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Pineal Crx Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Pineal Crx Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG Pou4f2 Mesi2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG Pou4f2 Mesi2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG Pou4f2 Mesi2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LHA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LHA Barhl2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LHA Barhl2 Glut: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SBPV-PVa Six6 Satb2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SBPV-PVa Six6 Satb2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SBPV-PVa Six6 Satb2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR D1 Sema5a Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR D1 Sema5a Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR D1 Sema5a Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SC Bnc2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SC Bnc2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SC Bnc2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SC Tnnt1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SC Tnnt1 Gli3 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SC Tnnt1 Gli3 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVHd-SBPV Six3 Prox1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVHd-SBPV Six3 Prox1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVHd-SBPV Six3 Prox1 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCm-PAG Cdh23 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCm-PAG Cdh23 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCm-PAG Cdh23 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCop Sln Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCop Sln Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCop Sln Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CS-PRNr-DR En1 Sox2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CS-PRNr-DR En1 Sox2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CS-PRNr-DR En1 Sox2 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-SC Pou4f1 Zic1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-SC Pou4f1 Zic1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-SC Pou4f1 Zic1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: TU-ARH Otp Six6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group TU-ARH Otp Six6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group TU-ARH Otp Six6 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: VMH Nr5a1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group VMH Nr5a1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group VMH Nr5a1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PARN-MDRNd-NTS Gbx2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVpo-VMPO-MPN Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells (< 50). Nestin_Control: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MY Prox1 Lmo7 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MY Prox1 Lmo7 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MY Prox1 Lmo7 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVH-SO-PVa Otp Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVH-SO-PVa Otp Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVH-SO-PVa Otp Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PRC-PAG Tcf7l2 Irx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: IF-RL-CLI-PAG Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group IF-RL-CLI-PAG Foxa1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group IF-RL-CLI-PAG Foxa1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: CU-ECU-SPVI Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CU-ECU-SPVI Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CU-ECU-SPVI Foxb1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group STR Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group STR Lhx8 Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-COA-BMA Ccdc42 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MEA-COA-BMA Ccdc42 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MEA-COA-BMA Ccdc42 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CS-PRNr-PCG Tmem163 Otp Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVT-PT Ntrk1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVT-PT Ntrk1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVT-PT Ntrk1 Glut: Not enough cells (< 50). Nestin_Control: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVHd-DMH Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PVHd-DMH Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PVHd-DMH Lhx6 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: LDT Fgf7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LDT Fgf7 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LDT Fgf7 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PPN-CUN-PCG Otp En1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PPN-CUN-PCG Otp En1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PPN-CUN-PCG Otp En1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PH-an Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PH-an Pitx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PH-an Pitx2 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCig-an-PPT Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCig-an-PPT Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCig-an-PPT Foxb1 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PAG-MRN-RN Foxa2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group PAG-MRN-RN Foxa2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group PAG-MRN-RN Foxa2 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: Astroependymal NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group Astroependymal NN: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group Astroependymal NN: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MV-SPIV Phox2b Ebf3 Lbx1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCs Dmbx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group SCs Dmbx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group SCs Dmbx1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LH Pou4f1 Sox1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group LH Pou4f1 Sox1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group LH Pou4f1 Sox1 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: RPA Pax6 Hoxb5 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DCO Il22 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DCO Il22 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DCO Il22 Gly-Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DMH-LHA Gsx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group DMH-LHA Gsx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group DMH-LHA Gsx1 Gaba: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: RN Spp1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group RN Spp1 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group RN Spp1 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: APN C1ql4 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Emx1 genotype...\n",
      "    Skipping Emx1 DGE for group APN C1ql4 Glut: Not enough cells (< 50). Emx1_Control: 0, Emx1_Mutant: 0\n",
      "  Running DGE for Nestin genotype...\n",
      "    Skipping Nestin DGE for group APN C1ql4 Glut: Not enough cells (< 50). Nestin_Control: 0, Nestin_Mutant: 1\n",
      "\n",
      "Summary of skipped groups (using key 'mapmycells_second_layer') for Emx1 saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Emx1_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "Summary of skipped groups (using key 'mapmycells_second_layer') for Nestin saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\geno_spec_cond_comp\\dge_Nestin_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "Genotype-Specific DGE analysis finished.\n",
      "\n",
      "Generating Volcano Plots for Genotype-Specific DGE results...\n",
      "\n",
      "--- Visualizing for Genotype: Emx1 ---\n",
      "  Generating volcano plot for: Emx1 - Astro-TE NN (Counts: Emx1_Ctrl=152, Emx1_Mut=220) -> dge_Emx1_Astro-TE_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - L6 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=36, Emx1_Mut=37.\n",
      "  Generating volcano plot for: Emx1 - CA3 Glut (Counts: Emx1_Ctrl=379, Emx1_Mut=463) -> dge_Emx1_CA3_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - Monocytes NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=3, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - L5 ET CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=46, Emx1_Mut=45.\n",
      "  Generating volcano plot for: Emx1 - L2/3 IT PPP Glut (Counts: Emx1_Ctrl=54, Emx1_Mut=410) -> dge_Emx1_L2_3_IT_PPP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - CA1-ProS Glut (Counts: Emx1_Ctrl=664, Emx1_Mut=751) -> dge_Emx1_CA1-ProS_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - MEA Slc17a7 Glut (Counts: Emx1_Ctrl=103, Emx1_Mut=68) -> dge_Emx1_MEA_Slc17a7_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - Astro-OLF NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - COAp Grxcr2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=23, Emx1_Mut=25.\n",
      "  Generating volcano plot for: Emx1 - LA-BLA-BMA-PA Glut (Counts: Emx1_Ctrl=135, Emx1_Mut=186) -> dge_Emx1_LA-BLA-BMA-PA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - DG Glut (Counts: Emx1_Ctrl=846, Emx1_Mut=402) -> dge_Emx1_DG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - Oligo NN (Counts: Emx1_Ctrl=133, Emx1_Mut=142) -> dge_Emx1_Oligo_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - NTS Phox2b Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=6, Emx1_Mut=3.\n",
      "  Generating volcano plot for: Emx1 - Sst Gaba (Counts: Emx1_Ctrl=106, Emx1_Mut=195) -> dge_Emx1_Sst_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - L2/3 IT PIR-ENTl Glut (Counts: Emx1_Ctrl=139, Emx1_Mut=169) -> dge_Emx1_L2_3_IT_PIR-ENTl_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - SUB-ProS Glut (Counts: Emx1_Ctrl=114, Emx1_Mut=173) -> dge_Emx1_SUB-ProS_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - Lymphoid NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=11, Emx1_Mut=10.\n",
      "  Skipping volcano plot for Emx1 - L6b CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=12, Emx1_Mut=13.\n",
      "  Skipping volcano plot for Emx1 - L5 NP CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=8, Emx1_Mut=13.\n",
      "  Generating volcano plot for: Emx1 - L2/3 IT CTX Glut (Counts: Emx1_Ctrl=89, Emx1_Mut=189) -> dge_Emx1_L2_3_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Emx1 - L2/3 IT ENT Glut (Counts: Emx1_Ctrl=186, Emx1_Mut=382) -> dge_Emx1_L2_3_IT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - HPF CR Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=6, Emx1_Mut=5.\n",
      "  Skipping volcano plot for Emx1 - Microglia NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=53, Emx1_Mut=27.\n",
      "  Skipping volcano plot for Emx1 - Vip Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=39, Emx1_Mut=67.\n",
      "  Generating volcano plot for: Emx1 - L6b/CT ENT Glut (Counts: Emx1_Ctrl=54, Emx1_Mut=128) -> dge_Emx1_L6b_CT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - L2 IT ENT-po Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=30, Emx1_Mut=126.\n",
      "  Skipping volcano plot for Emx1 - OB-in Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=5, Emx1_Mut=11.\n",
      "  Skipping volcano plot for Emx1 - CA2-FC-IG Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=46, Emx1_Mut=55.\n",
      "  Generating volcano plot for: Emx1 - L6 CT CTX Glut (Counts: Emx1_Ctrl=82, Emx1_Mut=161) -> dge_Emx1_L6_CT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - RHP-COA Ndnf Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=48, Emx1_Mut=62.\n",
      "  Skipping volcano plot for Emx1 - SC Lef1 Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - DG-PIR Ex IMN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=15, Emx1_Mut=5.\n",
      "  Skipping volcano plot for Emx1 - OPC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=40, Emx1_Mut=84.\n",
      "  Skipping volcano plot for Emx1 - Sncg Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=36, Emx1_Mut=71.\n",
      "  Generating volcano plot for: Emx1 - L4/5 IT CTX Glut (Counts: Emx1_Ctrl=87, Emx1_Mut=108) -> dge_Emx1_L4_5_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - IA Mgp Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=3, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - LSX Prdm12 Slit2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=4.\n",
      "  Skipping volcano plot for Emx1 - IT AON-TT-DP Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=55, Emx1_Mut=23.\n",
      "  Skipping volcano plot for Emx1 - Lamp5 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=31, Emx1_Mut=60.\n",
      "  Skipping volcano plot for Emx1 - CEA-BST Ebf1 Pdyn Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=3, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - Lamp5 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=21, Emx1_Mut=55.\n",
      "  Skipping volcano plot for Emx1 - IT EP-CLA Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=20, Emx1_Mut=36.\n",
      "  Generating volcano plot for: Emx1 - Pvalb Gaba (Counts: Emx1_Ctrl=67, Emx1_Mut=139) -> dge_Emx1_Pvalb_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Emx1 - VLMC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=30, Emx1_Mut=53.\n",
      "  Skipping volcano plot for Emx1 - L6b EPd Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=6, Emx1_Mut=6.\n",
      "  Skipping volcano plot for Emx1 - OB Meis2 Thsd7b Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=11, Emx1_Mut=9.\n",
      "  Skipping volcano plot for Emx1 - L5/6 IT TPE-ENT Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=41, Emx1_Mut=101.\n",
      "  Skipping volcano plot for Emx1 - IC Six3 En2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=4, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - ENTmv-PA-COAp Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=32, Emx1_Mut=35.\n",
      "  Skipping volcano plot for Emx1 - LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - Astro-NT NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=7, Emx1_Mut=12.\n",
      "  Skipping volcano plot for Emx1 - COAa-PAA-MEA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=7, Emx1_Mut=10.\n",
      "  Skipping volcano plot for Emx1 - L5 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=9, Emx1_Mut=11.\n",
      "  Skipping volcano plot for Emx1 - SMC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=6, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - Endo NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=26, Emx1_Mut=41.\n",
      "  Skipping volcano plot for Emx1 - ABC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=10, Emx1_Mut=32.\n",
      "  Skipping volcano plot for Emx1 - CLA-EPd-CTX Car3 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=21, Emx1_Mut=11.\n",
      "  Skipping volcano plot for Emx1 - SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - CT SUB Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=24, Emx1_Mut=74.\n",
      "  Skipping volcano plot for Emx1 - Peri NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=9, Emx1_Mut=9.\n",
      "  Skipping volcano plot for Emx1 - L2 IT PPP-APr Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=8, Emx1_Mut=24.\n",
      "  Skipping volcano plot for Emx1 - Pvalb chandelier Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=12, Emx1_Mut=15.\n",
      "  Skipping volcano plot for Emx1 - Ependymal NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=7, Emx1_Mut=6.\n",
      "  Skipping volcano plot for Emx1 - NP SUB Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=25, Emx1_Mut=55.\n",
      "  Skipping volcano plot for Emx1 - Tanycyte NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=4, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - OB-mi Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - OB Trdn Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVR Six3 Sox3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - MEA-BST Sox6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=4.\n",
      "  Skipping volcano plot for Emx1 - STR D2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=5, Emx1_Mut=13.\n",
      "  Skipping volcano plot for Emx1 - CHOR NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - IC Tfap2d Maf Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=3, Emx1_Mut=4.\n",
      "  Skipping volcano plot for Emx1 - NI-RPO Gata3 Nr4a2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - NLL Gata3 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - LSX Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MH Tac2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - OB Eomes Ms4a15 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - NTS-PARN Neurod2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - STR Prox1 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - OB-STR-CTX Inh IMN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=14, Emx1_Mut=13.\n",
      "  Skipping volcano plot for Emx1 - LSX Nkx2-1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=5, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - STR D1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=5, Emx1_Mut=7.\n",
      "  Skipping volcano plot for Emx1 - MY Lhx1 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - TRS-BAC Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=4.\n",
      "  Skipping volcano plot for Emx1 - CB Granule Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - L2/3 IT RSP Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=5, Emx1_Mut=5.\n",
      "  Skipping volcano plot for Emx1 - PRT Mecom Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - OB Dopa-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - CB PLI Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - BAM NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - RT-ZI Gnb3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SPVC Mafa Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - TH Prkcd Grin2c Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=8.\n",
      "  Skipping volcano plot for Emx1 - SNc-VTA-RAmb Foxa1 Dopa: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - VCO Mafa Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - STR-PAL Chst9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=15.\n",
      "  Skipping volcano plot for Emx1 - NP PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=6.\n",
      "  Skipping volcano plot for Emx1 - SC-PAG Lef1 Emx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - DC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=4, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - OB-out Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - CS-RPO Meis2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - LSX Prdm12 Zeb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - SCs Pax7 Nfia Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - ADP-MPO Trp73 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - Sst Chodl Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=2, Emx1_Mut=7.\n",
      "  Skipping volcano plot for Emx1 - OEC NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - NLOT Rho Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SCsg Gabrr2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PAG-ND-PCG Onecut1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - MB-MY Tph2 Glut-Sero: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PG-TRN-LRN Fat2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - LSX Sall3 Lmo1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=9.\n",
      "  Skipping volcano plot for Emx1 - PRC-PAG Pax6 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - SCsg Pde5a Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MEA-BST Lhx6 Nfib Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - MEA-BST Lhx6 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=1, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - CEA-BST Six3 Cyp26b1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=5.\n",
      "  Skipping volcano plot for Emx1 - Bergmann NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - L5 PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=6.\n",
      "  Skipping volcano plot for Emx1 - SCig Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - SI-MA-LPO-LHA Skor1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=6.\n",
      "  Skipping volcano plot for Emx1 - MV-SPIV Zic4 Neurod2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - LSX Sall3 Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - MG-POL-SGN Nts Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - LGv-ZI Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PH-SUM Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PH-LHA Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - LDT-PCG St18 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - SCH Six6 Cdc14a Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - L4 RSP-ACA Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=3.\n",
      "  Skipping volcano plot for Emx1 - PRT Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - PRP Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - DMH Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - STN-PSTN Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - OT D3 Folh1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=2.\n",
      "  Skipping volcano plot for Emx1 - MRN-PPN-CUN Pax8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - SCs Lef1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PDTg-PCG Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - PH-ant-LHA Otp Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - ZI Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=1.\n",
      "  Skipping volcano plot for Emx1 - LDT-PCG Vsx2 Lhx4 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PAG-SC Neurod2 Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MPN-MPO-PVpo Hmx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - BST-MPN Six3 Nrgn Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PAG-PPN Pax5 Sox21 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PRNc Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MPO-ADP Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MS-SF Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PB Sst Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - NDB-SI-ant Prdm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SI-MPO-LPO Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - Pineal Crx Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PAG Pou4f2 Mesi2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - LHA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SBPV-PVa Six6 Satb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - STR D1 Sema5a Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SC Bnc2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SC Tnnt1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVHd-SBPV Six3 Prox1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SCm-PAG Cdh23 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SCop Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - CS-PRNr-DR En1 Sox2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PAG-SC Pou4f1 Zic1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - TU-ARH Otp Six6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - VMH Nr5a1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MY Prox1 Lmo7 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVH-SO-PVa Otp Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - IF-RL-CLI-PAG Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - CU-ECU-SPVI Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - STR Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MEA-COA-BMA Ccdc42 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVT-PT Ntrk1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PVHd-DMH Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - LDT Fgf7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PPN-CUN-PCG Otp En1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PH-an Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SCig-an-PPT Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - PAG-MRN-RN Foxa2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - Astroependymal NN: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - SCs Dmbx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - LH Pou4f1 Sox1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - DCO Il22 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - DMH-LHA Gsx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - RN Spp1 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "  Skipping volcano plot for Emx1 - APN C1ql4 Glut: Not enough cells in one or both groups (< 50). Emx1_Ctrl=0, Emx1_Mut=0.\n",
      "\n",
      "--- Visualizing for Genotype: Nestin ---\n",
      "  Generating volcano plot for: Nestin - Astro-TE NN (Counts: Nestin_Ctrl=474, Nestin_Mut=159) -> dge_Nestin_Astro-TE_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - L6 IT CTX Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=31, Nestin_Mut=55.\n",
      "  Generating volcano plot for: Nestin - CA3 Glut (Counts: Nestin_Ctrl=287, Nestin_Mut=417) -> dge_Nestin_CA3_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Monocytes NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Generating volcano plot for: Nestin - L5 ET CTX Glut (Counts: Nestin_Ctrl=50, Nestin_Mut=114) -> dge_Nestin_L5_ET_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - L2/3 IT PPP Glut (Counts: Nestin_Ctrl=135, Nestin_Mut=442) -> dge_Nestin_L2_3_IT_PPP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - CA1-ProS Glut (Counts: Nestin_Ctrl=803, Nestin_Mut=1161) -> dge_Nestin_CA1-ProS_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - MEA Slc17a7 Glut (Counts: Nestin_Ctrl=429, Nestin_Mut=223) -> dge_Nestin_MEA_Slc17a7_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Astro-OLF NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=6, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - COAp Grxcr2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=7, Nestin_Mut=1.\n",
      "  Generating volcano plot for: Nestin - LA-BLA-BMA-PA Glut (Counts: Nestin_Ctrl=217, Nestin_Mut=102) -> dge_Nestin_LA-BLA-BMA-PA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - DG Glut (Counts: Nestin_Ctrl=1554, Nestin_Mut=922) -> dge_Nestin_DG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - Oligo NN (Counts: Nestin_Ctrl=800, Nestin_Mut=753) -> dge_Nestin_Oligo_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - NTS Phox2b Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=8, Nestin_Mut=1.\n",
      "  Generating volcano plot for: Nestin - Sst Gaba (Counts: Nestin_Ctrl=144, Nestin_Mut=261) -> dge_Nestin_Sst_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - L2/3 IT PIR-ENTl Glut (Counts: Nestin_Ctrl=232, Nestin_Mut=156) -> dge_Nestin_L2_3_IT_PIR-ENTl_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - SUB-ProS Glut (Counts: Nestin_Ctrl=153, Nestin_Mut=218) -> dge_Nestin_SUB-ProS_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Lymphoid NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=14, Nestin_Mut=6.\n",
      "  Skipping volcano plot for Nestin - L6b CTX Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=21.\n",
      "  Skipping volcano plot for Nestin - L5 NP CTX Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=9, Nestin_Mut=45.\n",
      "  Generating volcano plot for: Nestin - L2/3 IT CTX Glut (Counts: Nestin_Ctrl=208, Nestin_Mut=362) -> dge_Nestin_L2_3_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - L2/3 IT ENT Glut (Counts: Nestin_Ctrl=203, Nestin_Mut=460) -> dge_Nestin_L2_3_IT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - HPF CR Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=25, Nestin_Mut=37.\n",
      "  Generating volcano plot for: Nestin - Microglia NN (Counts: Nestin_Ctrl=355, Nestin_Mut=496) -> dge_Nestin_Microglia_NN_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - Vip Gaba (Counts: Nestin_Ctrl=62, Nestin_Mut=147) -> dge_Nestin_Vip_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Nestin - L6b/CT ENT Glut (Counts: Nestin_Ctrl=136, Nestin_Mut=227) -> dge_Nestin_L6b_CT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - L2 IT ENT-po Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=44, Nestin_Mut=116.\n",
      "  Skipping volcano plot for Nestin - OB-in Frmd7 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=50, Nestin_Mut=11.\n",
      "  Skipping volcano plot for Nestin - CA2-FC-IG Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=47, Nestin_Mut=39.\n",
      "  Generating volcano plot for: Nestin - L6 CT CTX Glut (Counts: Nestin_Ctrl=254, Nestin_Mut=494) -> dge_Nestin_L6_CT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - RHP-COA Ndnf Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=31, Nestin_Mut=70.\n",
      "  Skipping volcano plot for Nestin - SC Lef1 Otx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - DG-PIR Ex IMN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=101, Nestin_Mut=37.\n",
      "  Generating volcano plot for: Nestin - OPC NN (Counts: Nestin_Ctrl=294, Nestin_Mut=404) -> dge_Nestin_OPC_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Sncg Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=21, Nestin_Mut=21.\n",
      "  Generating volcano plot for: Nestin - L4/5 IT CTX Glut (Counts: Nestin_Ctrl=55, Nestin_Mut=248) -> dge_Nestin_L4_5_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - IA Mgp Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - LSX Prdm12 Slit2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - IT AON-TT-DP Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=97, Nestin_Mut=27.\n",
      "  Skipping volcano plot for Nestin - Lamp5 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=32, Nestin_Mut=83.\n",
      "  Skipping volcano plot for Nestin - CEA-BST Ebf1 Pdyn Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - Lamp5 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=45, Nestin_Mut=63.\n",
      "  Skipping volcano plot for Nestin - IT EP-CLA Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=33, Nestin_Mut=36.\n",
      "  Generating volcano plot for: Nestin - Pvalb Gaba (Counts: Nestin_Ctrl=77, Nestin_Mut=144) -> dge_Nestin_Pvalb_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - VLMC NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=40, Nestin_Mut=14.\n",
      "  Skipping volcano plot for Nestin - L6b EPd Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=7, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - OB Meis2 Thsd7b Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=14, Nestin_Mut=7.\n",
      "  Generating volcano plot for: Nestin - L5/6 IT TPE-ENT Glut (Counts: Nestin_Ctrl=52, Nestin_Mut=82) -> dge_Nestin_L5_6_IT_TPE-ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - IC Six3 En2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=6, Nestin_Mut=4.\n",
      "  Skipping volcano plot for Nestin - ENTmv-PA-COAp Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=44, Nestin_Mut=64.\n",
      "  Skipping volcano plot for Nestin - LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - Astro-NT NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=8, Nestin_Mut=9.\n",
      "  Skipping volcano plot for Nestin - COAa-PAA-MEA Barhl2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=16, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - L5 IT CTX Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=17, Nestin_Mut=26.\n",
      "  Skipping volcano plot for Nestin - SMC NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - Endo NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=47, Nestin_Mut=8.\n",
      "  Skipping volcano plot for Nestin - ABC NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=32, Nestin_Mut=13.\n",
      "  Skipping volcano plot for Nestin - CLA-EPd-CTX Car3 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=27, Nestin_Mut=27.\n",
      "  Skipping volcano plot for Nestin - SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Generating volcano plot for: Nestin - CT SUB Glut (Counts: Nestin_Ctrl=52, Nestin_Mut=86) -> dge_Nestin_CT_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Peri NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=18, Nestin_Mut=10.\n",
      "  Skipping volcano plot for Nestin - L2 IT PPP-APr Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=19, Nestin_Mut=80.\n",
      "  Skipping volcano plot for Nestin - Pvalb chandelier Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=9, Nestin_Mut=21.\n",
      "  Skipping volcano plot for Nestin - Ependymal NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=2.\n",
      "  Generating volcano plot for: Nestin - NP SUB Glut (Counts: Nestin_Ctrl=80, Nestin_Mut=86) -> dge_Nestin_NP_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Nestin - Tanycyte NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - OB-mi Frmd7 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - OB Trdn Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PVR Six3 Sox3 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - MEA-BST Sox6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=9, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - STR D2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=19, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - CHOR NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=8.\n",
      "  Skipping volcano plot for Nestin - IC Tfap2d Maf Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=6.\n",
      "  Skipping volcano plot for Nestin - NI-RPO Gata3 Nr4a2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - NLL Gata3 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=12, Nestin_Mut=6.\n",
      "  Skipping volcano plot for Nestin - LSX Otx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MH Tac2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=9.\n",
      "  Skipping volcano plot for Nestin - DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - OB Eomes Ms4a15 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=13, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - NTS-PARN Neurod2 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - STR Prox1 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - OB-STR-CTX Inh IMN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=17, Nestin_Mut=15.\n",
      "  Skipping volcano plot for Nestin - LSX Nkx2-1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - STR D1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=5, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - MY Lhx1 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - TRS-BAC Sln Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - CB Granule Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - L2/3 IT RSP Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=6, Nestin_Mut=9.\n",
      "  Skipping volcano plot for Nestin - PRT Mecom Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - OB Dopa-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=19, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - CB PLI Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - BAM NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=16, Nestin_Mut=7.\n",
      "  Skipping volcano plot for Nestin - RT-ZI Gnb3 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SPVC Mafa Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - TH Prkcd Grin2c Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SNc-VTA-RAmb Foxa1 Dopa: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=9, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - VCO Mafa Meis2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - STR-PAL Chst9 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=7, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - NP PPP Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=14.\n",
      "  Skipping volcano plot for Nestin - SC-PAG Lef1 Emx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - DC NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=6.\n",
      "  Skipping volcano plot for Nestin - SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=8, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - OB-out Frmd7 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=48, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - CS-RPO Meis2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - LSX Prdm12 Zeb2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCs Pax7 Nfia Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=4.\n",
      "  Skipping volcano plot for Nestin - ADP-MPO Trp73 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - Sst Chodl Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=7.\n",
      "  Skipping volcano plot for Nestin - OEC NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=39, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - NLOT Rho Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCsg Gabrr2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=7, Nestin_Mut=6.\n",
      "  Skipping volcano plot for Nestin - PAG-ND-PCG Onecut1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - MB-MY Tph2 Glut-Sero: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PG-TRN-LRN Fat2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - LSX Sall3 Lmo1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PRC-PAG Pax6 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCsg Pde5a Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - MEA-BST Lhx6 Nfib Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MEA-BST Lhx6 Sp9 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - CEA-BST Six3 Cyp26b1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=5, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - Bergmann NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - L5 PPP Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - SCig Foxb1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=6, Nestin_Mut=7.\n",
      "  Skipping volcano plot for Nestin - SI-MA-LPO-LHA Skor1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=7, Nestin_Mut=5.\n",
      "  Skipping volcano plot for Nestin - MV-SPIV Zic4 Neurod2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - LSX Sall3 Pax6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MG-POL-SGN Nts Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - LGv-ZI Otx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PH-SUM Foxa1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PH-LHA Foxb1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - LDT-PCG St18 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCH Six6 Cdc14a Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - L4 RSP-ACA Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - PRT Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=6, Nestin_Mut=9.\n",
      "  Skipping volcano plot for Nestin - PRP Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - DMH Hmx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - STN-PSTN Pitx2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - OT D3 Folh1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - MRN-PPN-CUN Pax8 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCs Lef1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PDTg-PCG Pax6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PH-ant-LHA Otp Bsx Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - ZI Pax6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - LDT-PCG Vsx2 Lhx4 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PAG-SC Neurod2 Meis2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=4, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - MPN-MPO-PVpo Hmx2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - BST-MPN Six3 Nrgn Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PAG-PPN Pax5 Sox21 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=3, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PRNc Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=5, Nestin_Mut=3.\n",
      "  Skipping volcano plot for Nestin - MPO-ADP Lhx8 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MS-SF Bsx Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - PB Sst Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - NDB-SI-ant Prdm12 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - SI-MPO-LPO Lhx8 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - Pineal Crx Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PAG Pou4f2 Mesi2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - LHA Barhl2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SBPV-PVa Six6 Satb2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - STR D1 Sema5a Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SC Bnc2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SC Tnnt1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PVHd-SBPV Six3 Prox1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - SCm-PAG Cdh23 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - SCop Sln Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - CS-PRNr-DR En1 Sox2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PAG-SC Pou4f1 Zic1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - TU-ARH Otp Six6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - VMH Nr5a1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=2, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - MY Prox1 Lmo7 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PVH-SO-PVa Otp Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - IF-RL-CLI-PAG Foxa1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - CU-ECU-SPVI Foxb1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - STR Lhx8 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - MEA-COA-BMA Ccdc42 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=0.\n",
      "  Skipping volcano plot for Nestin - PVT-PT Ntrk1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=1, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PVHd-DMH Lhx6 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=4.\n",
      "  Skipping volcano plot for Nestin - LDT Fgf7 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - PPN-CUN-PCG Otp En1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - PH-an Pitx2 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - SCig-an-PPT Foxb1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - PAG-MRN-RN Foxa2 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=2.\n",
      "  Skipping volcano plot for Nestin - Astroependymal NN: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - SCs Dmbx1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - LH Pou4f1 Sox1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=4.\n",
      "  Skipping volcano plot for Nestin - RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - DCO Il22 Gly-Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - DMH-LHA Gsx1 Gaba: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - RN Spp1 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "  Skipping volcano plot for Nestin - APN C1ql4 Glut: Not enough cells in one or both groups (< 50). Nestin_Ctrl=0, Nestin_Mut=1.\n",
      "\n",
      "Genotype-Specific DGE visualization finished.\n"
     ]
    }
   ],
   "source": [
    "# Genotype-Specific DGE (Mutant vs Control within each genotype) using the 'for_DEGs' layer, grouped by the DEG_BY\n",
    "print(f\"\\nRunning Genotype-Specific DGE (grouped by {DEG_BY})...\")\n",
    "dge_by_genotype = functions_degs.run_genotype_specific_dge(\n",
    "    adata,\n",
    "    grouping_key=DEG_BY,\n",
    "    dge_output_dir=DGE_OUTPUT_DIR,\n",
    "    plot_output_dir=PLOT_OUTPUT_DIR,\n",
    "    layer='for_DEGs' if 'for_DEGs' in adata.layers else None\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "39a4b0ca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running Genotype Comparison DGE (grouped by mapmycells_second_layer)...\n",
      "\n",
      "Starting DGE analysis: Genotype comparison within conditions...\n",
      "\n",
      "Processing group: Astro-TE NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 63 cells (max of 3 absolute and 10.0% of 626)...\n",
      "    Genes after filtering for Control: 6256\n",
      "      Calculating mean expression for 6256 genes (Control condition, group Astro-TE NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group Astro-TE NN).\n",
      "    Finished DGE for Control, group Astro-TE NN. Found 6256 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Astro-TE_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Astro-TE_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Astro-TE_NN' is writable.\n",
      "    Saved 1425 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Astro-TE_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 295 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Astro-TE_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 38 cells (max of 3 absolute and 10.0% of 379)...\n",
      "    Genes after filtering for Mutant: 4956\n",
      "      Calculating mean expression for 4956 genes (Mutant condition, group Astro-TE NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Astro-TE NN).\n",
      "    Finished DGE for Mutant, group Astro-TE NN. Found 4956 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Astro-TE_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Astro-TE_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Astro-TE_NN' is writable.\n",
      "    Saved 212 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Astro-TE_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 262 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Astro-TE_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: L6 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L6 IT CTX Glut: Not enough cells (< 50). Emx1_Control: 36, Nestin_Control: 31\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L6 IT CTX Glut: Not enough cells (< 50). Emx1_Mutant: 37, Nestin_Mutant: 55\n",
      "\n",
      "Processing group: CA3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 67 cells (max of 3 absolute and 10.0% of 666)...\n",
      "    Genes after filtering for Control: 10256\n",
      "      Calculating mean expression for 10256 genes (Control condition, group CA3 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group CA3 Glut).\n",
      "    Finished DGE for Control, group CA3 Glut. Found 10256 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_CA3_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_CA3_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA3_Glut' is writable.\n",
      "    Saved 1985 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA3_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 574 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA3_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 88 cells (max of 3 absolute and 10.0% of 880)...\n",
      "    Genes after filtering for Mutant: 10166\n",
      "      Calculating mean expression for 10166 genes (Mutant condition, group CA3 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group CA3 Glut).\n",
      "    Finished DGE for Mutant, group CA3 Glut. Found 10166 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CA3_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CA3_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA3_Glut' is writable.\n",
      "    Saved 1113 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA3_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 968 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA3_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Monocytes NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Monocytes NN: Not enough cells (< 50). Emx1_Control: 3, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Monocytes NN: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: L5 ET CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L5 ET CTX Glut: Not enough cells (< 50). Emx1_Control: 46, Nestin_Control: 50\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L5 ET CTX Glut: Not enough cells (< 50). Emx1_Mutant: 45, Nestin_Mutant: 114\n",
      "\n",
      "Processing group: L2/3 IT PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 189)...\n",
      "    Genes after filtering for Control: 10491\n",
      "      Calculating mean expression for 10491 genes (Control condition, group L2/3 IT PPP Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L2/3 IT PPP Glut).\n",
      "    Finished DGE for Control, group L2/3 IT PPP Glut. Found 10491 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_PPP_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_PPP_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Saved 90 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PPP_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 182 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PPP_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 86 cells (max of 3 absolute and 10.0% of 852)...\n",
      "    Genes after filtering for Mutant: 9625\n",
      "      Calculating mean expression for 9625 genes (Mutant condition, group L2/3 IT PPP Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L2/3 IT PPP Glut).\n",
      "    Finished DGE for Mutant, group L2/3 IT PPP Glut. Found 9625 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_PPP_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_PPP_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Saved 759 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PPP_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 933 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PPP_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: CA1-ProS Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 147 cells (max of 3 absolute and 10.0% of 1467)...\n",
      "    Genes after filtering for Control: 10280\n",
      "      Calculating mean expression for 10280 genes (Control condition, group CA1-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group CA1-ProS Glut).\n",
      "    Finished DGE for Control, group CA1-ProS Glut. Found 10280 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_CA1-ProS_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_CA1-ProS_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA1-ProS_Glut' is writable.\n",
      "    Saved 1192 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA1-ProS_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1177 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\CA1-ProS_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 192 cells (max of 3 absolute and 10.0% of 1912)...\n",
      "    Genes after filtering for Mutant: 9807\n",
      "      Calculating mean expression for 9807 genes (Mutant condition, group CA1-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group CA1-ProS Glut).\n",
      "    Finished DGE for Mutant, group CA1-ProS Glut. Found 9807 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CA1-ProS_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CA1-ProS_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA1-ProS_Glut' is writable.\n",
      "    Saved 1439 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA1-ProS_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1138 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CA1-ProS_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: MEA Slc17a7 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 54 cells (max of 3 absolute and 10.0% of 532)...\n",
      "    Genes after filtering for Control: 7772\n",
      "      Calculating mean expression for 7772 genes (Control condition, group MEA Slc17a7 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group MEA Slc17a7 Glut).\n",
      "    Finished DGE for Control, group MEA Slc17a7 Glut. Found 7772 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_MEA_Slc17a7_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_MEA_Slc17a7_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\MEA_Slc17a7_Glut' is writable.\n",
      "    Saved 15 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\MEA_Slc17a7_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 792 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\MEA_Slc17a7_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 30 cells (max of 3 absolute and 10.0% of 291)...\n",
      "    Genes after filtering for Mutant: 10094\n",
      "      Calculating mean expression for 10094 genes (Mutant condition, group MEA Slc17a7 Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group MEA Slc17a7 Glut).\n",
      "    Finished DGE for Mutant, group MEA Slc17a7 Glut. Found 10094 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_MEA_Slc17a7_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_MEA_Slc17a7_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\MEA_Slc17a7_Glut' is writable.\n",
      "    Saved 42 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\MEA_Slc17a7_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 70 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\MEA_Slc17a7_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Astro-OLF NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Astro-OLF NN: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 6\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Astro-OLF NN: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: COAp Grxcr2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group COAp Grxcr2 Glut: Not enough cells (< 50). Emx1_Control: 23, Nestin_Control: 7\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group COAp Grxcr2 Glut: Not enough cells (< 50). Emx1_Mutant: 25, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LA-BLA-BMA-PA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 36 cells (max of 3 absolute and 10.0% of 352)...\n",
      "    Genes after filtering for Control: 8064\n",
      "      Calculating mean expression for 8064 genes (Control condition, group LA-BLA-BMA-PA Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group LA-BLA-BMA-PA Glut).\n",
      "    Finished DGE for Control, group LA-BLA-BMA-PA Glut. Found 8064 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_LA-BLA-BMA-PA_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_LA-BLA-BMA-PA_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 33 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\LA-BLA-BMA-PA_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 189 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\LA-BLA-BMA-PA_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 29 cells (max of 3 absolute and 10.0% of 288)...\n",
      "    Genes after filtering for Mutant: 9248\n",
      "      Calculating mean expression for 9248 genes (Mutant condition, group LA-BLA-BMA-PA Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group LA-BLA-BMA-PA Glut).\n",
      "    Finished DGE for Mutant, group LA-BLA-BMA-PA Glut. Found 9248 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_LA-BLA-BMA-PA_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_LA-BLA-BMA-PA_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 94 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\LA-BLA-BMA-PA_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 195 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\LA-BLA-BMA-PA_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: DG Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 240 cells (max of 3 absolute and 10.0% of 2400)...\n",
      "    Genes after filtering for Control: 9649\n",
      "      Calculating mean expression for 9649 genes (Control condition, group DG Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group DG Glut).\n",
      "    Finished DGE for Control, group DG Glut. Found 9649 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_DG_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_DG_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\DG_Glut' is writable.\n",
      "    Saved 1769 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\DG_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1621 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\DG_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 133 cells (max of 3 absolute and 10.0% of 1324)...\n",
      "    Genes after filtering for Mutant: 8556\n",
      "      Calculating mean expression for 8556 genes (Mutant condition, group DG Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group DG Glut).\n",
      "    Finished DGE for Mutant, group DG Glut. Found 8556 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_DG_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_DG_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\DG_Glut' is writable.\n",
      "    Saved 877 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\DG_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1123 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\DG_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Oligo NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 94 cells (max of 3 absolute and 10.0% of 933)...\n",
      "    Genes after filtering for Control: 5369\n",
      "      Calculating mean expression for 5369 genes (Control condition, group Oligo NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group Oligo NN).\n",
      "    Finished DGE for Control, group Oligo NN. Found 5369 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Oligo_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Oligo_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Oligo_NN' is writable.\n",
      "    Saved 1332 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Oligo_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 285 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Oligo_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 90 cells (max of 3 absolute and 10.0% of 895)...\n",
      "    Genes after filtering for Mutant: 4325\n",
      "      Calculating mean expression for 4325 genes (Mutant condition, group Oligo NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Oligo NN).\n",
      "    Finished DGE for Mutant, group Oligo NN. Found 4325 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Oligo_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Oligo_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Oligo_NN' is writable.\n",
      "    Saved 504 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Oligo_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 224 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Oligo_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: NTS Phox2b Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NTS Phox2b Glut: Not enough cells (< 50). Emx1_Control: 6, Nestin_Control: 8\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NTS Phox2b Glut: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Sst Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 25 cells (max of 3 absolute and 10.0% of 250)...\n",
      "    Genes after filtering for Control: 10122\n",
      "      Calculating mean expression for 10122 genes (Control condition, group Sst Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group Sst Gaba).\n",
      "    Finished DGE for Control, group Sst Gaba. Found 10122 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Sst_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Sst_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Sst_Gaba' is writable.\n",
      "    Saved 161 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Sst_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 389 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Sst_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 46 cells (max of 3 absolute and 10.0% of 456)...\n",
      "    Genes after filtering for Mutant: 9436\n",
      "      Calculating mean expression for 9436 genes (Mutant condition, group Sst Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Sst Gaba).\n",
      "    Finished DGE for Mutant, group Sst Gaba. Found 9436 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Sst_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Sst_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Sst_Gaba' is writable.\n",
      "    Saved 417 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Sst_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 649 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Sst_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: L2/3 IT PIR-ENTl Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 38 cells (max of 3 absolute and 10.0% of 371)...\n",
      "    Genes after filtering for Control: 9830\n",
      "      Calculating mean expression for 9830 genes (Control condition, group L2/3 IT PIR-ENTl Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L2/3 IT PIR-ENTl Glut).\n",
      "    Finished DGE for Control, group L2/3 IT PIR-ENTl Glut. Found 9830 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_PIR-ENTl_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_PIR-ENTl_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Saved 30 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PIR-ENTl_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 2759 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_PIR-ENTl_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 33 cells (max of 3 absolute and 10.0% of 325)...\n",
      "    Genes after filtering for Mutant: 9428\n",
      "      Calculating mean expression for 9428 genes (Mutant condition, group L2/3 IT PIR-ENTl Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L2/3 IT PIR-ENTl Glut).\n",
      "    Finished DGE for Mutant, group L2/3 IT PIR-ENTl Glut. Found 9428 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_PIR-ENTl_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_PIR-ENTl_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Saved 147 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PIR-ENTl_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 495 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_PIR-ENTl_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: SUB-ProS Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 27 cells (max of 3 absolute and 10.0% of 267)...\n",
      "    Genes after filtering for Control: 11437\n",
      "      Calculating mean expression for 11437 genes (Control condition, group SUB-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group SUB-ProS Glut).\n",
      "    Finished DGE for Control, group SUB-ProS Glut. Found 11437 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_SUB-ProS_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_SUB-ProS_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\SUB-ProS_Glut' is writable.\n",
      "    Saved 790 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\SUB-ProS_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 323 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\SUB-ProS_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 40 cells (max of 3 absolute and 10.0% of 391)...\n",
      "    Genes after filtering for Mutant: 10435\n",
      "      Calculating mean expression for 10435 genes (Mutant condition, group SUB-ProS Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group SUB-ProS Glut).\n",
      "    Finished DGE for Mutant, group SUB-ProS Glut. Found 10435 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_SUB-ProS_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_SUB-ProS_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\SUB-ProS_Glut' is writable.\n",
      "    Saved 530 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\SUB-ProS_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 560 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\SUB-ProS_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Lymphoid NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Lymphoid NN: Not enough cells (< 50). Emx1_Control: 11, Nestin_Control: 14\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Lymphoid NN: Not enough cells (< 50). Emx1_Mutant: 10, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: L6b CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L6b CTX Glut: Not enough cells (< 50). Emx1_Control: 12, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L6b CTX Glut: Not enough cells (< 50). Emx1_Mutant: 13, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: L5 NP CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L5 NP CTX Glut: Not enough cells (< 50). Emx1_Control: 8, Nestin_Control: 9\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L5 NP CTX Glut: Not enough cells (< 50). Emx1_Mutant: 13, Nestin_Mutant: 45\n",
      "\n",
      "Processing group: L2/3 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 30 cells (max of 3 absolute and 10.0% of 297)...\n",
      "    Genes after filtering for Control: 9983\n",
      "      Calculating mean expression for 9983 genes (Control condition, group L2/3 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L2/3 IT CTX Glut).\n",
      "    Finished DGE for Control, group L2/3 IT CTX Glut. Found 9983 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Saved 46 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 168 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 56 cells (max of 3 absolute and 10.0% of 551)...\n",
      "    Genes after filtering for Mutant: 9952\n",
      "      Calculating mean expression for 9952 genes (Mutant condition, group L2/3 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L2/3 IT CTX Glut).\n",
      "    Finished DGE for Mutant, group L2/3 IT CTX Glut. Found 9952 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Saved 600 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 728 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: L2/3 IT ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 39 cells (max of 3 absolute and 10.0% of 389)...\n",
      "    Genes after filtering for Control: 11468\n",
      "      Calculating mean expression for 11468 genes (Control condition, group L2/3 IT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L2/3 IT ENT Glut).\n",
      "    Finished DGE for Control, group L2/3 IT ENT Glut. Found 11468 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_ENT_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L2_3_IT_ENT_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Saved 724 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_ENT_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 683 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L2_3_IT_ENT_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 85 cells (max of 3 absolute and 10.0% of 842)...\n",
      "    Genes after filtering for Mutant: 10600\n",
      "      Calculating mean expression for 10600 genes (Mutant condition, group L2/3 IT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L2/3 IT ENT Glut).\n",
      "    Finished DGE for Mutant, group L2/3 IT ENT Glut. Found 10600 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_ENT_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_3_IT_ENT_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Saved 787 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_ENT_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1192 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_3_IT_ENT_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: HPF CR Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group HPF CR Glut: Not enough cells (< 50). Emx1_Control: 6, Nestin_Control: 25\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group HPF CR Glut: Not enough cells (< 50). Emx1_Mutant: 5, Nestin_Mutant: 37\n",
      "\n",
      "Processing group: Microglia NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 41 cells (max of 3 absolute and 10.0% of 408)...\n",
      "    Genes after filtering for Control: 4484\n",
      "      Calculating mean expression for 4484 genes (Control condition, group Microglia NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group Microglia NN).\n",
      "    Finished DGE for Control, group Microglia NN. Found 4484 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Microglia_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Microglia_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Microglia_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Microglia_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Microglia_NN' is writable.\n",
      "    Saved 352 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Microglia_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 119 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Microglia_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Microglia NN: Not enough cells (< 50). Emx1_Mutant: 27, Nestin_Mutant: 496\n",
      "\n",
      "Processing group: Vip Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Vip Gaba: Not enough cells (< 50). Emx1_Control: 39, Nestin_Control: 62\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 22 cells (max of 3 absolute and 10.0% of 214)...\n",
      "    Genes after filtering for Mutant: 6940\n",
      "      Calculating mean expression for 6940 genes (Mutant condition, group Vip Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Vip Gaba).\n",
      "    Finished DGE for Mutant, group Vip Gaba. Found 6940 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Vip_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Vip_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Vip_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Vip_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Vip_Gaba' is writable.\n",
      "    Saved 101 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Vip_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 127 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Vip_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: L6b/CT ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 190)...\n",
      "    Genes after filtering for Control: 10022\n",
      "      Calculating mean expression for 10022 genes (Control condition, group L6b/CT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L6b/CT ENT Glut).\n",
      "    Finished DGE for Control, group L6b/CT ENT Glut. Found 10022 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L6b_CT_ENT_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L6b_CT_ENT_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6b_CT_ENT_Glut' is writable.\n",
      "    Saved 508 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6b_CT_ENT_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 134 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6b_CT_ENT_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 36 cells (max of 3 absolute and 10.0% of 355)...\n",
      "    Genes after filtering for Mutant: 8660\n",
      "      Calculating mean expression for 8660 genes (Mutant condition, group L6b/CT ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L6b/CT ENT Glut).\n",
      "    Finished DGE for Mutant, group L6b/CT ENT Glut. Found 8660 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L6b_CT_ENT_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L6b_CT_ENT_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6b_CT_ENT_Glut' is writable.\n",
      "    Saved 260 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6b_CT_ENT_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 533 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6b_CT_ENT_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: L2 IT ENT-po Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L2 IT ENT-po Glut: Not enough cells (< 50). Emx1_Control: 30, Nestin_Control: 44\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 25 cells (max of 3 absolute and 10.0% of 242)...\n",
      "    Genes after filtering for Mutant: 10255\n",
      "      Calculating mean expression for 10255 genes (Mutant condition, group L2 IT ENT-po Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L2 IT ENT-po Glut).\n",
      "    Finished DGE for Mutant, group L2 IT ENT-po Glut. Found 10255 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_IT_ENT-po_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L2_IT_ENT-po_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_IT_ENT-po_Glut' is writable.\n",
      "    Saved 396 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_IT_ENT-po_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 344 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L2_IT_ENT-po_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: OB-in Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB-in Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 5, Nestin_Control: 50\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB-in Frmd7 Gaba: Not enough cells (< 50). Emx1_Mutant: 11, Nestin_Mutant: 11\n",
      "\n",
      "Processing group: CA2-FC-IG Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CA2-FC-IG Glut: Not enough cells (< 50). Emx1_Control: 46, Nestin_Control: 47\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CA2-FC-IG Glut: Not enough cells (< 50). Emx1_Mutant: 55, Nestin_Mutant: 39\n",
      "\n",
      "Processing group: L6 CT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 34 cells (max of 3 absolute and 10.0% of 336)...\n",
      "    Genes after filtering for Control: 11415\n",
      "      Calculating mean expression for 11415 genes (Control condition, group L6 CT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L6 CT CTX Glut).\n",
      "    Finished DGE for Control, group L6 CT CTX Glut. Found 11415 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L6_CT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L6_CT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 244 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6_CT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 547 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L6_CT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 66 cells (max of 3 absolute and 10.0% of 655)...\n",
      "    Genes after filtering for Mutant: 11866\n",
      "      Calculating mean expression for 11866 genes (Mutant condition, group L6 CT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L6 CT CTX Glut).\n",
      "    Finished DGE for Mutant, group L6 CT CTX Glut. Found 11866 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L6_CT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L6_CT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 498 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6_CT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 1452 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L6_CT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: RHP-COA Ndnf Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group RHP-COA Ndnf Gaba: Not enough cells (< 50). Emx1_Control: 48, Nestin_Control: 31\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 14 cells (max of 3 absolute and 10.0% of 132)...\n",
      "    Genes after filtering for Mutant: 9608\n",
      "      Calculating mean expression for 9608 genes (Mutant condition, group RHP-COA Ndnf Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group RHP-COA Ndnf Gaba).\n",
      "    Finished DGE for Mutant, group RHP-COA Ndnf Gaba. Found 9608 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_RHP-COA_Ndnf_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_RHP-COA_Ndnf_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\RHP-COA_Ndnf_Gaba' is writable.\n",
      "    Saved 71 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\RHP-COA_Ndnf_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 133 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\RHP-COA_Ndnf_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: SC Lef1 Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SC Lef1 Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SC Lef1 Otx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DG-PIR Ex IMN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DG-PIR Ex IMN: Not enough cells (< 50). Emx1_Control: 15, Nestin_Control: 101\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DG-PIR Ex IMN: Not enough cells (< 50). Emx1_Mutant: 5, Nestin_Mutant: 37\n",
      "\n",
      "Processing group: OPC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OPC NN: Not enough cells (< 50). Emx1_Control: 40, Nestin_Control: 294\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 49 cells (max of 3 absolute and 10.0% of 488)...\n",
      "    Genes after filtering for Mutant: 4855\n",
      "      Calculating mean expression for 4855 genes (Mutant condition, group OPC NN)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group OPC NN).\n",
      "    Finished DGE for Mutant, group OPC NN. Found 4855 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_OPC_NN_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_OPC_NN_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\OPC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\OPC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\OPC_NN' is writable.\n",
      "    Saved 155 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\OPC_NN\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 157 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\OPC_NN\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Sncg Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Sncg Gaba: Not enough cells (< 50). Emx1_Control: 36, Nestin_Control: 21\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Sncg Gaba: Not enough cells (< 50). Emx1_Mutant: 71, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: L4/5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 15 cells (max of 3 absolute and 10.0% of 142)...\n",
      "    Genes after filtering for Control: 10628\n",
      "      Calculating mean expression for 10628 genes (Control condition, group L4/5 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group L4/5 IT CTX Glut).\n",
      "    Finished DGE for Control, group L4/5 IT CTX Glut. Found 10628 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L4_5_IT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_L4_5_IT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Saved 25 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L4_5_IT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 76 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\L4_5_IT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 36 cells (max of 3 absolute and 10.0% of 356)...\n",
      "    Genes after filtering for Mutant: 10150\n",
      "      Calculating mean expression for 10150 genes (Mutant condition, group L4/5 IT CTX Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L4/5 IT CTX Glut).\n",
      "    Finished DGE for Mutant, group L4/5 IT CTX Glut. Found 10150 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L4_5_IT_CTX_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L4_5_IT_CTX_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Saved 415 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L4_5_IT_CTX_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 354 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L4_5_IT_CTX_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: IA Mgp Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group IA Mgp Gaba: Not enough cells (< 50). Emx1_Control: 3, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IA Mgp Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LSX Prdm12 Slit2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Prdm12 Slit2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Prdm12 Slit2 Gaba: Not enough cells (< 50). Emx1_Mutant: 4, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: IT AON-TT-DP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 16 cells (max of 3 absolute and 10.0% of 152)...\n",
      "    Genes after filtering for Control: 6330\n",
      "      Calculating mean expression for 6330 genes (Control condition, group IT AON-TT-DP Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group IT AON-TT-DP Glut).\n",
      "    Finished DGE for Control, group IT AON-TT-DP Glut. Found 6330 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_IT_AON-TT-DP_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_IT_AON-TT-DP_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\IT_AON-TT-DP_Glut' is writable.\n",
      "    Saved 6 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\IT_AON-TT-DP_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 91 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\IT_AON-TT-DP_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IT AON-TT-DP Glut: Not enough cells (< 50). Emx1_Mutant: 23, Nestin_Mutant: 27\n",
      "\n",
      "Processing group: Lamp5 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Lamp5 Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 31, Nestin_Control: 32\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 15 cells (max of 3 absolute and 10.0% of 143)...\n",
      "    Genes after filtering for Mutant: 8895\n",
      "      Calculating mean expression for 8895 genes (Mutant condition, group Lamp5 Lhx6 Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Lamp5 Lhx6 Gaba).\n",
      "    Finished DGE for Mutant, group Lamp5 Lhx6 Gaba. Found 8895 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Lamp5_Lhx6_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Lamp5_Lhx6_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Lhx6_Gaba' is writable.\n",
      "    Saved 101 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Lhx6_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 125 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Lhx6_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: CEA-BST Ebf1 Pdyn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CEA-BST Ebf1 Pdyn Gaba: Not enough cells (< 50). Emx1_Control: 3, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CEA-BST Ebf1 Pdyn Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Lamp5 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Lamp5 Gaba: Not enough cells (< 50). Emx1_Control: 21, Nestin_Control: 45\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 12 cells (max of 3 absolute and 10.0% of 118)...\n",
      "    Genes after filtering for Mutant: 8501\n",
      "      Calculating mean expression for 8501 genes (Mutant condition, group Lamp5 Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Lamp5 Gaba).\n",
      "    Finished DGE for Mutant, group Lamp5 Gaba. Found 8501 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Lamp5_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Lamp5_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Gaba' is writable.\n",
      "    Saved 157 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 100 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Lamp5_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: IT EP-CLA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group IT EP-CLA Glut: Not enough cells (< 50). Emx1_Control: 20, Nestin_Control: 33\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IT EP-CLA Glut: Not enough cells (< 50). Emx1_Mutant: 36, Nestin_Mutant: 36\n",
      "\n",
      "Processing group: Pvalb Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Initial genes for Control condition: 26870\n",
      "    Filtering genes for Control: Requiring expression in >= 15 cells (max of 3 absolute and 10.0% of 144)...\n",
      "    Genes after filtering for Control: 9848\n",
      "      Calculating mean expression for 9848 genes (Control condition, group Pvalb Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Control condition, group Pvalb Gaba).\n",
      "    Finished DGE for Control, group Pvalb Gaba. Found 9848 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Pvalb_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_Pvalb_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Pvalb_Gaba' is writable.\n",
      "    Saved 261 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Pvalb_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 127 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Control\\Pvalb_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 29 cells (max of 3 absolute and 10.0% of 283)...\n",
      "    Genes after filtering for Mutant: 8842\n",
      "      Calculating mean expression for 8842 genes (Mutant condition, group Pvalb Gaba)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group Pvalb Gaba).\n",
      "    Finished DGE for Mutant, group Pvalb Gaba. Found 8842 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Pvalb_Gaba_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_Pvalb_Gaba_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Pvalb_Gaba' is writable.\n",
      "    Saved 224 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Pvalb_Gaba\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 392 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\Pvalb_Gaba\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: VLMC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group VLMC NN: Not enough cells (< 50). Emx1_Control: 30, Nestin_Control: 40\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group VLMC NN: Not enough cells (< 50). Emx1_Mutant: 53, Nestin_Mutant: 14\n",
      "\n",
      "Processing group: L6b EPd Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L6b EPd Glut: Not enough cells (< 50). Emx1_Control: 6, Nestin_Control: 7\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L6b EPd Glut: Not enough cells (< 50). Emx1_Mutant: 6, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: OB Meis2 Thsd7b Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB Meis2 Thsd7b Gaba: Not enough cells (< 50). Emx1_Control: 11, Nestin_Control: 14\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB Meis2 Thsd7b Gaba: Not enough cells (< 50). Emx1_Mutant: 9, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: L5/6 IT TPE-ENT Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L5/6 IT TPE-ENT Glut: Not enough cells (< 50). Emx1_Control: 41, Nestin_Control: 52\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 19 cells (max of 3 absolute and 10.0% of 183)...\n",
      "    Genes after filtering for Mutant: 10088\n",
      "      Calculating mean expression for 10088 genes (Mutant condition, group L5/6 IT TPE-ENT Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group L5/6 IT TPE-ENT Glut).\n",
      "    Finished DGE for Mutant, group L5/6 IT TPE-ENT Glut. Found 10088 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L5_6_IT_TPE-ENT_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_L5_6_IT_TPE-ENT_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L5_6_IT_TPE-ENT_Glut' is writable.\n",
      "    Saved 272 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L5_6_IT_TPE-ENT_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 224 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\L5_6_IT_TPE-ENT_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: IC Six3 En2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group IC Six3 En2 Gaba: Not enough cells (< 50). Emx1_Control: 4, Nestin_Control: 6\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IC Six3 En2 Gaba: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: ENTmv-PA-COAp Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group ENTmv-PA-COAp Glut: Not enough cells (< 50). Emx1_Control: 32, Nestin_Control: 44\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group ENTmv-PA-COAp Glut: Not enough cells (< 50). Emx1_Mutant: 35, Nestin_Mutant: 64\n",
      "\n",
      "Processing group: LDT Vsx2 Nkx6-1 Nfib Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: Astro-NT NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Astro-NT NN: Not enough cells (< 50). Emx1_Control: 7, Nestin_Control: 8\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Astro-NT NN: Not enough cells (< 50). Emx1_Mutant: 12, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: COAa-PAA-MEA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group COAa-PAA-MEA Barhl2 Glut: Not enough cells (< 50). Emx1_Control: 7, Nestin_Control: 16\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group COAa-PAA-MEA Barhl2 Glut: Not enough cells (< 50). Emx1_Mutant: 10, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: L5 IT CTX Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L5 IT CTX Glut: Not enough cells (< 50). Emx1_Control: 9, Nestin_Control: 17\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L5 IT CTX Glut: Not enough cells (< 50). Emx1_Mutant: 11, Nestin_Mutant: 26\n",
      "\n",
      "Processing group: SMC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SMC NN: Not enough cells (< 50). Emx1_Control: 6, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SMC NN: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: Endo NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Endo NN: Not enough cells (< 50). Emx1_Control: 26, Nestin_Control: 47\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Endo NN: Not enough cells (< 50). Emx1_Mutant: 41, Nestin_Mutant: 8\n",
      "\n",
      "Processing group: ABC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group ABC NN: Not enough cells (< 50). Emx1_Control: 10, Nestin_Control: 32\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group ABC NN: Not enough cells (< 50). Emx1_Mutant: 32, Nestin_Mutant: 13\n",
      "\n",
      "Processing group: CLA-EPd-CTX Car3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CLA-EPd-CTX Car3 Glut: Not enough cells (< 50). Emx1_Control: 21, Nestin_Control: 27\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CLA-EPd-CTX Car3 Glut: Not enough cells (< 50). Emx1_Mutant: 11, Nestin_Mutant: 27\n",
      "\n",
      "Processing group: SPVI-SPVC Sall3 Nfib Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CT SUB Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CT SUB Glut: Not enough cells (< 50). Emx1_Control: 24, Nestin_Control: 52\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 16 cells (max of 3 absolute and 10.0% of 160)...\n",
      "    Genes after filtering for Mutant: 8429\n",
      "      Calculating mean expression for 8429 genes (Mutant condition, group CT SUB Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group CT SUB Glut).\n",
      "    Finished DGE for Mutant, group CT SUB Glut. Found 8429 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CT_SUB_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_CT_SUB_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CT_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CT_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CT_SUB_Glut' is writable.\n",
      "    Saved 80 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CT_SUB_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 139 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\CT_SUB_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Peri NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Peri NN: Not enough cells (< 50). Emx1_Control: 9, Nestin_Control: 18\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Peri NN: Not enough cells (< 50). Emx1_Mutant: 9, Nestin_Mutant: 10\n",
      "\n",
      "Processing group: L2 IT PPP-APr Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L2 IT PPP-APr Glut: Not enough cells (< 50). Emx1_Control: 8, Nestin_Control: 19\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L2 IT PPP-APr Glut: Not enough cells (< 50). Emx1_Mutant: 24, Nestin_Mutant: 80\n",
      "\n",
      "Processing group: Pvalb chandelier Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Pvalb chandelier Gaba: Not enough cells (< 50). Emx1_Control: 12, Nestin_Control: 9\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Pvalb chandelier Gaba: Not enough cells (< 50). Emx1_Mutant: 15, Nestin_Mutant: 21\n",
      "\n",
      "Processing group: Ependymal NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Ependymal NN: Not enough cells (< 50). Emx1_Control: 7, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Ependymal NN: Not enough cells (< 50). Emx1_Mutant: 6, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: NP SUB Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NP SUB Glut: Not enough cells (< 50). Emx1_Control: 25, Nestin_Control: 80\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Initial genes for Mutant condition: 26870\n",
      "    Filtering genes for Mutant: Requiring expression in >= 15 cells (max of 3 absolute and 10.0% of 141)...\n",
      "    Genes after filtering for Mutant: 9041\n",
      "      Calculating mean expression for 9041 genes (Mutant condition, group NP SUB Glut)...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns (Mutant condition, group NP SUB Glut).\n",
      "    Finished DGE for Mutant, group NP SUB Glut. Found 9041 ranked genes.\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_NP_SUB_Glut_Nestin_vs_Emx1.csv\n",
      "    DGE results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_NP_SUB_Glut_Nestin_vs_Emx1.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\NP_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\NP_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\NP_SUB_Glut' is writable.\n",
      "    Saved 91 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\NP_SUB_Glut\\Nestin_vs_Emx1_up_significant.csv\n",
      "    Saved 165 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\sig_deg_lists\\Mutant\\NP_SUB_Glut\\Nestin_vs_Emx1_down_significant.csv\n",
      "\n",
      "Processing group: Tanycyte NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Tanycyte NN: Not enough cells (< 50). Emx1_Control: 4, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Tanycyte NN: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB-mi Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB-mi Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB-mi Frmd7 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB Trdn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB Trdn Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB Trdn Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVR Six3 Sox3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVR Six3 Sox3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVR Six3 Sox3 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-BST Sox6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MEA-BST Sox6 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 9\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MEA-BST Sox6 Gaba: Not enough cells (< 50). Emx1_Mutant: 4, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: STR D2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR D2 Gaba: Not enough cells (< 50). Emx1_Control: 5, Nestin_Control: 19\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR D2 Gaba: Not enough cells (< 50). Emx1_Mutant: 13, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: CHOR NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CHOR NN: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CHOR NN: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 8\n",
      "\n",
      "Processing group: IC Tfap2d Maf Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group IC Tfap2d Maf Glut: Not enough cells (< 50). Emx1_Control: 3, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IC Tfap2d Maf Glut: Not enough cells (< 50). Emx1_Mutant: 4, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: NI-RPO Gata3 Nr4a2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NI-RPO Gata3 Nr4a2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NI-RPO Gata3 Nr4a2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NLL Gata3 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NLL Gata3 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 12\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NLL Gata3 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: LSX Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Otx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MH Tac2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MH Tac2 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MH Tac2 Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: DTN-LDT-IPN Otp Pax3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OB Eomes Ms4a15 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB Eomes Ms4a15 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 13\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB Eomes Ms4a15 Glut: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NTS-PARN Neurod2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NTS-PARN Neurod2 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NTS-PARN Neurod2 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: STR Prox1 Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR Prox1 Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR Prox1 Lhx6 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: OB-STR-CTX Inh IMN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB-STR-CTX Inh IMN: Not enough cells (< 50). Emx1_Control: 14, Nestin_Control: 17\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB-STR-CTX Inh IMN: Not enough cells (< 50). Emx1_Mutant: 13, Nestin_Mutant: 15\n",
      "\n",
      "Processing group: LSX Nkx2-1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Nkx2-1 Gaba: Not enough cells (< 50). Emx1_Control: 5, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Nkx2-1 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: STR D1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR D1 Gaba: Not enough cells (< 50). Emx1_Control: 5, Nestin_Control: 5\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR D1 Gaba: Not enough cells (< 50). Emx1_Mutant: 7, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MY Lhx1 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MY Lhx1 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MY Lhx1 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: TRS-BAC Sln Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group TRS-BAC Sln Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group TRS-BAC Sln Glut: Not enough cells (< 50). Emx1_Mutant: 4, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CB Granule Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CB Granule Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CB Granule Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: L2/3 IT RSP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L2/3 IT RSP Glut: Not enough cells (< 50). Emx1_Control: 5, Nestin_Control: 6\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L2/3 IT RSP Glut: Not enough cells (< 50). Emx1_Mutant: 5, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: PRT Mecom Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRT Mecom Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRT Mecom Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: OB Dopa-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB Dopa-Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 19\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB Dopa-Gaba: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MDRNd Prox1 Pax6 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Nr2e1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: CB PLI Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CB PLI Gly-Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CB PLI Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: BAM NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group BAM NN: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 16\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group BAM NN: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: RT-ZI Gnb3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group RT-ZI Gnb3 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group RT-ZI Gnb3 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SPVC Mafa Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SPVC Mafa Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SPVC Mafa Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: TH Prkcd Grin2c Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group TH Prkcd Grin2c Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group TH Prkcd Grin2c Glut: Not enough cells (< 50). Emx1_Mutant: 8, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SNc-VTA-RAmb Foxa1 Dopa (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SNc-VTA-RAmb Foxa1 Dopa: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SNc-VTA-RAmb Foxa1 Dopa: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PGRN-PARN-MDRN Hoxb5 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 9\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: VCO Mafa Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group VCO Mafa Meis2 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group VCO Mafa Meis2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR-PAL Chst9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR-PAL Chst9 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 7\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR-PAL Chst9 Gaba: Not enough cells (< 50). Emx1_Mutant: 15, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: NP PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NP PPP Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NP PPP Glut: Not enough cells (< 50). Emx1_Mutant: 6, Nestin_Mutant: 14\n",
      "\n",
      "Processing group: SC-PAG Lef1 Emx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SC-PAG Lef1 Emx2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SC-PAG Lef1 Emx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: DC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DC NN: Not enough cells (< 50). Emx1_Control: 4, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DC NN: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: SPVI-SPVC Tlx3 Ebf3 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 8\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: OB-out Frmd7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OB-out Frmd7 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 48\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OB-out Frmd7 Gaba: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: CS-RPO Meis2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CS-RPO Meis2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CS-RPO Meis2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: AHN-SBPV-PVHd Pdrm12 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: LSX Prdm12 Zeb2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Prdm12 Zeb2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Prdm12 Zeb2 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LDT-PCG-CS Gata3 Lhx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCs Pax7 Nfia Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCs Pax7 Nfia Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCs Pax7 Nfia Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: ADP-MPO Trp73 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group ADP-MPO Trp73 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group ADP-MPO Trp73 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: Sst Chodl Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Sst Chodl Gaba: Not enough cells (< 50). Emx1_Control: 2, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Sst Chodl Gaba: Not enough cells (< 50). Emx1_Mutant: 7, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: OEC NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OEC NN: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 39\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OEC NN: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NLOT Rho Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NLOT Rho Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NLOT Rho Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCsg Gabrr2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCsg Gabrr2 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 7\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCsg Gabrr2 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 6\n",
      "\n",
      "Processing group: PAG-ND-PCG Onecut1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-ND-PCG Onecut1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-ND-PCG Onecut1 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MB-MY Tph2 Glut-Sero (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MB-MY Tph2 Glut-Sero: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MB-MY Tph2 Glut-Sero: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PG-TRN-LRN Fat2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PG-TRN-LRN Fat2 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PG-TRN-LRN Fat2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LSX Sall3 Lmo1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Sall3 Lmo1 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Sall3 Lmo1 Gaba: Not enough cells (< 50). Emx1_Mutant: 9, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PRC-PAG Pax6 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRC-PAG Pax6 Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRC-PAG Pax6 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCsg Pde5a Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCsg Pde5a Glut: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCsg Pde5a Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Nfib Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MEA-BST Lhx6 Nfib Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MEA-BST Lhx6 Nfib Gaba: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MEA-BST Lhx6 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MEA-BST Lhx6 Sp9 Gaba: Not enough cells (< 50). Emx1_Control: 1, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MEA-BST Lhx6 Sp9 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CEA-BST Six3 Cyp26b1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CEA-BST Six3 Cyp26b1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 5\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CEA-BST Six3 Cyp26b1 Gaba: Not enough cells (< 50). Emx1_Mutant: 5, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: Bergmann NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Bergmann NN: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Bergmann NN: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: L5 PPP Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L5 PPP Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L5 PPP Glut: Not enough cells (< 50). Emx1_Mutant: 6, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: SCig Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCig Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 6\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCig Foxb1 Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 7\n",
      "\n",
      "Processing group: SI-MA-LPO-LHA Skor1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SI-MA-LPO-LHA Skor1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 7\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SI-MA-LPO-LHA Skor1 Glut: Not enough cells (< 50). Emx1_Mutant: 6, Nestin_Mutant: 5\n",
      "\n",
      "Processing group: MV-SPIV Zic4 Neurod2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MV-SPIV Zic4 Neurod2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MV-SPIV Zic4 Neurod2 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LSX Sall3 Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LSX Sall3 Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LSX Sall3 Pax6 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NDB-SI-MA-STRv Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MG-POL-SGN Nts Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MG-POL-SGN Nts Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MG-POL-SGN Nts Glut: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: LGv-ZI Otx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LGv-ZI Otx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LGv-ZI Otx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PH-SUM Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PH-SUM Foxa1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PH-SUM Foxa1 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PH-LHA Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PH-LHA Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PH-LHA Foxb1 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LDT-PCG St18 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LDT-PCG St18 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LDT-PCG St18 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCH Six6 Cdc14a Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCH Six6 Cdc14a Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCH Six6 Cdc14a Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: L4 RSP-ACA Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group L4 RSP-ACA Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group L4 RSP-ACA Glut: Not enough cells (< 50). Emx1_Mutant: 3, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PRT Tcf7l2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRT Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 6\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRT Tcf7l2 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 9\n",
      "\n",
      "Processing group: PRP Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRP Otp Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRP Otp Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: DMH Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DMH Hmx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DMH Hmx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STN-PSTN Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STN-PSTN Pitx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STN-PSTN Pitx2 Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: OT D3 Folh1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group OT D3 Folh1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group OT D3 Folh1 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-RN Nkx2-2 Otx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells (< 50). Emx1_Mutant: 2, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MRN-PPN-CUN Pax8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MRN-PPN-CUN Pax8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MRN-PPN-CUN Pax8 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCs Lef1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCs Lef1 Gli3 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCs Lef1 Gli3 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PDTg-PCG Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PDTg-PCG Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PDTg-PCG Pax6 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PH-ant-LHA Otp Bsx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PH-ant-LHA Otp Bsx Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PH-ant-LHA Otp Bsx Glut: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: ZI Pax6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group ZI Pax6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group ZI Pax6 Gaba: Not enough cells (< 50). Emx1_Mutant: 1, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: LDT-PCG Vsx2 Lhx4 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LDT-PCG Vsx2 Lhx4 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LDT-PCG Vsx2 Lhx4 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CEA-AAA-BST Six3 Sp9 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PAG-SC Neurod2 Meis2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-SC Neurod2 Meis2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 4\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-SC Neurod2 Meis2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: MPN-MPO-PVpo Hmx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MPN-MPO-PVpo Hmx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MPN-MPO-PVpo Hmx2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: BST-MPN Six3 Nrgn Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group BST-MPN Six3 Nrgn Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group BST-MPN Six3 Nrgn Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-PPN Pax5 Sox21 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-PPN Pax5 Sox21 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 3\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-PPN Pax5 Sox21 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PRNc Otp Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRNc Otp Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 5\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRNc Otp Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 3\n",
      "\n",
      "Processing group: MPO-ADP Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MPO-ADP Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MPO-ADP Lhx8 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MS-SF Bsx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MS-SF Bsx Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MS-SF Bsx Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PB Sst Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PB Sst Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PB Sst Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SI-MA-ACB Ebf1 Bnc2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: NDB-SI-ant Prdm12 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group NDB-SI-ant Prdm12 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group NDB-SI-ant Prdm12 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SI-MPO-LPO Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SI-MPO-LPO Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SI-MPO-LPO Lhx8 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: Pineal Crx Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Pineal Crx Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Pineal Crx Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG Pou4f2 Mesi2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG Pou4f2 Mesi2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG Pou4f2 Mesi2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LHA Barhl2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LHA Barhl2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LHA Barhl2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SBPV-PVa Six6 Satb2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SBPV-PVa Six6 Satb2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SBPV-PVa Six6 Satb2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR D1 Sema5a Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR D1 Sema5a Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR D1 Sema5a Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SC Bnc2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SC Bnc2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SC Bnc2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SC Tnnt1 Gli3 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SC Tnnt1 Gli3 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SC Tnnt1 Gli3 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVHd-SBPV Six3 Prox1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVHd-SBPV Six3 Prox1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVHd-SBPV Six3 Prox1 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: SCm-PAG Cdh23 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCm-PAG Cdh23 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCm-PAG Cdh23 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCop Sln Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCop Sln Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCop Sln Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CS-PRNr-DR En1 Sox2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CS-PRNr-DR En1 Sox2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CS-PRNr-DR En1 Sox2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PAG-SC Pou4f1 Zic1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-SC Pou4f1 Zic1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-SC Pou4f1 Zic1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: TU-ARH Otp Six6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group TU-ARH Otp Six6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group TU-ARH Otp Six6 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: VMH Nr5a1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group VMH Nr5a1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group VMH Nr5a1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PARN-MDRNd-NTS Gbx2 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVpo-VMPO-MPN Hmx2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 2\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MY Prox1 Lmo7 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MY Prox1 Lmo7 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MY Prox1 Lmo7 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVH-SO-PVa Otp Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVH-SO-PVa Otp Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVH-SO-PVa Otp Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PRC-PAG Tcf7l2 Irx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: IF-RL-CLI-PAG Foxa1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group IF-RL-CLI-PAG Foxa1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group IF-RL-CLI-PAG Foxa1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: CU-ECU-SPVI Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CU-ECU-SPVI Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CU-ECU-SPVI Foxb1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: STR Lhx8 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group STR Lhx8 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group STR Lhx8 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: MEA-COA-BMA Ccdc42 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MEA-COA-BMA Ccdc42 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MEA-COA-BMA Ccdc42 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: CS-PRNr-PCG Tmem163 Otp Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 0\n",
      "\n",
      "Processing group: PVT-PT Ntrk1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVT-PT Ntrk1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 1\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVT-PT Ntrk1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PVHd-DMH Lhx6 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PVHd-DMH Lhx6 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PVHd-DMH Lhx6 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: LDT Fgf7 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LDT Fgf7 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LDT Fgf7 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PPN-CUN-PCG Otp En1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PPN-CUN-PCG Otp En1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PPN-CUN-PCG Otp En1 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: PH-an Pitx2 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PH-an Pitx2 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PH-an Pitx2 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCig-an-PPT Foxb1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCig-an-PPT Foxb1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCig-an-PPT Foxb1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: PAG-MRN-RN Foxa2 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group PAG-MRN-RN Foxa2 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group PAG-MRN-RN Foxa2 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 2\n",
      "\n",
      "Processing group: Astroependymal NN (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group Astroependymal NN: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group Astroependymal NN: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: MV-SPIV Phox2b Ebf3 Lbx1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: SCs Dmbx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group SCs Dmbx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group SCs Dmbx1 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: LH Pou4f1 Sox1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group LH Pou4f1 Sox1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group LH Pou4f1 Sox1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 4\n",
      "\n",
      "Processing group: RPA Pax6 Hoxb5 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DCO Il22 Gly-Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DCO Il22 Gly-Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DCO Il22 Gly-Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: DMH-LHA Gsx1 Gaba (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group DMH-LHA Gsx1 Gaba: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group DMH-LHA Gsx1 Gaba: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: RN Spp1 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group RN Spp1 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group RN Spp1 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Processing group: APN C1ql4 Glut (using key 'mapmycells_second_layer')\n",
      "  Running DGE for Control condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Control condition, group APN C1ql4 Glut: Not enough cells (< 50). Emx1_Control: 0, Nestin_Control: 0\n",
      "  Running DGE for Mutant condition (Nestin vs Emx1)...\n",
      "    Skipping DGE for Mutant condition, group APN C1ql4 Glut: Not enough cells (< 50). Emx1_Mutant: 0, Nestin_Mutant: 1\n",
      "\n",
      "Summary of skipped groups (using key 'mapmycells_second_layer') for Control condition comparison saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Control_cond_comparison_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "Summary of skipped groups (using key 'mapmycells_second_layer') for Mutant condition comparison saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\\cond_spec_geno_comp\\dge_Mutant_cond_comparison_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "DGE analysis (Genotype comparison within conditions) finished.\n",
      "\n",
      "Generating Volcano Plots for Genotype comparison within conditions...\n",
      "\n",
      "--- Visualizing for Condition: Control (Nestin vs Emx1) ---\n",
      "  Generating volcano plot for: Control Cond - Astro-TE NN (Counts: Emx1_Control=152, Nestin_Control=474) -> dge_Control_cond_Nestin_vs_Emx1_Astro-TE_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - L6 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Control=36, Nestin_Control=31.\n",
      "  Generating volcano plot for: Control Cond - CA3 Glut (Counts: Emx1_Control=379, Nestin_Control=287) -> dge_Control_cond_Nestin_vs_Emx1_CA3_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - Monocytes NN: Not enough cells in one or both groups (< 50). Emx1_Control=3, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - L5 ET CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Control=46, Nestin_Control=50.\n",
      "  Generating volcano plot for: Control Cond - L2/3 IT PPP Glut (Counts: Emx1_Control=54, Nestin_Control=135) -> dge_Control_cond_Nestin_vs_Emx1_L2_3_IT_PPP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - CA1-ProS Glut (Counts: Emx1_Control=664, Nestin_Control=803) -> dge_Control_cond_Nestin_vs_Emx1_CA1-ProS_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - MEA Slc17a7 Glut (Counts: Emx1_Control=103, Nestin_Control=429) -> dge_Control_cond_Nestin_vs_Emx1_MEA_Slc17a7_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - Astro-OLF NN: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=6.\n",
      "  Skipping volcano plot for Control Cond - COAp Grxcr2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=23, Nestin_Control=7.\n",
      "  Generating volcano plot for: Control Cond - LA-BLA-BMA-PA Glut (Counts: Emx1_Control=135, Nestin_Control=217) -> dge_Control_cond_Nestin_vs_Emx1_LA-BLA-BMA-PA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - DG Glut (Counts: Emx1_Control=846, Nestin_Control=1554) -> dge_Control_cond_Nestin_vs_Emx1_DG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - Oligo NN (Counts: Emx1_Control=133, Nestin_Control=800) -> dge_Control_cond_Nestin_vs_Emx1_Oligo_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - NTS Phox2b Glut: Not enough cells in one or both groups (< 50). Emx1_Control=6, Nestin_Control=8.\n",
      "  Generating volcano plot for: Control Cond - Sst Gaba (Counts: Emx1_Control=106, Nestin_Control=144) -> dge_Control_cond_Nestin_vs_Emx1_Sst_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - L2/3 IT PIR-ENTl Glut (Counts: Emx1_Control=139, Nestin_Control=232) -> dge_Control_cond_Nestin_vs_Emx1_L2_3_IT_PIR-ENTl_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - SUB-ProS Glut (Counts: Emx1_Control=114, Nestin_Control=153) -> dge_Control_cond_Nestin_vs_Emx1_SUB-ProS_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - Lymphoid NN: Not enough cells in one or both groups (< 50). Emx1_Control=11, Nestin_Control=14.\n",
      "  Skipping volcano plot for Control Cond - L6b CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Control=12, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - L5 NP CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Control=8, Nestin_Control=9.\n",
      "  Generating volcano plot for: Control Cond - L2/3 IT CTX Glut (Counts: Emx1_Control=89, Nestin_Control=208) -> dge_Control_cond_Nestin_vs_Emx1_L2_3_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Control Cond - L2/3 IT ENT Glut (Counts: Emx1_Control=186, Nestin_Control=203) -> dge_Control_cond_Nestin_vs_Emx1_L2_3_IT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - HPF CR Glut: Not enough cells in one or both groups (< 50). Emx1_Control=6, Nestin_Control=25.\n",
      "  Generating volcano plot for: Control Cond - Microglia NN (Counts: Emx1_Control=53, Nestin_Control=355) -> dge_Control_cond_Nestin_vs_Emx1_Microglia_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - Vip Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=39, Nestin_Control=62.\n",
      "  Generating volcano plot for: Control Cond - L6b/CT ENT Glut (Counts: Emx1_Control=54, Nestin_Control=136) -> dge_Control_cond_Nestin_vs_Emx1_L6b_CT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - L2 IT ENT-po Glut: Not enough cells in one or both groups (< 50). Emx1_Control=30, Nestin_Control=44.\n",
      "  Skipping volcano plot for Control Cond - OB-in Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=5, Nestin_Control=50.\n",
      "  Skipping volcano plot for Control Cond - CA2-FC-IG Glut: Not enough cells in one or both groups (< 50). Emx1_Control=46, Nestin_Control=47.\n",
      "  Generating volcano plot for: Control Cond - L6 CT CTX Glut (Counts: Emx1_Control=82, Nestin_Control=254) -> dge_Control_cond_Nestin_vs_Emx1_L6_CT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - RHP-COA Ndnf Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=48, Nestin_Control=31.\n",
      "  Skipping volcano plot for Control Cond - SC Lef1 Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - DG-PIR Ex IMN: Not enough cells in one or both groups (< 50). Emx1_Control=15, Nestin_Control=101.\n",
      "  Skipping volcano plot for Control Cond - OPC NN: Not enough cells in one or both groups (< 50). Emx1_Control=40, Nestin_Control=294.\n",
      "  Skipping volcano plot for Control Cond - Sncg Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=36, Nestin_Control=21.\n",
      "  Generating volcano plot for: Control Cond - L4/5 IT CTX Glut (Counts: Emx1_Control=87, Nestin_Control=55) -> dge_Control_cond_Nestin_vs_Emx1_L4_5_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - IA Mgp Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=3, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - LSX Prdm12 Slit2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=3.\n",
      "  Generating volcano plot for: Control Cond - IT AON-TT-DP Glut (Counts: Emx1_Control=55, Nestin_Control=97) -> dge_Control_cond_Nestin_vs_Emx1_IT_AON-TT-DP_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - Lamp5 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=31, Nestin_Control=32.\n",
      "  Skipping volcano plot for Control Cond - CEA-BST Ebf1 Pdyn Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=3, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - Lamp5 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=21, Nestin_Control=45.\n",
      "  Skipping volcano plot for Control Cond - IT EP-CLA Glut: Not enough cells in one or both groups (< 50). Emx1_Control=20, Nestin_Control=33.\n",
      "  Generating volcano plot for: Control Cond - Pvalb Gaba (Counts: Emx1_Control=67, Nestin_Control=77) -> dge_Control_cond_Nestin_vs_Emx1_Pvalb_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Control Cond - VLMC NN: Not enough cells in one or both groups (< 50). Emx1_Control=30, Nestin_Control=40.\n",
      "  Skipping volcano plot for Control Cond - L6b EPd Glut: Not enough cells in one or both groups (< 50). Emx1_Control=6, Nestin_Control=7.\n",
      "  Skipping volcano plot for Control Cond - OB Meis2 Thsd7b Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=11, Nestin_Control=14.\n",
      "  Skipping volcano plot for Control Cond - L5/6 IT TPE-ENT Glut: Not enough cells in one or both groups (< 50). Emx1_Control=41, Nestin_Control=52.\n",
      "  Skipping volcano plot for Control Cond - IC Six3 En2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=4, Nestin_Control=6.\n",
      "  Skipping volcano plot for Control Cond - ENTmv-PA-COAp Glut: Not enough cells in one or both groups (< 50). Emx1_Control=32, Nestin_Control=44.\n",
      "  Skipping volcano plot for Control Cond - LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - Astro-NT NN: Not enough cells in one or both groups (< 50). Emx1_Control=7, Nestin_Control=8.\n",
      "  Skipping volcano plot for Control Cond - COAa-PAA-MEA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=7, Nestin_Control=16.\n",
      "  Skipping volcano plot for Control Cond - L5 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Control=9, Nestin_Control=17.\n",
      "  Skipping volcano plot for Control Cond - SMC NN: Not enough cells in one or both groups (< 50). Emx1_Control=6, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - Endo NN: Not enough cells in one or both groups (< 50). Emx1_Control=26, Nestin_Control=47.\n",
      "  Skipping volcano plot for Control Cond - ABC NN: Not enough cells in one or both groups (< 50). Emx1_Control=10, Nestin_Control=32.\n",
      "  Skipping volcano plot for Control Cond - CLA-EPd-CTX Car3 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=21, Nestin_Control=27.\n",
      "  Skipping volcano plot for Control Cond - SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CT SUB Glut: Not enough cells in one or both groups (< 50). Emx1_Control=24, Nestin_Control=52.\n",
      "  Skipping volcano plot for Control Cond - Peri NN: Not enough cells in one or both groups (< 50). Emx1_Control=9, Nestin_Control=18.\n",
      "  Skipping volcano plot for Control Cond - L2 IT PPP-APr Glut: Not enough cells in one or both groups (< 50). Emx1_Control=8, Nestin_Control=19.\n",
      "  Skipping volcano plot for Control Cond - Pvalb chandelier Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=12, Nestin_Control=9.\n",
      "  Skipping volcano plot for Control Cond - Ependymal NN: Not enough cells in one or both groups (< 50). Emx1_Control=7, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - NP SUB Glut: Not enough cells in one or both groups (< 50). Emx1_Control=25, Nestin_Control=80.\n",
      "  Skipping volcano plot for Control Cond - Tanycyte NN: Not enough cells in one or both groups (< 50). Emx1_Control=4, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - OB-mi Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - OB Trdn Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - PVR Six3 Sox3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MEA-BST Sox6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=9.\n",
      "  Skipping volcano plot for Control Cond - STR D2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=5, Nestin_Control=19.\n",
      "  Skipping volcano plot for Control Cond - CHOR NN: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - IC Tfap2d Maf Glut: Not enough cells in one or both groups (< 50). Emx1_Control=3, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - NI-RPO Gata3 Nr4a2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - NLL Gata3 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=12.\n",
      "  Skipping volcano plot for Control Cond - LSX Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - MH Tac2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - OB Eomes Ms4a15 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=13.\n",
      "  Skipping volcano plot for Control Cond - NTS-PARN Neurod2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - STR Prox1 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - OB-STR-CTX Inh IMN: Not enough cells in one or both groups (< 50). Emx1_Control=14, Nestin_Control=17.\n",
      "  Skipping volcano plot for Control Cond - LSX Nkx2-1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=5, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - STR D1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=5, Nestin_Control=5.\n",
      "  Skipping volcano plot for Control Cond - MY Lhx1 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - TRS-BAC Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - CB Granule Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - L2/3 IT RSP Glut: Not enough cells in one or both groups (< 50). Emx1_Control=5, Nestin_Control=6.\n",
      "  Skipping volcano plot for Control Cond - PRT Mecom Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - OB Dopa-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=19.\n",
      "  Skipping volcano plot for Control Cond - MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CB PLI Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - BAM NN: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=16.\n",
      "  Skipping volcano plot for Control Cond - RT-ZI Gnb3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SPVC Mafa Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - TH Prkcd Grin2c Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SNc-VTA-RAmb Foxa1 Dopa: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=9.\n",
      "  Skipping volcano plot for Control Cond - VCO Mafa Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - STR-PAL Chst9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=7.\n",
      "  Skipping volcano plot for Control Cond - NP PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - SC-PAG Lef1 Emx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - DC NN: Not enough cells in one or both groups (< 50). Emx1_Control=4, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=8.\n",
      "  Skipping volcano plot for Control Cond - OB-out Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=48.\n",
      "  Skipping volcano plot for Control Cond - CS-RPO Meis2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - LSX Prdm12 Zeb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - SCs Pax7 Nfia Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - ADP-MPO Trp73 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - Sst Chodl Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=2, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - OEC NN: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=39.\n",
      "  Skipping volcano plot for Control Cond - NLOT Rho Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SCsg Gabrr2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=7.\n",
      "  Skipping volcano plot for Control Cond - PAG-ND-PCG Onecut1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - MB-MY Tph2 Glut-Sero: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PG-TRN-LRN Fat2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - LSX Sall3 Lmo1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PRC-PAG Pax6 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SCsg Pde5a Glut: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MEA-BST Lhx6 Nfib Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - MEA-BST Lhx6 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=1, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - CEA-BST Six3 Cyp26b1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=5.\n",
      "  Skipping volcano plot for Control Cond - Bergmann NN: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - L5 PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - SCig Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=6.\n",
      "  Skipping volcano plot for Control Cond - SI-MA-LPO-LHA Skor1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=7.\n",
      "  Skipping volcano plot for Control Cond - MV-SPIV Zic4 Neurod2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - LSX Sall3 Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - MG-POL-SGN Nts Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - LGv-ZI Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PH-SUM Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - PH-LHA Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - LDT-PCG St18 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SCH Six6 Cdc14a Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - L4 RSP-ACA Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - PRT Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=6.\n",
      "  Skipping volcano plot for Control Cond - PRP Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - DMH Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - STN-PSTN Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - OT D3 Folh1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MRN-PPN-CUN Pax8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SCs Lef1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PDTg-PCG Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PH-ant-LHA Otp Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - ZI Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - LDT-PCG Vsx2 Lhx4 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - PAG-SC Neurod2 Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=4.\n",
      "  Skipping volcano plot for Control Cond - MPN-MPO-PVpo Hmx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - BST-MPN Six3 Nrgn Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PAG-PPN Pax5 Sox21 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=3.\n",
      "  Skipping volcano plot for Control Cond - PRNc Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=5.\n",
      "  Skipping volcano plot for Control Cond - MPO-ADP Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MS-SF Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - PB Sst Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - NDB-SI-ant Prdm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - SI-MPO-LPO Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - Pineal Crx Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - PAG Pou4f2 Mesi2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - LHA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - SBPV-PVa Six6 Satb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - STR D1 Sema5a Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SC Bnc2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SC Tnnt1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PVHd-SBPV Six3 Prox1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - SCm-PAG Cdh23 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - SCop Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CS-PRNr-DR En1 Sox2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PAG-SC Pou4f1 Zic1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - TU-ARH Otp Six6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - VMH Nr5a1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=2.\n",
      "  Skipping volcano plot for Control Cond - MY Prox1 Lmo7 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PVH-SO-PVa Otp Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - IF-RL-CLI-PAG Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CU-ECU-SPVI Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - STR Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - MEA-COA-BMA Ccdc42 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PVT-PT Ntrk1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=1.\n",
      "  Skipping volcano plot for Control Cond - PVHd-DMH Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - LDT Fgf7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PPN-CUN-PCG Otp En1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PH-an Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SCig-an-PPT Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - PAG-MRN-RN Foxa2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - Astroependymal NN: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - SCs Dmbx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - LH Pou4f1 Sox1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - DCO Il22 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - DMH-LHA Gsx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - RN Spp1 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "  Skipping volcano plot for Control Cond - APN C1ql4 Glut: Not enough cells in one or both groups (< 50). Emx1_Control=0, Nestin_Control=0.\n",
      "\n",
      "--- Visualizing for Condition: Mutant (Nestin vs Emx1) ---\n",
      "  Generating volcano plot for: Mutant Cond - Astro-TE NN (Counts: Emx1_Mutant=220, Nestin_Mutant=159) -> dge_Mutant_cond_Nestin_vs_Emx1_Astro-TE_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - L6 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=37, Nestin_Mutant=55.\n",
      "  Generating volcano plot for: Mutant Cond - CA3 Glut (Counts: Emx1_Mutant=463, Nestin_Mutant=417) -> dge_Mutant_cond_Nestin_vs_Emx1_CA3_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Monocytes NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - L5 ET CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=45, Nestin_Mutant=114.\n",
      "  Generating volcano plot for: Mutant Cond - L2/3 IT PPP Glut (Counts: Emx1_Mutant=410, Nestin_Mutant=442) -> dge_Mutant_cond_Nestin_vs_Emx1_L2_3_IT_PPP_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - CA1-ProS Glut (Counts: Emx1_Mutant=751, Nestin_Mutant=1161) -> dge_Mutant_cond_Nestin_vs_Emx1_CA1-ProS_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - MEA Slc17a7 Glut (Counts: Emx1_Mutant=68, Nestin_Mutant=223) -> dge_Mutant_cond_Nestin_vs_Emx1_MEA_Slc17a7_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Astro-OLF NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - COAp Grxcr2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=25, Nestin_Mutant=1.\n",
      "  Generating volcano plot for: Mutant Cond - LA-BLA-BMA-PA Glut (Counts: Emx1_Mutant=186, Nestin_Mutant=102) -> dge_Mutant_cond_Nestin_vs_Emx1_LA-BLA-BMA-PA_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - DG Glut (Counts: Emx1_Mutant=402, Nestin_Mutant=922) -> dge_Mutant_cond_Nestin_vs_Emx1_DG_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - Oligo NN (Counts: Emx1_Mutant=142, Nestin_Mutant=753) -> dge_Mutant_cond_Nestin_vs_Emx1_Oligo_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - NTS Phox2b Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=1.\n",
      "  Generating volcano plot for: Mutant Cond - Sst Gaba (Counts: Emx1_Mutant=195, Nestin_Mutant=261) -> dge_Mutant_cond_Nestin_vs_Emx1_Sst_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - L2/3 IT PIR-ENTl Glut (Counts: Emx1_Mutant=169, Nestin_Mutant=156) -> dge_Mutant_cond_Nestin_vs_Emx1_L2_3_IT_PIR-ENTl_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - SUB-ProS Glut (Counts: Emx1_Mutant=173, Nestin_Mutant=218) -> dge_Mutant_cond_Nestin_vs_Emx1_SUB-ProS_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Lymphoid NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=10, Nestin_Mutant=6.\n",
      "  Skipping volcano plot for Mutant Cond - L6b CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=13, Nestin_Mutant=21.\n",
      "  Skipping volcano plot for Mutant Cond - L5 NP CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=13, Nestin_Mutant=45.\n",
      "  Generating volcano plot for: Mutant Cond - L2/3 IT CTX Glut (Counts: Emx1_Mutant=189, Nestin_Mutant=362) -> dge_Mutant_cond_Nestin_vs_Emx1_L2_3_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - L2/3 IT ENT Glut (Counts: Emx1_Mutant=382, Nestin_Mutant=460) -> dge_Mutant_cond_Nestin_vs_Emx1_L2_3_IT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - HPF CR Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=5, Nestin_Mutant=37.\n",
      "  Skipping volcano plot for Mutant Cond - Microglia NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=27, Nestin_Mutant=496.\n",
      "  Generating volcano plot for: Mutant Cond - Vip Gaba (Counts: Emx1_Mutant=67, Nestin_Mutant=147) -> dge_Mutant_cond_Nestin_vs_Emx1_Vip_Gaba_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - L6b/CT ENT Glut (Counts: Emx1_Mutant=128, Nestin_Mutant=227) -> dge_Mutant_cond_Nestin_vs_Emx1_L6b_CT_ENT_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - L2 IT ENT-po Glut (Counts: Emx1_Mutant=126, Nestin_Mutant=116) -> dge_Mutant_cond_Nestin_vs_Emx1_L2_IT_ENT-po_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - OB-in Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=11, Nestin_Mutant=11.\n",
      "  Skipping volcano plot for Mutant Cond - CA2-FC-IG Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=55, Nestin_Mutant=39.\n",
      "  Generating volcano plot for: Mutant Cond - L6 CT CTX Glut (Counts: Emx1_Mutant=161, Nestin_Mutant=494) -> dge_Mutant_cond_Nestin_vs_Emx1_L6_CT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Generating volcano plot for: Mutant Cond - RHP-COA Ndnf Gaba (Counts: Emx1_Mutant=62, Nestin_Mutant=70) -> dge_Mutant_cond_Nestin_vs_Emx1_RHP-COA_Ndnf_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - SC Lef1 Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - DG-PIR Ex IMN: Not enough cells in one or both groups (< 50). Emx1_Mutant=5, Nestin_Mutant=37.\n",
      "  Generating volcano plot for: Mutant Cond - OPC NN (Counts: Emx1_Mutant=84, Nestin_Mutant=404) -> dge_Mutant_cond_Nestin_vs_Emx1_OPC_NN_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Sncg Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=71, Nestin_Mutant=21.\n",
      "  Generating volcano plot for: Mutant Cond - L4/5 IT CTX Glut (Counts: Emx1_Mutant=108, Nestin_Mutant=248) -> dge_Mutant_cond_Nestin_vs_Emx1_L4_5_IT_CTX_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - IA Mgp Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Prdm12 Slit2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=4, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - IT AON-TT-DP Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=23, Nestin_Mutant=27.\n",
      "  Generating volcano plot for: Mutant Cond - Lamp5 Lhx6 Gaba (Counts: Emx1_Mutant=60, Nestin_Mutant=83) -> dge_Mutant_cond_Nestin_vs_Emx1_Lamp5_Lhx6_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - CEA-BST Ebf1 Pdyn Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=1.\n",
      "  Generating volcano plot for: Mutant Cond - Lamp5 Gaba (Counts: Emx1_Mutant=55, Nestin_Mutant=63) -> dge_Mutant_cond_Nestin_vs_Emx1_Lamp5_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - IT EP-CLA Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=36, Nestin_Mutant=36.\n",
      "  Generating volcano plot for: Mutant Cond - Pvalb Gaba (Counts: Emx1_Mutant=139, Nestin_Mutant=144) -> dge_Mutant_cond_Nestin_vs_Emx1_Pvalb_Gaba_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - VLMC NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=53, Nestin_Mutant=14.\n",
      "  Skipping volcano plot for Mutant Cond - L6b EPd Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=6, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - OB Meis2 Thsd7b Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=9, Nestin_Mutant=7.\n",
      "  Generating volcano plot for: Mutant Cond - L5/6 IT TPE-ENT Glut (Counts: Emx1_Mutant=101, Nestin_Mutant=82) -> dge_Mutant_cond_Nestin_vs_Emx1_L5_6_IT_TPE-ENT_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - IC Six3 En2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=4.\n",
      "  Skipping volcano plot for Mutant Cond - ENTmv-PA-COAp Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=35, Nestin_Mutant=64.\n",
      "  Skipping volcano plot for Mutant Cond - LDT Vsx2 Nkx6-1 Nfib Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - Astro-NT NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=12, Nestin_Mutant=9.\n",
      "  Skipping volcano plot for Mutant Cond - COAa-PAA-MEA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=10, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - L5 IT CTX Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=11, Nestin_Mutant=26.\n",
      "  Skipping volcano plot for Mutant Cond - SMC NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - Endo NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=41, Nestin_Mutant=8.\n",
      "  Skipping volcano plot for Mutant Cond - ABC NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=32, Nestin_Mutant=13.\n",
      "  Skipping volcano plot for Mutant Cond - CLA-EPd-CTX Car3 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=11, Nestin_Mutant=27.\n",
      "  Skipping volcano plot for Mutant Cond - SPVI-SPVC Sall3 Nfib Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Generating volcano plot for: Mutant Cond - CT SUB Glut (Counts: Emx1_Mutant=74, Nestin_Mutant=86) -> dge_Mutant_cond_Nestin_vs_Emx1_CT_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Peri NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=9, Nestin_Mutant=10.\n",
      "  Skipping volcano plot for Mutant Cond - L2 IT PPP-APr Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=24, Nestin_Mutant=80.\n",
      "  Skipping volcano plot for Mutant Cond - Pvalb chandelier Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=15, Nestin_Mutant=21.\n",
      "  Skipping volcano plot for Mutant Cond - Ependymal NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=6, Nestin_Mutant=2.\n",
      "  Generating volcano plot for: Mutant Cond - NP SUB Glut (Counts: Emx1_Mutant=55, Nestin_Mutant=86) -> dge_Mutant_cond_Nestin_vs_Emx1_NP_SUB_Glut_volcano.png\n",
      "---\n",
      "  Skipping volcano plot for Mutant Cond - Tanycyte NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - OB-mi Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - OB Trdn Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PVR Six3 Sox3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - MEA-BST Sox6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=4, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - STR D2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=13, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - CHOR NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=8.\n",
      "  Skipping volcano plot for Mutant Cond - IC Tfap2d Maf Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=4, Nestin_Mutant=6.\n",
      "  Skipping volcano plot for Mutant Cond - NI-RPO Gata3 Nr4a2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - NLL Gata3 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=6.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MH Tac2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=9.\n",
      "  Skipping volcano plot for Mutant Cond - DTN-LDT-IPN Otp Pax3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - OB Eomes Ms4a15 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - NTS-PARN Neurod2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - STR Prox1 Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - OB-STR-CTX Inh IMN: Not enough cells in one or both groups (< 50). Emx1_Mutant=13, Nestin_Mutant=15.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Nkx2-1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - STR D1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=7, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - MY Lhx1 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - TRS-BAC Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=4, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - CB Granule Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - L2/3 IT RSP Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=5, Nestin_Mutant=9.\n",
      "  Skipping volcano plot for Mutant Cond - PRT Mecom Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - OB Dopa-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MDRNd Prox1 Pax6 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MEA-BST Lhx6 Nr2e1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - CB PLI Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - BAM NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=7.\n",
      "  Skipping volcano plot for Mutant Cond - RT-ZI Gnb3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SPVC Mafa Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - TH Prkcd Grin2c Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=8, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SNc-VTA-RAmb Foxa1 Dopa: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PGRN-PARN-MDRN Hoxb5 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - VCO Mafa Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - STR-PAL Chst9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=15, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - NP PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=6, Nestin_Mutant=14.\n",
      "  Skipping volcano plot for Mutant Cond - SC-PAG Lef1 Emx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - DC NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=6.\n",
      "  Skipping volcano plot for Mutant Cond - SPVI-SPVC Tlx3 Ebf3 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - OB-out Frmd7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - CS-RPO Meis2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - AHN-SBPV-PVHd Pdrm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Prdm12 Zeb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - LDT-PCG-CS Gata3 Lhx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCs Pax7 Nfia Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=4.\n",
      "  Skipping volcano plot for Mutant Cond - ADP-MPO Trp73 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - Sst Chodl Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=7, Nestin_Mutant=7.\n",
      "  Skipping volcano plot for Mutant Cond - OEC NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - NLOT Rho Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCsg Gabrr2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=6.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-ND-PCG Onecut1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - MB-MY Tph2 Glut-Sero: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PG-TRN-LRN Fat2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Sall3 Lmo1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=9, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PRC-PAG Pax6 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCsg Pde5a Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - MEA-BST Lhx6 Nfib Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MEA-BST Lhx6 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - CEA-BST Six3 Cyp26b1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=5, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - Bergmann NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - L5 PPP Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=6, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - SCig Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=7.\n",
      "  Skipping volcano plot for Mutant Cond - SI-MA-LPO-LHA Skor1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=6, Nestin_Mutant=5.\n",
      "  Skipping volcano plot for Mutant Cond - MV-SPIV Zic4 Neurod2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - LSX Sall3 Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - NDB-SI-MA-STRv Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MG-POL-SGN Nts Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - LGv-ZI Otx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PH-SUM Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PH-LHA Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - LDT-PCG St18 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCH Six6 Cdc14a Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - L4 RSP-ACA Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=3, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - PRT Tcf7l2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=9.\n",
      "  Skipping volcano plot for Mutant Cond - PRP Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - DMH Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - STN-PSTN Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - OT D3 Folh1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-RN Nkx2-2 Otx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=2, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - MRN-PPN-CUN Pax8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCs Lef1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PDTg-PCG Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PH-ant-LHA Otp Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - ZI Pax6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=1, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - LDT-PCG Vsx2 Lhx4 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - CEA-AAA-BST Six3 Sp9 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-SC Neurod2 Meis2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - MPN-MPO-PVpo Hmx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - BST-MPN Six3 Nrgn Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-PPN Pax5 Sox21 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PRNc Otp Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=3.\n",
      "  Skipping volcano plot for Mutant Cond - MPO-ADP Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MS-SF Bsx Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - PB Sst Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SI-MA-ACB Ebf1 Bnc2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - NDB-SI-ant Prdm12 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - SI-MPO-LPO Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - Pineal Crx Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PAG Pou4f2 Mesi2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - LHA Barhl2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SBPV-PVa Six6 Satb2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - STR D1 Sema5a Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SC Bnc2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SC Tnnt1 Gli3 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PVHd-SBPV Six3 Prox1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - SCm-PAG Cdh23 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - SCop Sln Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - CS-PRNr-DR En1 Sox2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-SC Pou4f1 Zic1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - TU-ARH Otp Six6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - VMH Nr5a1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PARN-MDRNd-NTS Gbx2 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PVpo-VMPO-MPN Hmx2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - MY Prox1 Lmo7 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PVH-SO-PVa Otp Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PRC-PAG Tcf7l2 Irx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - IF-RL-CLI-PAG Foxa1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - CU-ECU-SPVI Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - STR Lhx8 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - MEA-COA-BMA Ccdc42 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - CS-PRNr-PCG Tmem163 Otp Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=0.\n",
      "  Skipping volcano plot for Mutant Cond - PVT-PT Ntrk1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PVHd-DMH Lhx6 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=4.\n",
      "  Skipping volcano plot for Mutant Cond - LDT Fgf7 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - PPN-CUN-PCG Otp En1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - PH-an Pitx2 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - SCig-an-PPT Foxb1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - PAG-MRN-RN Foxa2 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=2.\n",
      "  Skipping volcano plot for Mutant Cond - Astroependymal NN: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - MV-SPIV Phox2b Ebf3 Lbx1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - SCs Dmbx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - LH Pou4f1 Sox1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=4.\n",
      "  Skipping volcano plot for Mutant Cond - RPA Pax6 Hoxb5 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - DCO Il22 Gly-Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - DMH-LHA Gsx1 Gaba: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - RN Spp1 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "  Skipping volcano plot for Mutant Cond - APN C1ql4 Glut: Not enough cells in one or both groups (< 50). Emx1_Mutant=0, Nestin_Mutant=1.\n",
      "\n",
      "DGE visualization (Genotype comparison within conditions) finished.\n"
     ]
    }
   ],
   "source": [
    "# Genotype Comparison DGE (Nestin vs Emx1 within each condition) using the 'for_DEGs' layer, grouped by the DEG_BY\n",
    "print(f\"\\nRunning Genotype Comparison DGE (grouped by {DEG_BY})...\")\n",
    "dge_genotype_within_condition = functions_degs.run_genotype_comparison_dge(\n",
    "    adata,\n",
    "    grouping_key=DEG_BY,\n",
    "    dge_output_dir=DGE_OUTPUT_DIR,\n",
    "    plot_output_dir=PLOT_OUTPUT_DIR,\n",
    "    layer='for_DEGs' if 'for_DEGs' in adata.layers else None\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8abc4cd6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Running Cell Type Comparison DGE (Markers for mapmycells_second_layer)...\n",
      "\n",
      "Starting Cluster Comparison DGE analysis (Markers for 'mapmycells_second_layer') into D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers...\n",
      "  Initial genes: 26870\n",
      "  Filtering genes globally: Requiring expression in >= 3 cells...\n",
      "  Genes after global filtering: 26870\n",
      "  Warning: Skipping group 'LDT Vsx2 Nkx6-1 Nfib Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'NI-RPO Gata3 Nr4a2 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'RT-ZI Gnb3 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SPVC Mafa Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'SNc-VTA-RAmb Foxa1 Dopa' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'CS-RPO Meis2 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'AHN-SBPV-PVHd Pdrm12 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'NLOT Rho Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PAG-ND-PCG Onecut1 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MB-MY Tph2 Glut-Sero' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCsg Pde5a Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MEA-BST Lhx6 Sp9 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'Bergmann NN' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'LSX Sall3 Pax6 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'MG-POL-SGN Nts Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'LGv-ZI Otx2 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'LDT-PCG St18 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCH Six6 Cdc14a Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PRP Otp Gly-Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'DMH Hmx2 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'STN-PSTN Pitx2 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'OT D3 Folh1 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MRN-PPN-CUN Pax8 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCs Lef1 Gli3 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PDTg-PCG Pax6 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PH-ant-LHA Otp Bsx Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'LDT-PCG Vsx2 Lhx4 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'CEA-AAA-BST Six3 Sp9 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MPN-MPO-PVpo Hmx2 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'BST-MPN Six3 Nrgn Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MPO-ADP Lhx8 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PB Sst Gly-Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SI-MA-ACB Ebf1 Bnc2 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SI-MPO-LPO Lhx8 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PAG Pou4f2 Mesi2 Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'LHA Barhl2 Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'SBPV-PVa Six6 Satb2 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'STR D1 Sema5a Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SC Bnc2 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SC Tnnt1 Gli3 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PVHd-SBPV Six3 Prox1 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCop Sln Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'CS-PRNr-DR En1 Sox2 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'TU-ARH Otp Six6 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'VMH Nr5a1 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PARN-MDRNd-NTS Gbx2 Gly-Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PVpo-VMPO-MPN Hmx2 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'MY Prox1 Lmo7 Gly-Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PVH-SO-PVa Otp Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'IF-RL-CLI-PAG Foxa1 Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'CU-ECU-SPVI Foxb1 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'STR Lhx8 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'MEA-COA-BMA Ccdc42 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'CS-PRNr-PCG Tmem163 Otp Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PVT-PT Ntrk1 Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'LDT Fgf7 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PPN-CUN-PCG Otp En1 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'PH-an Pitx2 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCig-an-PPT Foxb1 Glut' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'PAG-MRN-RN Foxa2 Gaba' from DGE as it only contains 2 cells (less than 3).\n",
      "  Warning: Skipping group 'Astroependymal NN' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'MV-SPIV Phox2b Ebf3 Lbx1 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'SCs Dmbx1 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'RPA Pax6 Hoxb5 Gly-Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'DCO Il22 Gly-Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'DMH-LHA Gsx1 Gaba' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'RN Spp1 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Warning: Skipping group 'APN C1ql4 Glut' from DGE as it only contains 1 cells (less than 3).\n",
      "  Proceeding with DGE for 137 groups. Removed groups: ['LDT Vsx2 Nkx6-1 Nfib Glut', 'NI-RPO Gata3 Nr4a2 Gaba', 'RT-ZI Gnb3 Gaba', 'SPVC Mafa Glut', 'SNc-VTA-RAmb Foxa1 Dopa', 'CS-RPO Meis2 Gaba', 'AHN-SBPV-PVHd Pdrm12 Gaba', 'NLOT Rho Glut', 'PAG-ND-PCG Onecut1 Gaba', 'MB-MY Tph2 Glut-Sero', 'SCsg Pde5a Glut', 'MEA-BST Lhx6 Sp9 Gaba', 'Bergmann NN', 'LSX Sall3 Pax6 Gaba', 'MG-POL-SGN Nts Glut', 'LGv-ZI Otx2 Gaba', 'LDT-PCG St18 Gaba', 'SCH Six6 Cdc14a Gaba', 'PRP Otp Gly-Gaba', 'DMH Hmx2 Gaba', 'STN-PSTN Pitx2 Glut', 'OT D3 Folh1 Gaba', 'MRN-PPN-CUN Pax8 Gaba', 'SCs Lef1 Gli3 Gaba', 'PDTg-PCG Pax6 Gaba', 'PH-ant-LHA Otp Bsx Glut', 'LDT-PCG Vsx2 Lhx4 Glut', 'CEA-AAA-BST Six3 Sp9 Gaba', 'MPN-MPO-PVpo Hmx2 Glut', 'BST-MPN Six3 Nrgn Gaba', 'MPO-ADP Lhx8 Gaba', 'PB Sst Gly-Gaba', 'SI-MA-ACB Ebf1 Bnc2 Gaba', 'SI-MPO-LPO Lhx8 Gaba', 'PAG Pou4f2 Mesi2 Glut', 'LHA Barhl2 Glut', 'SBPV-PVa Six6 Satb2 Gaba', 'STR D1 Sema5a Gaba', 'SC Bnc2 Glut', 'SC Tnnt1 Gli3 Gaba', 'PVHd-SBPV Six3 Prox1 Gaba', 'SCop Sln Glut', 'CS-PRNr-DR En1 Sox2 Gaba', 'TU-ARH Otp Six6 Gaba', 'VMH Nr5a1 Glut', 'PARN-MDRNd-NTS Gbx2 Gly-Gaba', 'PVpo-VMPO-MPN Hmx2 Gaba', 'MY Prox1 Lmo7 Gly-Gaba', 'PVH-SO-PVa Otp Glut', 'IF-RL-CLI-PAG Foxa1 Glut', 'CU-ECU-SPVI Foxb1 Glut', 'STR Lhx8 Gaba', 'MEA-COA-BMA Ccdc42 Glut', 'CS-PRNr-PCG Tmem163 Otp Gaba', 'PVT-PT Ntrk1 Glut', 'LDT Fgf7 Gaba', 'PPN-CUN-PCG Otp En1 Gaba', 'PH-an Pitx2 Glut', 'SCig-an-PPT Foxb1 Glut', 'PAG-MRN-RN Foxa2 Gaba', 'Astroependymal NN', 'MV-SPIV Phox2b Ebf3 Lbx1 Glut', 'SCs Dmbx1 Gaba', 'RPA Pax6 Hoxb5 Gly-Gaba', 'DCO Il22 Gly-Gaba', 'DMH-LHA Gsx1 Gaba', 'RN Spp1 Glut', 'APN C1ql4 Glut']\n",
      "  Running sc.tl.rank_genes_groups (groupby='mapmycells_second_layer', method='wilcoxon')...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:456: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"names\"] = self.var_names[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:458: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"scores\"] = scores[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:461: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals\"] = pvals[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:471: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"pvals_adj\"] = pvals_adj[global_indices]\n",
      "d:\\Programs\\miniforge3\\envs\\bioinf\\Lib\\site-packages\\scanpy\\tools\\_rank_genes_groups.py:482: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  self.stats[group_name, \"logfoldchanges\"] = np.log2(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  rank_genes_groups finished.\n",
      "    Calculating mean expression for Cluster ABC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster ABC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ABC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ABC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ABC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ABC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ABC_NN' is writable.\n",
      "    Saved 591 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ABC_NN\\Cluster_ABC NN_vs_Rest_up_significant.csv\n",
      "    Saved 3109 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ABC_NN\\Cluster_ABC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster ADP-MPO Trp73 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster ADP-MPO Trp73 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ADP-MPO_Trp73_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ADP-MPO_Trp73_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ADP-MPO_Trp73_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ADP-MPO_Trp73_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ADP-MPO_Trp73_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_ADP-MPO Trp73 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ADP-MPO_Trp73_Glut\\Cluster_ADP-MPO Trp73 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_ADP-MPO Trp73 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ADP-MPO_Trp73_Glut\\Cluster_ADP-MPO Trp73 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Astro-NT NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Astro-NT NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-NT_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-NT_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-NT_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-NT_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-NT_NN' is writable.\n",
      "    Saved 180 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-NT_NN\\Cluster_Astro-NT NN_vs_Rest_up_significant.csv\n",
      "    Saved 1964 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-NT_NN\\Cluster_Astro-NT NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Astro-OLF NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Astro-OLF NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-OLF_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-OLF_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-OLF_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-OLF_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-OLF_NN' is writable.\n",
      "    No significant UP genes found for Cluster_Astro-OLF NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-OLF_NN\\Cluster_Astro-OLF NN_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_Astro-OLF NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-OLF_NN\\Cluster_Astro-OLF NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Astro-TE NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Astro-TE NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-TE_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Astro-TE_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-TE_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-TE_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-TE_NN' is writable.\n",
      "    Saved 1519 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-TE_NN\\Cluster_Astro-TE NN_vs_Rest_up_significant.csv\n",
      "    Saved 6245 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Astro-TE_NN\\Cluster_Astro-TE NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster BAM NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster BAM NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_BAM_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_BAM_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\BAM_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\BAM_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\BAM_NN' is writable.\n",
      "    Saved 165 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\BAM_NN\\Cluster_BAM NN_vs_Rest_up_significant.csv\n",
      "    Saved 1559 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\BAM_NN\\Cluster_BAM NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CA1-ProS Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CA1-ProS Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA1-ProS_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA1-ProS_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA1-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA1-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA1-ProS_Glut' is writable.\n",
      "    Saved 2840 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA1-ProS_Glut\\Cluster_CA1-ProS Glut_vs_Rest_up_significant.csv\n",
      "    Saved 926 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA1-ProS_Glut\\Cluster_CA1-ProS Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CA2-FC-IG Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CA2-FC-IG Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA2-FC-IG_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA2-FC-IG_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA2-FC-IG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA2-FC-IG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA2-FC-IG_Glut' is writable.\n",
      "    Saved 752 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA2-FC-IG_Glut\\Cluster_CA2-FC-IG Glut_vs_Rest_up_significant.csv\n",
      "    Saved 1078 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA2-FC-IG_Glut\\Cluster_CA2-FC-IG Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CA3 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CA3 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA3_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CA3_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA3_Glut' is writable.\n",
      "    Saved 3451 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA3_Glut\\Cluster_CA3 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 951 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CA3_Glut\\Cluster_CA3 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CB Granule Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CB Granule Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CB_Granule_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CB_Granule_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_Granule_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_Granule_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_Granule_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_CB Granule Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_Granule_Glut\\Cluster_CB Granule Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_CB Granule Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_Granule_Glut\\Cluster_CB Granule Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CB PLI Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CB PLI Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CB_PLI_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CB_PLI_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_PLI_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_PLI_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_PLI_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_CB PLI Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_PLI_Gly-Gaba\\Cluster_CB PLI Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_CB PLI Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CB_PLI_Gly-Gaba\\Cluster_CB PLI Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CEA-BST Ebf1 Pdyn Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CEA-BST Ebf1 Pdyn Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CEA-BST_Ebf1_Pdyn_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CEA-BST_Ebf1_Pdyn_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Ebf1_Pdyn_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Ebf1_Pdyn_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Ebf1_Pdyn_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_CEA-BST Ebf1 Pdyn Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Ebf1_Pdyn_Gaba\\Cluster_CEA-BST Ebf1 Pdyn Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_CEA-BST Ebf1 Pdyn Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Ebf1_Pdyn_Gaba\\Cluster_CEA-BST Ebf1 Pdyn Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CEA-BST Six3 Cyp26b1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CEA-BST Six3 Cyp26b1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CEA-BST_Six3_Cyp26b1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CEA-BST_Six3_Cyp26b1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Six3_Cyp26b1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Six3_Cyp26b1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Six3_Cyp26b1_Gaba' is writable.\n",
      "    Saved 46 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Six3_Cyp26b1_Gaba\\Cluster_CEA-BST Six3 Cyp26b1 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 8 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CEA-BST_Six3_Cyp26b1_Gaba\\Cluster_CEA-BST Six3 Cyp26b1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CHOR NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CHOR NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CHOR_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CHOR_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CHOR_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CHOR_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CHOR_NN' is writable.\n",
      "    Saved 51 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CHOR_NN\\Cluster_CHOR NN_vs_Rest_up_significant.csv\n",
      "    Saved 67 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CHOR_NN\\Cluster_CHOR NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CLA-EPd-CTX Car3 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CLA-EPd-CTX Car3 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CLA-EPd-CTX_Car3_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CLA-EPd-CTX_Car3_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CLA-EPd-CTX_Car3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CLA-EPd-CTX_Car3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CLA-EPd-CTX_Car3_Glut' is writable.\n",
      "    Saved 1256 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CLA-EPd-CTX_Car3_Glut\\Cluster_CLA-EPd-CTX Car3 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 88 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CLA-EPd-CTX_Car3_Glut\\Cluster_CLA-EPd-CTX Car3 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster COAa-PAA-MEA Barhl2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster COAa-PAA-MEA Barhl2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_COAa-PAA-MEA_Barhl2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_COAa-PAA-MEA_Barhl2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAa-PAA-MEA_Barhl2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAa-PAA-MEA_Barhl2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAa-PAA-MEA_Barhl2_Glut' is writable.\n",
      "    Saved 14 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAa-PAA-MEA_Barhl2_Glut\\Cluster_COAa-PAA-MEA Barhl2 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 278 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAa-PAA-MEA_Barhl2_Glut\\Cluster_COAa-PAA-MEA Barhl2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster COAp Grxcr2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster COAp Grxcr2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_COAp_Grxcr2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_COAp_Grxcr2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAp_Grxcr2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAp_Grxcr2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAp_Grxcr2_Glut' is writable.\n",
      "    Saved 43 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAp_Grxcr2_Glut\\Cluster_COAp Grxcr2 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 368 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\COAp_Grxcr2_Glut\\Cluster_COAp Grxcr2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster CT SUB Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster CT SUB Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CT_SUB_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_CT_SUB_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CT_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CT_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CT_SUB_Glut' is writable.\n",
      "    Saved 1049 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CT_SUB_Glut\\Cluster_CT SUB Glut_vs_Rest_up_significant.csv\n",
      "    Saved 479 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\CT_SUB_Glut\\Cluster_CT SUB Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster DC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster DC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DC_NN' is writable.\n",
      "    Saved 7 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DC_NN\\Cluster_DC NN_vs_Rest_up_significant.csv\n",
      "    Saved 535 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DC_NN\\Cluster_DC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster DG Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster DG Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DG_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DG_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG_Glut' is writable.\n",
      "    Saved 3021 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG_Glut\\Cluster_DG Glut_vs_Rest_up_significant.csv\n",
      "    Saved 2032 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG_Glut\\Cluster_DG Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster DG-PIR Ex IMN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster DG-PIR Ex IMN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DG-PIR_Ex_IMN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DG-PIR_Ex_IMN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG-PIR_Ex_IMN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG-PIR_Ex_IMN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG-PIR_Ex_IMN' is writable.\n",
      "    Saved 271 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG-PIR_Ex_IMN\\Cluster_DG-PIR Ex IMN_vs_Rest_up_significant.csv\n",
      "    Saved 3643 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DG-PIR_Ex_IMN\\Cluster_DG-PIR Ex IMN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster DTN-LDT-IPN Otp Pax3 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster DTN-LDT-IPN Otp Pax3 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DTN-LDT-IPN_Otp_Pax3_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_DTN-LDT-IPN_Otp_Pax3_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DTN-LDT-IPN_Otp_Pax3_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DTN-LDT-IPN_Otp_Pax3_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DTN-LDT-IPN_Otp_Pax3_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_DTN-LDT-IPN Otp Pax3 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DTN-LDT-IPN_Otp_Pax3_Gaba\\Cluster_DTN-LDT-IPN Otp Pax3 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_DTN-LDT-IPN Otp Pax3 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\DTN-LDT-IPN_Otp_Pax3_Gaba\\Cluster_DTN-LDT-IPN Otp Pax3 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster ENTmv-PA-COAp Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster ENTmv-PA-COAp Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ENTmv-PA-COAp_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ENTmv-PA-COAp_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ENTmv-PA-COAp_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ENTmv-PA-COAp_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ENTmv-PA-COAp_Glut' is writable.\n",
      "    Saved 2161 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ENTmv-PA-COAp_Glut\\Cluster_ENTmv-PA-COAp Glut_vs_Rest_up_significant.csv\n",
      "    Saved 103 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ENTmv-PA-COAp_Glut\\Cluster_ENTmv-PA-COAp Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Endo NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Endo NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Endo_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Endo_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Endo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Endo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Endo_NN' is writable.\n",
      "    Saved 820 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Endo_NN\\Cluster_Endo NN_vs_Rest_up_significant.csv\n",
      "    Saved 3992 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Endo_NN\\Cluster_Endo NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Ependymal NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Ependymal NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Ependymal_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Ependymal_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Ependymal_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Ependymal_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Ependymal_NN' is writable.\n",
      "    Saved 572 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Ependymal_NN\\Cluster_Ependymal NN_vs_Rest_up_significant.csv\n",
      "    Saved 473 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Ependymal_NN\\Cluster_Ependymal NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster HPF CR Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster HPF CR Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_HPF_CR_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_HPF_CR_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\HPF_CR_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\HPF_CR_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\HPF_CR_Glut' is writable.\n",
      "    Saved 163 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\HPF_CR_Glut\\Cluster_HPF CR Glut_vs_Rest_up_significant.csv\n",
      "    Saved 3156 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\HPF_CR_Glut\\Cluster_HPF CR Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster IA Mgp Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster IA Mgp Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IA_Mgp_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IA_Mgp_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IA_Mgp_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IA_Mgp_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IA_Mgp_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_IA Mgp Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IA_Mgp_Gaba\\Cluster_IA Mgp Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_IA Mgp Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IA_Mgp_Gaba\\Cluster_IA Mgp Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster IC Six3 En2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster IC Six3 En2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IC_Six3_En2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IC_Six3_En2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Six3_En2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Six3_En2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Six3_En2_Gaba' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Six3_En2_Gaba\\Cluster_IC Six3 En2 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 1456 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Six3_En2_Gaba\\Cluster_IC Six3 En2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster IC Tfap2d Maf Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster IC Tfap2d Maf Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IC_Tfap2d_Maf_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IC_Tfap2d_Maf_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Tfap2d_Maf_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Tfap2d_Maf_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Tfap2d_Maf_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_IC Tfap2d Maf Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Tfap2d_Maf_Glut\\Cluster_IC Tfap2d Maf Glut_vs_Rest_up_significant.csv\n",
      "    Saved 76 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IC_Tfap2d_Maf_Glut\\Cluster_IC Tfap2d Maf Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster IT AON-TT-DP Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster IT AON-TT-DP Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IT_AON-TT-DP_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IT_AON-TT-DP_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_AON-TT-DP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_AON-TT-DP_Glut' is writable.\n",
      "    Saved 87 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_AON-TT-DP_Glut\\Cluster_IT AON-TT-DP Glut_vs_Rest_up_significant.csv\n",
      "    Saved 5455 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_AON-TT-DP_Glut\\Cluster_IT AON-TT-DP Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster IT EP-CLA Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster IT EP-CLA Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IT_EP-CLA_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_IT_EP-CLA_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_EP-CLA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_EP-CLA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_EP-CLA_Glut' is writable.\n",
      "    Saved 1482 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_EP-CLA_Glut\\Cluster_IT EP-CLA Glut_vs_Rest_up_significant.csv\n",
      "    Saved 107 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\IT_EP-CLA_Glut\\Cluster_IT EP-CLA Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L2 IT ENT-po Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2 IT ENT-po Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_IT_ENT-po_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_IT_ENT-po_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_ENT-po_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_ENT-po_Glut' is writable.\n",
      "    Saved 3959 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_ENT-po_Glut\\Cluster_L2 IT ENT-po Glut_vs_Rest_up_significant.csv\n",
      "    Saved 341 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_ENT-po_Glut\\Cluster_L2 IT ENT-po Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L2 IT PPP-APr Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2 IT PPP-APr Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_IT_PPP-APr_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_IT_PPP-APr_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_PPP-APr_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_PPP-APr_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_PPP-APr_Glut' is writable.\n",
      "    Saved 423 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_PPP-APr_Glut\\Cluster_L2 IT PPP-APr Glut_vs_Rest_up_significant.csv\n",
      "    Saved 324 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_IT_PPP-APr_Glut\\Cluster_L2 IT PPP-APr Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L2/3 IT CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2/3 IT CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_CTX_Glut' is writable.\n",
      "    Skipping cluster L2/3 IT CTX Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_CTX_Glut\\Cluster_L2'\n",
      "    Calculating mean expression for Cluster L2/3 IT ENT Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2/3 IT ENT Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_ENT_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_ENT_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_ENT_Glut' is writable.\n",
      "    Skipping cluster L2/3 IT ENT Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_ENT_Glut\\Cluster_L2'\n",
      "    Calculating mean expression for Cluster L2/3 IT PIR-ENTl Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2/3 IT PIR-ENTl Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_PIR-ENTl_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_PIR-ENTl_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut' is writable.\n",
      "    Skipping cluster L2/3 IT PIR-ENTl Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PIR-ENTl_Glut\\Cluster_L2'\n",
      "    Calculating mean expression for Cluster L2/3 IT PPP Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2/3 IT PPP Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_PPP_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_PPP_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PPP_Glut' is writable.\n",
      "    Skipping cluster L2/3 IT PPP Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_PPP_Glut\\Cluster_L2'\n",
      "    Calculating mean expression for Cluster L2/3 IT RSP Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L2/3 IT RSP Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_RSP_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L2_3_IT_RSP_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_RSP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_RSP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_RSP_Glut' is writable.\n",
      "    Skipping cluster L2/3 IT RSP Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L2_3_IT_RSP_Glut\\Cluster_L2'\n",
      "    Calculating mean expression for Cluster L4 RSP-ACA Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L4 RSP-ACA Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L4_RSP-ACA_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L4_RSP-ACA_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_RSP-ACA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_RSP-ACA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_RSP-ACA_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_L4 RSP-ACA Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_RSP-ACA_Glut\\Cluster_L4 RSP-ACA Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_L4 RSP-ACA Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_RSP-ACA_Glut\\Cluster_L4 RSP-ACA Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L4/5 IT CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L4/5 IT CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L4_5_IT_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L4_5_IT_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_5_IT_CTX_Glut' is writable.\n",
      "    Skipping cluster L4/5 IT CTX Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L4_5_IT_CTX_Glut\\Cluster_L4'\n",
      "    Calculating mean expression for Cluster L5 ET CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L5 ET CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_ET_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_ET_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_ET_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_ET_CTX_Glut' is writable.\n",
      "    Saved 1791 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_ET_CTX_Glut\\Cluster_L5 ET CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 107 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_ET_CTX_Glut\\Cluster_L5 ET CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L5 IT CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L5 IT CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_IT_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_IT_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_IT_CTX_Glut' is writable.\n",
      "    Saved 548 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_IT_CTX_Glut\\Cluster_L5 IT CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 20 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_IT_CTX_Glut\\Cluster_L5 IT CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L5 NP CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L5 NP CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_NP_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_NP_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_NP_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_NP_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_NP_CTX_Glut' is writable.\n",
      "    Saved 160 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_NP_CTX_Glut\\Cluster_L5 NP CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 113 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_NP_CTX_Glut\\Cluster_L5 NP CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L5 PPP Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L5 PPP Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_PPP_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_PPP_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_PPP_Glut' is writable.\n",
      "    Saved 130 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_PPP_Glut\\Cluster_L5 PPP Glut_vs_Rest_up_significant.csv\n",
      "    Saved 9 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_PPP_Glut\\Cluster_L5 PPP Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L5/6 IT TPE-ENT Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L5/6 IT TPE-ENT Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_6_IT_TPE-ENT_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L5_6_IT_TPE-ENT_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut' is writable.\n",
      "    Skipping cluster L5/6 IT TPE-ENT Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L5_6_IT_TPE-ENT_Glut\\Cluster_L5'\n",
      "    Calculating mean expression for Cluster L6 CT CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L6 CT CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6_CT_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6_CT_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_CT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_CT_CTX_Glut' is writable.\n",
      "    Saved 6350 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_CT_CTX_Glut\\Cluster_L6 CT CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 862 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_CT_CTX_Glut\\Cluster_L6 CT CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L6 IT CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L6 IT CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6_IT_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6_IT_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_IT_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_IT_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_IT_CTX_Glut' is writable.\n",
      "    Saved 99 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_IT_CTX_Glut\\Cluster_L6 IT CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 1323 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6_IT_CTX_Glut\\Cluster_L6 IT CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L6b CTX Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L6b CTX Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_CTX_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_CTX_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CTX_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CTX_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CTX_Glut' is writable.\n",
      "    Saved 596 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CTX_Glut\\Cluster_L6b CTX Glut_vs_Rest_up_significant.csv\n",
      "    Saved 172 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CTX_Glut\\Cluster_L6b CTX Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L6b EPd Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L6b EPd Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_EPd_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_EPd_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_EPd_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_EPd_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_EPd_Glut' is writable.\n",
      "    Saved 172 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_EPd_Glut\\Cluster_L6b EPd Glut_vs_Rest_up_significant.csv\n",
      "    Saved 45 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_EPd_Glut\\Cluster_L6b EPd Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster L6b/CT ENT Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster L6b/CT ENT Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_CT_ENT_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_L6b_CT_ENT_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CT_ENT_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CT_ENT_Glut' is writable.\n",
      "    Skipping cluster L6b/CT ENT Glut: Error during processing - Cannot save file into a non-existent directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\L6b_CT_ENT_Glut\\Cluster_L6b'\n",
      "    Calculating mean expression for Cluster LA-BLA-BMA-PA Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LA-BLA-BMA-PA Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LA-BLA-BMA-PA_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LA-BLA-BMA-PA_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LA-BLA-BMA-PA_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LA-BLA-BMA-PA_Glut' is writable.\n",
      "    Saved 178 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LA-BLA-BMA-PA_Glut\\Cluster_LA-BLA-BMA-PA Glut_vs_Rest_up_significant.csv\n",
      "    Saved 4041 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LA-BLA-BMA-PA_Glut\\Cluster_LA-BLA-BMA-PA Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LDT-PCG-CS Gata3 Lhx1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LDT-PCG-CS Gata3 Lhx1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LDT-PCG-CS_Gata3_Lhx1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LDT-PCG-CS_Gata3_Lhx1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LDT-PCG-CS_Gata3_Lhx1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LDT-PCG-CS_Gata3_Lhx1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LDT-PCG-CS_Gata3_Lhx1_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LDT-PCG-CS Gata3 Lhx1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LDT-PCG-CS_Gata3_Lhx1_Gaba\\Cluster_LDT-PCG-CS Gata3 Lhx1 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_LDT-PCG-CS Gata3 Lhx1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LDT-PCG-CS_Gata3_Lhx1_Gaba\\Cluster_LDT-PCG-CS Gata3 Lhx1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba\\Cluster_LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 223 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LGv-SPFp-SPFm_Nkx2-2_Tcf7l2_Gaba\\Cluster_LGv-SPFp-SPFm Nkx2-2 Tcf7l2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LH Pou4f1 Sox1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LH Pou4f1 Sox1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LH_Pou4f1_Sox1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LH_Pou4f1_Sox1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LH_Pou4f1_Sox1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LH_Pou4f1_Sox1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LH_Pou4f1_Sox1_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_LH Pou4f1 Sox1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LH_Pou4f1_Sox1_Glut\\Cluster_LH Pou4f1 Sox1 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_LH Pou4f1 Sox1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LH_Pou4f1_Sox1_Glut\\Cluster_LH Pou4f1 Sox1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LSX Nkx2-1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LSX Nkx2-1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Nkx2-1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Nkx2-1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Nkx2-1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Nkx2-1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Nkx2-1_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LSX Nkx2-1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Nkx2-1_Gaba\\Cluster_LSX Nkx2-1 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 26 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Nkx2-1_Gaba\\Cluster_LSX Nkx2-1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LSX Otx2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LSX Otx2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Otx2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Otx2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Otx2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Otx2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Otx2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LSX Otx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Otx2_Gaba\\Cluster_LSX Otx2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_LSX Otx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Otx2_Gaba\\Cluster_LSX Otx2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LSX Prdm12 Slit2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LSX Prdm12 Slit2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Prdm12_Slit2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Prdm12_Slit2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Slit2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Slit2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Slit2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LSX Prdm12 Slit2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Slit2_Gaba\\Cluster_LSX Prdm12 Slit2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_LSX Prdm12 Slit2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Slit2_Gaba\\Cluster_LSX Prdm12 Slit2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LSX Prdm12 Zeb2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LSX Prdm12 Zeb2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Prdm12_Zeb2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Prdm12_Zeb2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Zeb2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Zeb2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Zeb2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_LSX Prdm12 Zeb2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Zeb2_Gaba\\Cluster_LSX Prdm12 Zeb2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_LSX Prdm12 Zeb2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Prdm12_Zeb2_Gaba\\Cluster_LSX Prdm12 Zeb2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster LSX Sall3 Lmo1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster LSX Sall3 Lmo1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Sall3_Lmo1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_LSX_Sall3_Lmo1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Sall3_Lmo1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Sall3_Lmo1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Sall3_Lmo1_Gaba' is writable.\n",
      "    Saved 119 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Sall3_Lmo1_Gaba\\Cluster_LSX Sall3 Lmo1 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 14 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\LSX_Sall3_Lmo1_Gaba\\Cluster_LSX Sall3 Lmo1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Lamp5 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Lamp5 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lamp5_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lamp5_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Gaba' is writable.\n",
      "    Saved 798 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Gaba\\Cluster_Lamp5 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 605 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Gaba\\Cluster_Lamp5 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Lamp5 Lhx6 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Lamp5 Lhx6 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lamp5_Lhx6_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lamp5_Lhx6_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Lhx6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Lhx6_Gaba' is writable.\n",
      "    Saved 1558 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Lhx6_Gaba\\Cluster_Lamp5 Lhx6 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 540 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lamp5_Lhx6_Gaba\\Cluster_Lamp5 Lhx6 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Lymphoid NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Lymphoid NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lymphoid_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Lymphoid_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lymphoid_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lymphoid_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lymphoid_NN' is writable.\n",
      "    Saved 60 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lymphoid_NN\\Cluster_Lymphoid NN_vs_Rest_up_significant.csv\n",
      "    Saved 3622 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Lymphoid_NN\\Cluster_Lymphoid NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MDRNd Prox1 Pax6 Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MDRNd Prox1 Pax6 Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MDRNd_Prox1_Pax6_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MDRNd_Prox1_Pax6_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MDRNd_Prox1_Pax6_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MDRNd_Prox1_Pax6_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MDRNd_Prox1_Pax6_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_MDRNd Prox1 Pax6 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MDRNd_Prox1_Pax6_Gly-Gaba\\Cluster_MDRNd Prox1 Pax6 Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MDRNd Prox1 Pax6 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MDRNd_Prox1_Pax6_Gly-Gaba\\Cluster_MDRNd Prox1 Pax6 Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MEA Slc17a7 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MEA Slc17a7 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA_Slc17a7_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA_Slc17a7_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA_Slc17a7_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA_Slc17a7_Glut' is writable.\n",
      "    Saved 68 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA_Slc17a7_Glut\\Cluster_MEA Slc17a7 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 4464 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA_Slc17a7_Glut\\Cluster_MEA Slc17a7 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MEA-BST Lhx6 Nfib Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MEA-BST Lhx6 Nfib Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Lhx6_Nfib_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Lhx6_Nfib_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nfib_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nfib_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nfib_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_MEA-BST Lhx6 Nfib Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nfib_Gaba\\Cluster_MEA-BST Lhx6 Nfib Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MEA-BST Lhx6 Nfib Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nfib_Gaba\\Cluster_MEA-BST Lhx6 Nfib Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MEA-BST Lhx6 Nr2e1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MEA-BST Lhx6 Nr2e1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Lhx6_Nr2e1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Lhx6_Nr2e1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nr2e1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nr2e1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nr2e1_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_MEA-BST Lhx6 Nr2e1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nr2e1_Gaba\\Cluster_MEA-BST Lhx6 Nr2e1 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MEA-BST Lhx6 Nr2e1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Lhx6_Nr2e1_Gaba\\Cluster_MEA-BST Lhx6 Nr2e1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MEA-BST Sox6 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MEA-BST Sox6 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Sox6_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MEA-BST_Sox6_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Sox6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Sox6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Sox6_Gaba' is writable.\n",
      "    Saved 3 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Sox6_Gaba\\Cluster_MEA-BST Sox6 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 277 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MEA-BST_Sox6_Gaba\\Cluster_MEA-BST Sox6 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MH Tac2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MH Tac2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MH_Tac2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MH_Tac2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MH_Tac2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MH_Tac2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MH_Tac2_Glut' is writable.\n",
      "    Saved 58 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MH_Tac2_Glut\\Cluster_MH Tac2 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 5 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MH_Tac2_Glut\\Cluster_MH Tac2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MS-SF Bsx Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MS-SF Bsx Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MS-SF_Bsx_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MS-SF_Bsx_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MS-SF_Bsx_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MS-SF_Bsx_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MS-SF_Bsx_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_MS-SF Bsx Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MS-SF_Bsx_Glut\\Cluster_MS-SF Bsx Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MS-SF Bsx Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MS-SF_Bsx_Glut\\Cluster_MS-SF Bsx Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MV-SPIV Zic4 Neurod2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MV-SPIV Zic4 Neurod2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MV-SPIV_Zic4_Neurod2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MV-SPIV_Zic4_Neurod2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MV-SPIV_Zic4_Neurod2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MV-SPIV_Zic4_Neurod2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MV-SPIV_Zic4_Neurod2_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_MV-SPIV Zic4 Neurod2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MV-SPIV_Zic4_Neurod2_Glut\\Cluster_MV-SPIV Zic4 Neurod2 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MV-SPIV Zic4 Neurod2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MV-SPIV_Zic4_Neurod2_Glut\\Cluster_MV-SPIV Zic4 Neurod2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster MY Lhx1 Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster MY Lhx1 Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MY_Lhx1_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_MY_Lhx1_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MY_Lhx1_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MY_Lhx1_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MY_Lhx1_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_MY Lhx1 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MY_Lhx1_Gly-Gaba\\Cluster_MY Lhx1 Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_MY Lhx1 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\MY_Lhx1_Gly-Gaba\\Cluster_MY Lhx1 Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Microglia NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Microglia NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Microglia_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Microglia_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Microglia_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Microglia_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Microglia_NN' is writable.\n",
      "    Saved 1379 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Microglia_NN\\Cluster_Microglia NN_vs_Rest_up_significant.csv\n",
      "    Saved 8503 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Microglia_NN\\Cluster_Microglia NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Monocytes NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Monocytes NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Monocytes_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Monocytes_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Monocytes_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Monocytes_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Monocytes_NN' is writable.\n",
      "    No significant UP genes found for Cluster_Monocytes NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Monocytes_NN\\Cluster_Monocytes NN_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_Monocytes NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Monocytes_NN\\Cluster_Monocytes NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NDB-SI-MA-STRv Lhx8 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NDB-SI-MA-STRv Lhx8 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NDB-SI-MA-STRv_Lhx8_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NDB-SI-MA-STRv_Lhx8_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-MA-STRv_Lhx8_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-MA-STRv_Lhx8_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-MA-STRv_Lhx8_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_NDB-SI-MA-STRv Lhx8 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-MA-STRv_Lhx8_Gaba\\Cluster_NDB-SI-MA-STRv Lhx8 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_NDB-SI-MA-STRv Lhx8 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-MA-STRv_Lhx8_Gaba\\Cluster_NDB-SI-MA-STRv Lhx8 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NDB-SI-ant Prdm12 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NDB-SI-ant Prdm12 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NDB-SI-ant_Prdm12_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NDB-SI-ant_Prdm12_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-ant_Prdm12_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-ant_Prdm12_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-ant_Prdm12_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_NDB-SI-ant Prdm12 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-ant_Prdm12_Gaba\\Cluster_NDB-SI-ant Prdm12 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_NDB-SI-ant Prdm12 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NDB-SI-ant_Prdm12_Gaba\\Cluster_NDB-SI-ant Prdm12 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NLL Gata3 Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NLL Gata3 Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NLL_Gata3_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NLL_Gata3_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NLL_Gata3_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NLL_Gata3_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NLL_Gata3_Gly-Gaba' is writable.\n",
      "    Saved 6 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NLL_Gata3_Gly-Gaba\\Cluster_NLL Gata3 Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 1561 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NLL_Gata3_Gly-Gaba\\Cluster_NLL Gata3 Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NP PPP Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NP PPP Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NP_PPP_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NP_PPP_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_PPP_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_PPP_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_PPP_Glut' is writable.\n",
      "    Saved 280 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_PPP_Glut\\Cluster_NP PPP Glut_vs_Rest_up_significant.csv\n",
      "    Saved 70 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_PPP_Glut\\Cluster_NP PPP Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NP SUB Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NP SUB Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NP_SUB_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NP_SUB_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_SUB_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_SUB_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_SUB_Glut' is writable.\n",
      "    Saved 2335 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_SUB_Glut\\Cluster_NP SUB Glut_vs_Rest_up_significant.csv\n",
      "    Saved 405 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NP_SUB_Glut\\Cluster_NP SUB Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NTS Phox2b Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NTS Phox2b Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NTS_Phox2b_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NTS_Phox2b_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS_Phox2b_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS_Phox2b_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS_Phox2b_Glut' is writable.\n",
      "    Saved 7 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS_Phox2b_Glut\\Cluster_NTS Phox2b Glut_vs_Rest_up_significant.csv\n",
      "    Saved 1862 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS_Phox2b_Glut\\Cluster_NTS Phox2b Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster NTS-PARN Neurod2 Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster NTS-PARN Neurod2 Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NTS-PARN_Neurod2_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_NTS-PARN_Neurod2_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS-PARN_Neurod2_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS-PARN_Neurod2_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS-PARN_Neurod2_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_NTS-PARN Neurod2 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS-PARN_Neurod2_Gly-Gaba\\Cluster_NTS-PARN Neurod2 Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_NTS-PARN Neurod2 Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\NTS-PARN_Neurod2_Gly-Gaba\\Cluster_NTS-PARN Neurod2 Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB Dopa-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB Dopa-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Dopa-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Dopa-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Dopa-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Dopa-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Dopa-Gaba' is writable.\n",
      "    Saved 5 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Dopa-Gaba\\Cluster_OB Dopa-Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 118 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Dopa-Gaba\\Cluster_OB Dopa-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB Eomes Ms4a15 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB Eomes Ms4a15 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Eomes_Ms4a15_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Eomes_Ms4a15_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Eomes_Ms4a15_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Eomes_Ms4a15_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Eomes_Ms4a15_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_OB Eomes Ms4a15 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Eomes_Ms4a15_Glut\\Cluster_OB Eomes Ms4a15 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 52 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Eomes_Ms4a15_Glut\\Cluster_OB Eomes Ms4a15 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB Meis2 Thsd7b Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB Meis2 Thsd7b Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Meis2_Thsd7b_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Meis2_Thsd7b_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Meis2_Thsd7b_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Meis2_Thsd7b_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Meis2_Thsd7b_Gaba' is writable.\n",
      "    Saved 10 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Meis2_Thsd7b_Gaba\\Cluster_OB Meis2 Thsd7b Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 3390 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Meis2_Thsd7b_Gaba\\Cluster_OB Meis2 Thsd7b Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB Trdn Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB Trdn Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Trdn_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB_Trdn_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Trdn_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Trdn_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Trdn_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_OB Trdn Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Trdn_Gaba\\Cluster_OB Trdn Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_OB Trdn Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB_Trdn_Gaba\\Cluster_OB Trdn Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB-STR-CTX Inh IMN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB-STR-CTX Inh IMN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-STR-CTX_Inh_IMN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-STR-CTX_Inh_IMN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-STR-CTX_Inh_IMN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-STR-CTX_Inh_IMN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-STR-CTX_Inh_IMN' is writable.\n",
      "    Saved 31 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-STR-CTX_Inh_IMN\\Cluster_OB-STR-CTX Inh IMN_vs_Rest_up_significant.csv\n",
      "    Saved 2916 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-STR-CTX_Inh_IMN\\Cluster_OB-STR-CTX Inh IMN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB-in Frmd7 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB-in Frmd7 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-in_Frmd7_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-in_Frmd7_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-in_Frmd7_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-in_Frmd7_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-in_Frmd7_Gaba' is writable.\n",
      "    Saved 65 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-in_Frmd7_Gaba\\Cluster_OB-in Frmd7 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 3181 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-in_Frmd7_Gaba\\Cluster_OB-in Frmd7 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB-mi Frmd7 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB-mi Frmd7 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-mi_Frmd7_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-mi_Frmd7_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-mi_Frmd7_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-mi_Frmd7_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-mi_Frmd7_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_OB-mi Frmd7 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-mi_Frmd7_Gaba\\Cluster_OB-mi Frmd7 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_OB-mi Frmd7 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-mi_Frmd7_Gaba\\Cluster_OB-mi Frmd7 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OB-out Frmd7 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OB-out Frmd7 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-out_Frmd7_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OB-out_Frmd7_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-out_Frmd7_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-out_Frmd7_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-out_Frmd7_Gaba' is writable.\n",
      "    Saved 172 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-out_Frmd7_Gaba\\Cluster_OB-out Frmd7 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 1035 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OB-out_Frmd7_Gaba\\Cluster_OB-out Frmd7 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OEC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OEC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OEC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OEC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OEC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OEC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OEC_NN' is writable.\n",
      "    Saved 390 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OEC_NN\\Cluster_OEC NN_vs_Rest_up_significant.csv\n",
      "    Saved 1192 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OEC_NN\\Cluster_OEC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster OPC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster OPC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OPC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_OPC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OPC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OPC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OPC_NN' is writable.\n",
      "    Saved 955 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OPC_NN\\Cluster_OPC NN_vs_Rest_up_significant.csv\n",
      "    Saved 6661 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\OPC_NN\\Cluster_OPC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Oligo NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Oligo NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Oligo_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Oligo_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Oligo_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Oligo_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Oligo_NN' is writable.\n",
      "    Saved 1317 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Oligo_NN\\Cluster_Oligo NN_vs_Rest_up_significant.csv\n",
      "    Saved 8534 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Oligo_NN\\Cluster_Oligo NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PAG-PPN Pax5 Sox21 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PAG-PPN Pax5 Sox21 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-PPN_Pax5_Sox21_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-PPN_Pax5_Sox21_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-PPN_Pax5_Sox21_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-PPN_Pax5_Sox21_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-PPN_Pax5_Sox21_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PAG-PPN Pax5 Sox21 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-PPN_Pax5_Sox21_Gaba\\Cluster_PAG-PPN Pax5 Sox21 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PAG-PPN Pax5 Sox21 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-PPN_Pax5_Sox21_Gaba\\Cluster_PAG-PPN Pax5 Sox21 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PAG-RN Nkx2-2 Otx1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PAG-RN Nkx2-2 Otx1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-RN_Nkx2-2_Otx1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-RN_Nkx2-2_Otx1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-RN_Nkx2-2_Otx1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-RN_Nkx2-2_Otx1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-RN_Nkx2-2_Otx1_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PAG-RN Nkx2-2 Otx1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-RN_Nkx2-2_Otx1_Gaba\\Cluster_PAG-RN Nkx2-2 Otx1 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PAG-RN Nkx2-2 Otx1 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-RN_Nkx2-2_Otx1_Gaba\\Cluster_PAG-RN Nkx2-2 Otx1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PAG-SC Neurod2 Meis2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PAG-SC Neurod2 Meis2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-SC_Neurod2_Meis2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-SC_Neurod2_Meis2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Neurod2_Meis2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Neurod2_Meis2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Neurod2_Meis2_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PAG-SC Neurod2 Meis2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Neurod2_Meis2_Glut\\Cluster_PAG-SC Neurod2 Meis2 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PAG-SC Neurod2 Meis2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Neurod2_Meis2_Glut\\Cluster_PAG-SC Neurod2 Meis2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PAG-SC Pou4f1 Zic1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PAG-SC Pou4f1 Zic1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-SC_Pou4f1_Zic1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PAG-SC_Pou4f1_Zic1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Pou4f1_Zic1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Pou4f1_Zic1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Pou4f1_Zic1_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PAG-SC Pou4f1 Zic1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Pou4f1_Zic1_Glut\\Cluster_PAG-SC Pou4f1 Zic1 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PAG-SC Pou4f1 Zic1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PAG-SC_Pou4f1_Zic1_Glut\\Cluster_PAG-SC Pou4f1 Zic1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PG-TRN-LRN Fat2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PG-TRN-LRN Fat2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PG-TRN-LRN_Fat2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PG-TRN-LRN_Fat2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PG-TRN-LRN_Fat2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PG-TRN-LRN_Fat2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PG-TRN-LRN_Fat2_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PG-TRN-LRN Fat2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PG-TRN-LRN_Fat2_Glut\\Cluster_PG-TRN-LRN Fat2 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PG-TRN-LRN Fat2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PG-TRN-LRN_Fat2_Glut\\Cluster_PG-TRN-LRN Fat2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PGRN-PARN-MDRN Hoxb5 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PGRN-PARN-MDRN Hoxb5 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PGRN-PARN-MDRN_Hoxb5_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PGRN-PARN-MDRN_Hoxb5_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PGRN-PARN-MDRN_Hoxb5_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PGRN-PARN-MDRN_Hoxb5_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PGRN-PARN-MDRN_Hoxb5_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PGRN-PARN-MDRN Hoxb5 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PGRN-PARN-MDRN_Hoxb5_Glut\\Cluster_PGRN-PARN-MDRN Hoxb5 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PGRN-PARN-MDRN Hoxb5 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PGRN-PARN-MDRN_Hoxb5_Glut\\Cluster_PGRN-PARN-MDRN Hoxb5 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PH-LHA Foxb1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PH-LHA Foxb1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PH-LHA_Foxb1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PH-LHA_Foxb1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-LHA_Foxb1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-LHA_Foxb1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-LHA_Foxb1_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PH-LHA Foxb1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-LHA_Foxb1_Glut\\Cluster_PH-LHA Foxb1 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PH-LHA Foxb1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-LHA_Foxb1_Glut\\Cluster_PH-LHA Foxb1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PH-SUM Foxa1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PH-SUM Foxa1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PH-SUM_Foxa1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PH-SUM_Foxa1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-SUM_Foxa1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-SUM_Foxa1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-SUM_Foxa1_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PH-SUM Foxa1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-SUM_Foxa1_Glut\\Cluster_PH-SUM Foxa1 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PH-SUM Foxa1 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PH-SUM_Foxa1_Glut\\Cluster_PH-SUM Foxa1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PRC-PAG Pax6 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PRC-PAG Pax6 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRC-PAG_Pax6_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRC-PAG_Pax6_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Pax6_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Pax6_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Pax6_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PRC-PAG Pax6 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Pax6_Glut\\Cluster_PRC-PAG Pax6 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PRC-PAG Pax6 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Pax6_Glut\\Cluster_PRC-PAG Pax6 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PRC-PAG Tcf7l2 Irx2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PRC-PAG Tcf7l2 Irx2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRC-PAG_Tcf7l2_Irx2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRC-PAG_Tcf7l2_Irx2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Tcf7l2_Irx2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Tcf7l2_Irx2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Tcf7l2_Irx2_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_PRC-PAG Tcf7l2 Irx2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Tcf7l2_Irx2_Glut\\Cluster_PRC-PAG Tcf7l2 Irx2 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PRC-PAG Tcf7l2 Irx2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRC-PAG_Tcf7l2_Irx2_Glut\\Cluster_PRC-PAG Tcf7l2 Irx2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PRNc Otp Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PRNc Otp Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRNc_Otp_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRNc_Otp_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRNc_Otp_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRNc_Otp_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRNc_Otp_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PRNc Otp Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRNc_Otp_Gly-Gaba\\Cluster_PRNc Otp Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PRNc Otp Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRNc_Otp_Gly-Gaba\\Cluster_PRNc Otp Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PRT Mecom Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PRT Mecom Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRT_Mecom_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRT_Mecom_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Mecom_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Mecom_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Mecom_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PRT Mecom Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Mecom_Gaba\\Cluster_PRT Mecom Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PRT Mecom Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Mecom_Gaba\\Cluster_PRT Mecom Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PRT Tcf7l2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PRT Tcf7l2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRT_Tcf7l2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PRT_Tcf7l2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Tcf7l2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Tcf7l2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Tcf7l2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PRT Tcf7l2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Tcf7l2_Gaba\\Cluster_PRT Tcf7l2 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 27 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PRT_Tcf7l2_Gaba\\Cluster_PRT Tcf7l2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PVHd-DMH Lhx6 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PVHd-DMH Lhx6 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PVHd-DMH_Lhx6_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PVHd-DMH_Lhx6_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVHd-DMH_Lhx6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVHd-DMH_Lhx6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVHd-DMH_Lhx6_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PVHd-DMH Lhx6 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVHd-DMH_Lhx6_Gaba\\Cluster_PVHd-DMH Lhx6 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PVHd-DMH Lhx6 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVHd-DMH_Lhx6_Gaba\\Cluster_PVHd-DMH Lhx6 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster PVR Six3 Sox3 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster PVR Six3 Sox3 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PVR_Six3_Sox3_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_PVR_Six3_Sox3_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVR_Six3_Sox3_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVR_Six3_Sox3_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVR_Six3_Sox3_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_PVR Six3 Sox3 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVR_Six3_Sox3_Gaba\\Cluster_PVR Six3 Sox3 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_PVR Six3 Sox3 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\PVR_Six3_Sox3_Gaba\\Cluster_PVR Six3 Sox3 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Peri NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Peri NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Peri_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Peri_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Peri_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Peri_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Peri_NN' is writable.\n",
      "    Saved 132 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Peri_NN\\Cluster_Peri NN_vs_Rest_up_significant.csv\n",
      "    Saved 3504 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Peri_NN\\Cluster_Peri NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Pineal Crx Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Pineal Crx Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pineal_Crx_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pineal_Crx_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pineal_Crx_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pineal_Crx_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pineal_Crx_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_Pineal Crx Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pineal_Crx_Glut\\Cluster_Pineal Crx Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_Pineal Crx Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pineal_Crx_Glut\\Cluster_Pineal Crx Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Pvalb Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Pvalb Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pvalb_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pvalb_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_Gaba' is writable.\n",
      "    Saved 2559 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_Gaba\\Cluster_Pvalb Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 1134 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_Gaba\\Cluster_Pvalb Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Pvalb chandelier Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Pvalb chandelier Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pvalb_chandelier_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Pvalb_chandelier_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_chandelier_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_chandelier_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_chandelier_Gaba' is writable.\n",
      "    Saved 725 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_chandelier_Gaba\\Cluster_Pvalb chandelier Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 316 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Pvalb_chandelier_Gaba\\Cluster_Pvalb chandelier Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster RHP-COA Ndnf Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster RHP-COA Ndnf Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_RHP-COA_Ndnf_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_RHP-COA_Ndnf_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\RHP-COA_Ndnf_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\RHP-COA_Ndnf_Gaba' is writable.\n",
      "    Saved 1516 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\RHP-COA_Ndnf_Gaba\\Cluster_RHP-COA Ndnf Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 483 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\RHP-COA_Ndnf_Gaba\\Cluster_RHP-COA Ndnf Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SC Lef1 Otx2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SC Lef1 Otx2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SC_Lef1_Otx2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SC_Lef1_Otx2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC_Lef1_Otx2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC_Lef1_Otx2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC_Lef1_Otx2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SC Lef1 Otx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC_Lef1_Otx2_Gaba\\Cluster_SC Lef1 Otx2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SC Lef1 Otx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC_Lef1_Otx2_Gaba\\Cluster_SC Lef1 Otx2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SC-PAG Lef1 Emx2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SC-PAG Lef1 Emx2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SC-PAG_Lef1_Emx2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SC-PAG_Lef1_Emx2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC-PAG_Lef1_Emx2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC-PAG_Lef1_Emx2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC-PAG_Lef1_Emx2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SC-PAG Lef1 Emx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC-PAG_Lef1_Emx2_Gaba\\Cluster_SC-PAG Lef1 Emx2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SC-PAG Lef1 Emx2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SC-PAG_Lef1_Emx2_Gaba\\Cluster_SC-PAG Lef1 Emx2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SCig Foxb1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SCig Foxb1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCig_Foxb1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCig_Foxb1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCig_Foxb1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCig_Foxb1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCig_Foxb1_Glut' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCig_Foxb1_Glut\\Cluster_SCig Foxb1 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 324 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCig_Foxb1_Glut\\Cluster_SCig Foxb1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SCm-PAG Cdh23 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SCm-PAG Cdh23 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCm-PAG_Cdh23_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCm-PAG_Cdh23_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCm-PAG_Cdh23_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCm-PAG_Cdh23_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCm-PAG_Cdh23_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SCm-PAG Cdh23 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCm-PAG_Cdh23_Gaba\\Cluster_SCm-PAG Cdh23 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SCm-PAG Cdh23 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCm-PAG_Cdh23_Gaba\\Cluster_SCm-PAG Cdh23 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SCs Pax7 Nfia Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SCs Pax7 Nfia Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCs_Pax7_Nfia_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCs_Pax7_Nfia_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCs_Pax7_Nfia_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCs_Pax7_Nfia_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCs_Pax7_Nfia_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SCs Pax7 Nfia Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCs_Pax7_Nfia_Gaba\\Cluster_SCs Pax7 Nfia Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SCs Pax7 Nfia Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCs_Pax7_Nfia_Gaba\\Cluster_SCs Pax7 Nfia Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SCsg Gabrr2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SCsg Gabrr2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCsg_Gabrr2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SCsg_Gabrr2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCsg_Gabrr2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCsg_Gabrr2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCsg_Gabrr2_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SCsg Gabrr2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCsg_Gabrr2_Gaba\\Cluster_SCsg Gabrr2 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SCsg Gabrr2 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SCsg_Gabrr2_Gaba\\Cluster_SCsg Gabrr2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SI-MA-LPO-LHA Skor1 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SI-MA-LPO-LHA Skor1 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SI-MA-LPO-LHA_Skor1_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SI-MA-LPO-LHA_Skor1_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SI-MA-LPO-LHA_Skor1_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SI-MA-LPO-LHA_Skor1_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SI-MA-LPO-LHA_Skor1_Glut' is writable.\n",
      "    Saved 2 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SI-MA-LPO-LHA_Skor1_Glut\\Cluster_SI-MA-LPO-LHA Skor1 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 1444 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SI-MA-LPO-LHA_Skor1_Glut\\Cluster_SI-MA-LPO-LHA Skor1 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SMC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SMC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SMC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SMC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SMC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SMC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SMC_NN' is writable.\n",
      "    Saved 76 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SMC_NN\\Cluster_SMC NN_vs_Rest_up_significant.csv\n",
      "    Saved 432 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SMC_NN\\Cluster_SMC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut' is writable.\n",
      "    Saved 1 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut\\Cluster_SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut_vs_Rest_up_significant.csv\n",
      "    Saved 202 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPA-SPFm-SPFp-POL-PIL-PoT_Sp9_Glut\\Cluster_SPA-SPFm-SPFp-POL-PIL-PoT Sp9 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SPVI-SPVC Sall3 Nfib Gly-Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SPVI-SPVC Sall3 Nfib Gly-Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPVI-SPVC_Sall3_Nfib_Gly-Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPVI-SPVC_Sall3_Nfib_Gly-Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Sall3_Nfib_Gly-Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Sall3_Nfib_Gly-Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Sall3_Nfib_Gly-Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_SPVI-SPVC Sall3 Nfib Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Sall3_Nfib_Gly-Gaba\\Cluster_SPVI-SPVC Sall3 Nfib Gly-Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SPVI-SPVC Sall3 Nfib Gly-Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Sall3_Nfib_Gly-Gaba\\Cluster_SPVI-SPVC Sall3 Nfib Gly-Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SPVI-SPVC Tlx3 Ebf3 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SPVI-SPVC Tlx3 Ebf3 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPVI-SPVC_Tlx3_Ebf3_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SPVI-SPVC_Tlx3_Ebf3_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Tlx3_Ebf3_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Tlx3_Ebf3_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Tlx3_Ebf3_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_SPVI-SPVC Tlx3 Ebf3 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Tlx3_Ebf3_Glut\\Cluster_SPVI-SPVC Tlx3 Ebf3 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_SPVI-SPVC Tlx3 Ebf3 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SPVI-SPVC_Tlx3_Ebf3_Glut\\Cluster_SPVI-SPVC Tlx3 Ebf3 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster STR D1 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster STR D1 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_D1_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_D1_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D1_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D1_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D1_Gaba' is writable.\n",
      "    Saved 224 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D1_Gaba\\Cluster_STR D1 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 44 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D1_Gaba\\Cluster_STR D1 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster STR D2 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster STR D2 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_D2_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_D2_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D2_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D2_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D2_Gaba' is writable.\n",
      "    Saved 200 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D2_Gaba\\Cluster_STR D2 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 151 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_D2_Gaba\\Cluster_STR D2 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster STR Prox1 Lhx6 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster STR Prox1 Lhx6 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_Prox1_Lhx6_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR_Prox1_Lhx6_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_Prox1_Lhx6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_Prox1_Lhx6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_Prox1_Lhx6_Gaba' is writable.\n",
      "    Saved 33 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_Prox1_Lhx6_Gaba\\Cluster_STR Prox1 Lhx6 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 1 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR_Prox1_Lhx6_Gaba\\Cluster_STR Prox1 Lhx6 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster STR-PAL Chst9 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster STR-PAL Chst9 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR-PAL_Chst9_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_STR-PAL_Chst9_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR-PAL_Chst9_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR-PAL_Chst9_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR-PAL_Chst9_Gaba' is writable.\n",
      "    Saved 105 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR-PAL_Chst9_Gaba\\Cluster_STR-PAL Chst9 Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 65 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\STR-PAL_Chst9_Gaba\\Cluster_STR-PAL Chst9 Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster SUB-ProS Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster SUB-ProS Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SUB-ProS_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_SUB-ProS_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SUB-ProS_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SUB-ProS_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SUB-ProS_Glut' is writable.\n",
      "    Saved 4792 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SUB-ProS_Glut\\Cluster_SUB-ProS Glut_vs_Rest_up_significant.csv\n",
      "    Saved 371 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\SUB-ProS_Glut\\Cluster_SUB-ProS Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Sncg Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Sncg Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sncg_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sncg_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sncg_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sncg_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sncg_Gaba' is writable.\n",
      "    Saved 1260 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sncg_Gaba\\Cluster_Sncg Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 681 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sncg_Gaba\\Cluster_Sncg Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Sst Chodl Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Sst Chodl Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sst_Chodl_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sst_Chodl_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Chodl_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Chodl_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Chodl_Gaba' is writable.\n",
      "    Saved 178 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Chodl_Gaba\\Cluster_Sst Chodl Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 36 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Chodl_Gaba\\Cluster_Sst Chodl Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Sst Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Sst Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sst_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Sst_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Gaba' is writable.\n",
      "    Saved 2922 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Gaba\\Cluster_Sst Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 916 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Sst_Gaba\\Cluster_Sst Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster TH Prkcd Grin2c Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster TH Prkcd Grin2c Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_TH_Prkcd_Grin2c_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_TH_Prkcd_Grin2c_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TH_Prkcd_Grin2c_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TH_Prkcd_Grin2c_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TH_Prkcd_Grin2c_Glut' is writable.\n",
      "    Saved 4 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TH_Prkcd_Grin2c_Glut\\Cluster_TH Prkcd Grin2c Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_TH Prkcd Grin2c Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TH_Prkcd_Grin2c_Glut\\Cluster_TH Prkcd Grin2c Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster TRS-BAC Sln Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster TRS-BAC Sln Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_TRS-BAC_Sln_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_TRS-BAC_Sln_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TRS-BAC_Sln_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TRS-BAC_Sln_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TRS-BAC_Sln_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_TRS-BAC Sln Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TRS-BAC_Sln_Glut\\Cluster_TRS-BAC Sln Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_TRS-BAC Sln Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\TRS-BAC_Sln_Glut\\Cluster_TRS-BAC Sln Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Tanycyte NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Tanycyte NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Tanycyte_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Tanycyte_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Tanycyte_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Tanycyte_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Tanycyte_NN' is writable.\n",
      "    No significant UP genes found for Cluster_Tanycyte NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Tanycyte_NN\\Cluster_Tanycyte NN_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_Tanycyte NN_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Tanycyte_NN\\Cluster_Tanycyte NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster VCO Mafa Meis2 Glut vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster VCO Mafa Meis2 Glut.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_VCO_Mafa_Meis2_Glut_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_VCO_Mafa_Meis2_Glut_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VCO_Mafa_Meis2_Glut'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VCO_Mafa_Meis2_Glut'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VCO_Mafa_Meis2_Glut' is writable.\n",
      "    No significant UP genes found for Cluster_VCO Mafa Meis2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VCO_Mafa_Meis2_Glut\\Cluster_VCO Mafa Meis2 Glut_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_VCO Mafa Meis2 Glut_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VCO_Mafa_Meis2_Glut\\Cluster_VCO Mafa Meis2 Glut_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster VLMC NN vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster VLMC NN.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_VLMC_NN_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_VLMC_NN_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VLMC_NN'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VLMC_NN'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VLMC_NN' is writable.\n",
      "    Saved 442 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VLMC_NN\\Cluster_VLMC NN_vs_Rest_up_significant.csv\n",
      "    Saved 4305 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\VLMC_NN\\Cluster_VLMC NN_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster Vip Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster Vip Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Vip_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_Vip_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Vip_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Vip_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Vip_Gaba' is writable.\n",
      "    Saved 594 significant UP genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Vip_Gaba\\Cluster_Vip Gaba_vs_Rest_up_significant.csv\n",
      "    Saved 2632 significant DOWN genes with stats to D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\Vip_Gaba\\Cluster_Vip Gaba_vs_Rest_down_significant.csv\n",
      "    Calculating mean expression for Cluster ZI Pax6 Gaba vs Rest...\n",
      "      Using data from layer: for_DEGs\n",
      "      Added mean expression columns for Cluster ZI Pax6 Gaba.\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ZI_Pax6_Gaba_vs_Rest_top50.csv\n",
      "    Top 50 Cluster marker results saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_ZI_Pax6_Gaba_vs_Rest_top50.xlsx\n",
      "    DEBUG: Attempting to create list_output_dir: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ZI_Pax6_Gaba'\n",
      "    DEBUG: Successfully created/confirmed directory: 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ZI_Pax6_Gaba'\n",
      "    DEBUG: Directory 'D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ZI_Pax6_Gaba' is writable.\n",
      "    No significant UP genes found for Cluster_ZI Pax6 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ZI_Pax6_Gaba\\Cluster_ZI Pax6 Gaba_vs_Rest_up_significant.csv\n",
      "    No significant DOWN genes found for Cluster_ZI Pax6 Gaba_vs_Rest. Empty CSV file with header created: D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\sig_deg_lists\\ZI_Pax6_Gaba\\Cluster_ZI Pax6 Gaba_vs_Rest_down_significant.csv\n",
      "\n",
      "Summary of skipped clusters (markers for 'mapmycells_second_layer') saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\dge_cluster_comparison_skipped_mapmycells_second_layer.csv\n",
      "\n",
      "Cluster Comparison DGE analysis (Marker identification) finished.\n",
      "\n",
      "Generating marker plot for 'mapmycells_second_layer' clusters...\n",
      "WARNING: saving figure to file D:\\Github\\SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\rank_genes_groups_mapmycells_second_layer_cluster_markers_mapmycells_second_layer.png\n",
      "  Marker plot saved to D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\\rank_genes_groups_cluster_markers_mapmycells_second_layer.png\n"
     ]
    }
   ],
   "source": [
    "# Cell Type Comparison DGE (Each cell type vs Rest) using the 'for_DEGs' layer, using the DEG_BY\n",
    "print(f\"\\nRunning Cell Type Comparison DGE (Markers for {DEG_BY})...\")\n",
    "dge_mapmycells_second_layer_markers = functions_degs.run_cluster_comparison_dge(\n",
    "    adata,\n",
    "    grouping_key=DEG_BY, \n",
    "    dge_output_dir=BIOMARKER_OUTPUT_DIR, \n",
    "    plot_output_dir=BIOMARKER_OUTPUT_DIR,\n",
    "    layer='for_DEGs' if 'for_DEGs' in adata.layers else None,\n",
    "    method='wilcoxon' # or 't-test'\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d932115c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "All analysis and output generation complete.\n",
      "Overall/Genotype DGE outputs saved in: D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\dge_res\n",
      "Biomarker DGE outputs saved in: D:/Github/SRF_Linda_RNA\\combine_data\\results_from_raw\\DEGs_mapmycells_L2FC_0_25\\biomarkers\n"
     ]
    }
   ],
   "source": [
    "print(\"\\nAll analysis and output generation complete.\")\n",
    "print(f\"Overall/Genotype DGE outputs saved in: {DGE_OUTPUT_DIR}\")\n",
    "print(f\"Biomarker DGE outputs saved in: {BIOMARKER_OUTPUT_DIR}\")"
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "cell_metadata_filter": "-all",
   "main_language": "python",
   "notebook_metadata_filter": "-all"
  },
  "kernelspec": {
   "display_name": "bioinf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
